<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Bayes multiparameter models | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Bayes multiparameter models" /><meta name="author" content="Aki Vehtari, Markus Paasiniemi, Dylan Lawless" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/src/bayesian_demo3_1_4.Rmd" /><meta property="og:url" content="http://localhost:4000/src/bayesian_demo3_1_4.Rmd" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Bayes multiparameter models" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","author":{"@type":"Person","name":"Aki Vehtari, Markus Paasiniemi, Dylan Lawless"},"description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Bayes multiparameter models","url":"http://localhost:4000/src/bayesian_demo3_1_4.Rmd"}</script> <script type="text/javascript" defer src="/assets/js/mathjax-script-type.js"> </script> <script type="text/javascript" id="MathJax-script" defer src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"> </script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/storage.html" class="nav-list-link">storage</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"> <h# Bayesian: Multiparameter models Last update: ```{r, date, echo = FALSE} print(Sys.Date()) ``` This doc was built with: `rmarkdown::render("Bayesian_example1.Rmd", output_file = "../pages/bayesian_example1.md")` ```{r setup, echo=FALSE, warning=FALSE, message=FALSE} knitr::opts_chunk$set(echo = TRUE, fig.path='../assets/images/') library(ggplot2); theme_set(theme_bw()) library(grid) library(gridExtra) library(tidyr) ``` This doc was built with: `rmarkdown::render("bayesian_demo3_1_4.Rmd", output_file = "../pages/bayesian_multiparameter_models.md")` This example is described in the textbook: Bayesian Data Analysis, by Andrew Gelman, John Carlin, Hal Stern, David Dunson, Aki Vehtari, and Donald Rubin. Third edition, (BDA3),<http://www.stat.columbia.edu/~gelman/book/>, chapter 3. The code is based on a version by Aki Vehtari and Markus Paasiniemi. - **Section 3.2**: Discusses the normal distribution with unknown mean and variance. - **Sections 3.4 and 3.5**: Address inference for the multinomial and multivariate normal distributions, the simplest models for discrete and continuous multivariate data, respectively. - **Conclusion**: The chapter concludes with an examination of a nonconjugate logistic regression model, implemented via numerical computation of the posterior density on a grid. ## Introduction to nuisance parameters in Bayesian statistics In the realm of statistics, nearly every practical problem encompasses multiple unknown or unobservable quantities. The simplicity of the Bayesian approach shines when managing such complexities, offering clear advantages over alternative methods of inference. Often, although a problem may present several parameters of interest, attention is usually focused on just one or a few. {: .important-title } > Definitions > > > **Marginal posterior distribution**: The probability distribution of the parameters of interest, obtained by integrating out the nuisance parameters from the joint posterior distribution. > > **Nuisance parameters**: Parameters included in the model to enhance its realism and validity, though they are not the primary focus of analysis. > > **Joint posterior distribution**: The probability distribution representing the likelihood of all parameter values given the observed data. ### Aim of Bayesian analysis The primary goal of Bayesian analysis is to determine the marginal posterior distribution of the parameters of interest. This is achieved through a two-step process: 1. **Obtaining the joint posterior distribution**: Initially, the complete joint posterior distribution encompassing all unknowns is required. 2. **Integrating over nuisance parameters**: Subsequently, this distribution is integrated over the parameters that are not of immediate interest, yielding the desired marginal distribution. $$ p(\theta_1 \mid y) = \int p(\theta_1, \theta_2 \mid y) \, d\theta_2 $$ Alternatively, by using simulation: 3. **Simulation sampling**: Samples are drawn from the joint posterior distribution, concentrating on the parameters of interest and disregarding the values of other unknowns. ### Nuisance parameters Frequently in statistical problems, many parameters, although essential for constructing a realistic model, are not the focus of inference. These parameters are typically referred to as *nuisance parameters*. **Example**: An exemplary instance of nuisance parameters is the scale of random errors in measurement problems. ## Averaging over ‘Nuisance Parameters’ {: .important-title } > Explanation of symbols > > > $$\theta$$: Total parameters of interest, split into two components $$\theta_1$$ and $$\theta_2$$. > > $$\theta_1$$ and $$\theta_2$$: $$\theta_1$$ represents parameters of primary interest, while $$\theta_2$$ represents nuisance parameters. > > $$y$$: Observed data. > > $$\mu$$ and $$\sigma^2$$: Parameters commonly used in normal distribution models, where $$\mu$$ is the mean and $$\sigma^2$$ is the variance. > > $$p(\theta_1 \mid y)$$: Marginal posterior distribution of $$\theta_1$$ given data $$y$$. > > $$p(\theta_1, \theta_2 \mid y)$$: Joint posterior distribution of $$\theta_1$$ and $$\theta_2$$ given data $$y$$. > > $$\mathcal{N}(\mu, \sigma^2)$$: Normal distribution with mean $$\mu$$ and variance $$\sigma^2$$. To express the ideas of joint and marginal posterior distributions mathematically, suppose $$\theta$$ has two parts, each of which can be a vector, $$\theta = (\theta_1, \theta_2)$$, and further suppose that we are only interested (at least for the moment) in inference for $$\theta_1$$, so $$\theta_2$$ may be considered a ‘nuisance’ parameter. For instance, in the simple example, $$ y \mid \mu, \sigma^2 \sim \mathcal{N}(\mu, \sigma^2), $$ in which both $$\mu$$ (equivalent to $$\theta_1$$) and $$\sigma^2$$ (equivalent to $$\theta_2$$) are unknown, interest commonly centers on $$\mu$$. We seek the conditional distribution of the parameter of interest given the observed data; in this case, $$p(\theta_1 \mid y)$$. This is derived from the joint posterior density, $$ p(\theta_1, \theta_2 \mid y) \propto p(y \mid \theta_1, \theta_2)p(\theta_1, \theta_2), $$ by averaging over $$\theta_2$$: $$ p(\theta_1 \mid y) = \int p(\theta_1, \theta_2 \mid y) \, d\theta_2. $$ Alternatively, the joint posterior density can be factored to yield $$ \begin{equation} p(\theta_1 \mid y) = \int Z p(\theta_1 \mid \theta_2, y) p(\theta_2 \mid y) d\theta_2, \tag{1}\label{eq:one} \end{equation} $$ which shows that the posterior distribution of interest, $$p(\theta_1 \mid y)$$, is a mixture of the conditional posterior distributions given the nuisance parameter, $$\theta_2$$, where $$p(\theta_2 \mid y)$$ is a weighting function for the different possible values of $$\theta_2$$. The weights depend on the posterior density of $$\theta_2$$ and thus on a combination of evidence from data and prior model. The averaging over nuisance parameters $$\theta_2$$ can be interpreted generally; for example, $$\theta_2$$ can include a discrete component representing different possible sub-models. We rarely evaluate the integral (eqn \\ref{eq:one}) explicitly, but it suggests an important practical strategy for both constructing and computing with multiparameter models. Posterior distributions can be computed by marginal and conditional simulation, first drawing $$\theta_2$$ from its marginal posterior distribution and then $$\theta_1$$ from its conditional posterior distribution, given the drawn value of $$\theta_2$$. In this way, the integration embodied in (eqn \\ref{eq:one}) is performed indirectly. A canonical example of this form of analysis is provided by the normal model with unknown mean and variance, to which we now turn. ### Explanation of averaging over nuisance parameters Let's break down the equation to show how the joint posterior distribution is constructed from the likelihood and the prior distribution. #### Objective To derive the marginal posterior distribution of $$\theta_1$$ by integrating out $$\theta_2$$, considered a nuisance parameter, from the joint posterior distribution. #### Step-by-step breakdown 1. **Joint posterior distribution**: - **Equation**: $$ p(\theta_1, \theta_2 \mid y) \propto p(y \mid \theta_1, \theta_2) \times p(\theta_1, \theta_2) $$ - **Explanation**: This formula shows the joint posterior distribution as a product of the likelihood of the data given the parameters and the prior beliefs about these parameters. The term "proportional to" indicates that after considering the likelihood and prior, the result must be normalised to ensure it sums to one across all possible values of $$\theta_1$$ and $$\theta_2$$. 2. **Visual breakdown**: - **Prior layer**: Represents initial beliefs about the parameters before seeing the data. - **Likelihood layer**: Adjusts beliefs based on how likely the observed data is under various parameter values. - **Normalisation**: Ensures the total probability across all parameter values equals one. 3. **Marginal posterior distribution**: - **Equation**: $$ p(\theta_1 \mid y) = \int p(\theta_1, \theta_2 \mid y) \, d\theta_2 $$ - **Explanation**: By integrating out $$\theta_2$$, this operation focuses solely on $$\theta_1$$, summing over all potential influences of $$\theta_2$$ to isolate the effect on $$\theta_1$$. This represents averaging over all possible values of $$\theta_2$$, which adjusts $$\theta_1$$'s distribution to reflect its combined impact. 4. **Factorisation and practical computation**: - **Equation**: $$ p(\theta_1 \mid y) = \int Z p(\theta_1 \mid \theta_2, y) p(\theta_2 \mid y) d\theta_2 $$ - **Components**: - $$p(\theta_1 \mid \theta_2, y)$$: The conditional posterior of $$\theta_1$$ given $$\theta_2$$ and data. - $$p(\theta_2 \mid y)$$: Acts as a weighting function in the integration, representing the marginal posterior of $$\theta_2$$. - **Explanation**: This factorisation views the marginal posterior as a weighted average of conditional posteriors, emphasising a practical simulation-based method to handle complex integrations. 5. **Simulation as a practical solution**: - **Procedure**: - **Draw $$\theta_2$$**: From its marginal posterior distribution. - **Draw $$\theta_1$$**: From the conditional distribution given the sampled $$\theta_2$$. - **Explanation**: This simulation effectively performs the integration by sampling, which is useful when direct computation is infeasible or impractical. To visualise how these components come together, you can think of it as layers: - **Base layer (Prior distribution)**: Start with your initial beliefs about the parameters (before seeing the data). - **Middle layer (Likelihood)**: Adjust these beliefs by the likelihood of the observed data under various parameter values. - **Top layer (Normalisation)**: Scale the result so that the total probability across all parameter values equals 1. ### Step-by-step visualization The goal of these visualizations is to explore the effects of the nuisance parameter $$\theta_2$$ on the distribution of $$\theta_1$$. We first observe the overall distribution of $$\theta_2$$ and then focus on how specific values influence $$\theta_1$$. #### Step 1: Visualizing the distribution of $$\theta_2$$ We start by drawing a histogram to understand the typical range and distribution of $$\theta_2$$. This visualization helps identify the common values and the spread of $$\theta_2$$. For detailed analysis, we specifically highlight the bar at $$x = 1$$ to observe its frequency and to mark it for further scrutiny. ```{r} # Simulate some data set.seed(123) data <- data.frame( theta1 = rnorm(10000), theta2 = rnorm(10000) ) ``` ```{r, echo=FALSE, warning=FALSE, message=FALSE} # Assuming 'data' is already loaded # Plot the distributions p_theta1 <- ggplot(data, aes(x = theta1)) + geom_histogram(binwidth = 0.1, fill = "#71BAD2", alpha = 0.7) + labs(title = "Distribution of theta1\n") p_theta2 <- ggplot(data, aes(x = theta2)) + geom_histogram(binwidth = 0.1, fill = "#FFC387", alpha = 0.7) + labs(title = "Distribution of theta2\n(Nuisance Parameter)") # Not shown # p_theta2 # Not shown library(patchwork) print(p_theta1 + p_theta2) ``` ```{r,echo=FALSE, warning=FALSE, message=FALSE} # Calculate histogram data # To use `geom_bar()`, you need to pre-calculate the histogram counts. This method gives you more control over the appearance of individual bars, such as highlighting a specific bar at $$x=0$$. # Assuming 'data' is already loaded with theta2 # Calculate histogram bins manually binwidth <- 0.1 theta2_bins <- data %>% mutate(bin = floor(theta2 / binwidth) * binwidth) %>% group_by(bin) %>% summarise(count = n(), .groups = 'drop') # Draw the histogram with geom_bar p_theta2_bar <- ggplot(theta2_bins, aes(x = bin, y = count)) + geom_bar(stat = "identity", fill = ifelse(theta2_bins$bin == 1, "red", "#FFC387"), alpha = 0.7) + labs(title = "Distribution of theta2 (Nuisance parameter) with highlight on x=1", x = "theta2") p_theta2_bar ``` #### Step 2: Examining the effect of a highlighted $$\theta_2$$ on $$\theta_1$$ Next, we demonstrate how varying $$\theta_2$$ affects $$\theta_1$$ by creating conditional histograms. We highlight the range around $$x = 1$$ to see how $$\theta_1$$ behaves when $$\theta_2$$ is near this specific value. This step visualizes the conditional effects of $$\theta_2$$ on $$\theta_1$$ across different slices, with a particular focus on the highlighted region. ```{r, echo=FALSE, warning=FALSE, message=FALSE} # Define the range of theta2 to highlight around 1 highlight_range <- c(0.95, 1.05) # Narrow range around 1 # Enhance data with a slicing variable, and compute the minimum theta2 value per slice for ordering data <- data %>% mutate( slice = cut(theta2, breaks = 12, include.lowest = TRUE) ) %>% group_by(slice) %>% mutate(min_theta2 = min(theta2)) %>% ungroup() %>% arrange(min_theta2) %>% # Convert slice to an ordered factor based on the sorted min_theta2 mutate(slice = factor(slice, levels = unique(slice))) # Create conditional histograms of theta1 p_conditional_highlight <- ggplot(data, aes(x = theta1)) + # Histogram for values outside the highlighted range (blue) geom_histogram(data = subset(data, theta2 < highlight_range[1] | theta2 > highlight_range[2]), binwidth = 0.1, fill = "#71BAD2", alpha = 0.9) + # Histogram for values within the highlighted range (red) geom_histogram(data = subset(data, theta2 >= highlight_range[1] & theta2 <= highlight_range[2]), binwidth = 0.1, fill = "red", alpha = 0.8) + facet_wrap(~slice, scales = "free_y") + labs(title = "Conditional distributions of theta1 given slices of theta2", subtitle = "Highlighted range includes theta2 around x=1") p_conditional_highlight ``` Integrating out $$\theta_2$$ in Bayesian analysis is about marginalizing over this parameter to refine our estimates for $$\theta_1$$. This involves averaging over all possible values of $$\theta_2$$, where each value is weighted by its likelihood: - **Highlighting an example value**: By highlighting $$x = 1$$ in the visualizations, we're not limiting our analysis to this point but rather using it to exemplify how specific values within the range of $$\theta_2$$ can influence the outcome for $$\theta_1$$. It serves as a focal point for comparison. - **Complete integration**: The full integration over $$\theta_2$$ considers every part of its distribution. Each histogram slice contributes to a comprehensive understanding of how $$\theta_1$$ is influenced by the entire spectrum of $$\theta_2$$. ## Normal model with unknown mean and variance (BDA3 section 3.2 on p. 64). Multivariate joint distribution, conditional distribution, marginal distribution, marginalization and posterior predictive distribution #### Generic part common for Demos 3.1-3.4 Data ```{r } y <- c(93, 112, 122, 135, 122, 150, 118, 90, 124, 114) ``` Sufficient statistics ```{r } n <- length(y) s2 <- var(y) my <- mean(y) ``` Factorize the joint posterior p(mu,sigma2|y) to p(sigma2|y)p(mu|sigma2,y) Sample from the joint posterior using this factorization ```{r } # helper functions to sample from and evaluate # scaled inverse chi-squared distribution rsinvchisq <- function(n, nu, s2, ...) nu*s2 / rchisq(n , nu, ...) dsinvchisq <- function(x, nu, s2){ exp(log(nu/2)*nu/2 - lgamma(nu/2) + log(s2)/2*nu - log(x)*(nu/2+1) - (nu*s2/2)/x) } ``` Sample 1000 random numbers from p(sigma2|y) ```{r } ns <- 1000 sigma2 <- rsinvchisq(ns, n-1, s2) ``` Sample from p(mu|sigma2,y) ```{r } mu <- my + sqrt(sigma2/n)*rnorm(length(sigma2)) ``` Create a variable sigma and sample from predictive distribution p(ynew|y) for each draw of (mu, sigma) ```{r } sigma <- sqrt(sigma2) ynew <- rnorm(ns, mu, sigma) ``` For mu, sigma and ynew compute the density in a grid ranges for the grids ```{r } t1l <- c(90, 150) t2l <- c(10, 60) nl <- c(50, 185) t1 <- seq(t1l[1], t1l[2], length.out = ns) t2 <- seq(t2l[1], t2l[2], length.out = ns) xynew <- seq(nl[1], nl[2], length.out = ns) ``` Compute the exact marginal density of mu ```{r } # multiplication by 1./sqrt(s2/n) is due to the transformation of # variable z=(x-mean(y))/sqrt(s2/n), see BDA3 p. 21 pm <- dt((t1-my) / sqrt(s2/n), n-1) / sqrt(s2/n) ``` Estimate the marginal density using samples and ad hoc Gaussian kernel approximation ```{r } pmk <- density(mu, adjust = 2, n = ns, from = t1l[1], to = t1l[2])$y ```<hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
