<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Home | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Home" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/" /><meta property="og:url" content="http://localhost:4000/" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Home" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebSite","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Home","name":"Dev docs","url":"http://localhost:4000/"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item active"><a href="/" class="nav-list-link active">Home</a><li class="nav-list-item"><a href="/pages/altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/storage.html" class="nav-list-link">storage</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><h1 style="display:center;"> <img style="float: left; padding-top: 1px" src="/assets/images/logos/SwissPedHealth_Pipeline_Devs.png" alt="Pipe dev logo image" width="70" /> Dev docs</h1><hr /><h2 id="welcome"> <a href="#welcome" class="anchor-heading" aria-labelledby="welcome"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Welcome</h2><p>This is the documentation index for bioinformatic pipeline development. You can visit the project home page at <a href="https://switzerlandomics.ch">https://switzerlandomics.ch</a>.</p><h2 id="quick-start"> <a href="#quick-start" class="anchor-heading" aria-labelledby="quick-start"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Quick start</h2><p>Most likely, you are looking for one of our design documents. Check the <strong>side bar</strong> or go here: <a href="pages/design_doc.html">Index page for design documents</a>. These pages describe the processes for our pipelines, start to finish, with links to supporting information.</p><h2 id="latest-updates"> <a href="#latest-updates" class="anchor-heading" aria-labelledby="latest-updates"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Latest updates</h2><p>See the <a href="/pages/platform_updates.html">platform release updates</a> for the most recent features.</p><h2 id="full-site-navigation"> <a href="#full-site-navigation" class="anchor-heading" aria-labelledby="full-site-navigation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Full site navigation</h2><div class="nav-columns specific-grid"><ul><li><a href="/404">404</a><li><a href="/pages/Phoenix_Sepsis_Score_logic.html">Sepsis score: Phoenix</a><li><a href="/pages/acat.html">ACAT</a><li><a href="/pages/acmg_criteria_table_main.html">ACMG criteria</a><li><a href="/pages/aggregate_multiplex.html">Aggregate multiplexed data</a><li><a href="/pages/altman_bland_analysis_of_methods.html">Stats Analysis of methods</a><li><a href="/pages/altman_bland_ci_from_p.html">Stats CI from P</a><li><a href="/pages/altman_bland_correlation.html">Stats Correlation, regression and repeated data</a><li><a href="/pages/altman_bland_odds_ratios.html">Stats Odds ratios, SE &amp; CI</a><li><a href="/pages/altman_bland_roc_curve.html">Stats Receiver operating characteristic plots</a><li><a href="/pages/altman_bland_sensitivity_specificity.html">Stats Sensitivity and specificity</a><li><a href="/pages/annotation_table.html">Annotation table</a><li><a href="/pages/bam2fastq.html">BWA</a><li><a href="/pages/bayesian_example1.html">Bayes 1 probability in placenta previa</a><li><a href="/pages/bayesian_example2.html">Bayesian 2 probability in placenta previa</a><li><a href="/pages/bayesian_genetic_carrier.html">Bayesian discrete probability example in genetics</a><li><a href="/pages/bayesian_mcmc_samplers.html">Bayes MCMC samplers</a><li><a href="/pages/bayesian_multiparameter_bioassay.html">Bayes multiparameter bioassay demo</a><li><a href="/pages/bayesian_multiparameter_models.html">Bayes multiparameter models</a><li><a href="/pages/benchmark_pipelines.html">Benchmarking pipeline output</a><li><a href="/pages/bevimed.html">BeviMed</a><li><a href="/pages/biomedit.html">BioMedIT</a><li><a href="/pages/bookmarks.html">Bookmarks</a><li><a href="/pages/bwa.html">BWA</a><li><a href="/pages/causal_inference_stats.html">Causal inference stats</a><li><a href="/pages/causal_inference_whole_game.html">Causal inference mosquito nets</a><li><a href="/pages/data_concepts.html">Data concepts</a><li><a href="/pages/data_stream.html">Data stream</a><li><a href="/pages/design_PCA_SNV_INDEL_V1.html">Design PCA SNV INDEL v1</a><li><a href="/pages/design_dna_snvindel_v1.html">Design DNA SNV INDEL v1</a><li><a href="/pages/design_dna_snvindel_v1_release.html">Design release DNA SNV INDEL v1</a><li><a href="/pages/design_dna_snvindel_v2_release.html">Design release DNA SNV INDEL v2</a><li><a href="/pages/design_doc.html">Design documents</a><li><a href="/pages/design_qv_snvindel_v1.html">Design QV SNV INDEL v1</a><li><a href="/pages/design_statistical_genomics_v1.html">Design statistical genomics v1</a><li><a href="/pages/dna_annotation.html">DNA annotation</a><li><a href="/pages/dna_interpretation.html">DNA interpretation</a><li><a href="/pages/dna_qc.html">DNA QC</a><li><a href="/pages/docker_singularity.html">Docker with singularity</a><li><a href="/pages/documentation_log.html">Documentation log</a><li><a href="/pages/fastp.html">FASTP</a><li><a href="/pages/fastq.html">FASTQ format data</a><li><a href="/pages/filter_vcf_bcftools.html">Filter VCF with bcftools</a><li><a href="/pages/gatk_bsqr.html">GATK BQSR</a><li><a href="/pages/gatk_dbimport.html">GATK Genomic db import</a><li><a href="/pages/gatk_duplicates.html">GATK Duplicates</a><li><a href="/pages/gatk_genotypegvcf.html">GATK Genotyping gVCFs</a><li><a href="/pages/gatk_genotyperefine.html">GATK Genotype refine</a><li><a href="/pages/gatk_hc.html">GATK Haplotype caller</a><li><a href="/pages/gatk_vqsr.html">GATK VQSR</a><li><a href="/pages/guru.html">Guru variant interpretation</a><li><a href="/pages/gwas.html">GWAS analysis</a><li><a href="/pages/hpc.html">HPC infrastructure</a><li><a href="/">Home</a><li><a href="/pages/inf_causal_metab_sem.html">Inference of causal metabolite networks</a><li><a href="/pages/layout.html">Layout</a><li><a href="/pages/mathjax.html">MathJax config</a><li><a href="/pages/mbdf_models.html">MBDF models</a><li><a href="/pages/mbdf_supervised.html">MBDF supervised</a><li><a href="/pages/mbdf_unsupervised.html">MBDF unsupervised</a><li><a href="/pages/metadata.html">WGS metadata</a><li><a href="/pages/metrics_bcftoolscounts.html">Metrics Bcftools counts</a><li><a href="/pages/metrics_bcftoolsstats.html">Metrics Bcftools stats</a><li><a href="/pages/metrics_collectwgsmetrics.html">Metrics CollectWgsMetrics</a><li><a href="/pages/multiblock_data_fusion.html">Multiblock data fusion</a><li><a href="/pages/panels_disease.html">Panels disease gene</a><li><a href="/pages/pca_biplot_1kg.html">PCA biplot 1000genomes</a><li><a href="/pages/pca_features.html">PCA features</a><li><a href="/pages/platform_updates.html">Platform updates</a><li><a href="/pages/pre_anno_maf.html">Pre-annotation MAF</a><li><a href="/pages/pre_annoprocess.html">Pre-annotation processing</a><li><a href="/pages/precision_med.html">Precision Medicine Unit (PMU)</a><li><a href="/pages/present/presentations.html">Presentations</a><li><a href="/pages/read_group.html">Read group</a><li><a href="/pages/ref.html">Reference genome</a><li><a href="/pages/rl_finte_mdp.html">RL finite MDP</a><li><a href="/pages/slurm_manager.html">SLURM monitoring</a><li><a href="/pages/slurm_sbatch.html">SLURM sbatch headers</a><li><a href="/pages/storage.html">storage</a><li><a href="/pages/style.html">Style page guide</a><li><a href="/pages/style_writing.html">Style writing guide</a><li><a href="/pages/sv.html">Structural variation detection</a><li><a href="/pages/synth_data.html">Synthetic data</a><li><a href="/pages/variables.html">Variables</a><li><a href="/pages/variant_concept.html">Variant to RDF concept</a><li><a href="/pages/vcf.html">VCF - Variant Call Format</a><li><a href="/pages/vcf_gvcf.html">VCF and gVCF</a><li><a href="/pages/virtual_panels.html">Virtual gene panels</a><li><a href="/pages/vsat.html">VSAT with SKAT</a><li><a href="/pages/vsat_setID.html">SetID for VSAT</a></ul></div><h2 id="about"> <a href="#about" class="anchor-heading" aria-labelledby="about"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> About</h2><h3 id="general"> <a href="#general" class="anchor-heading" aria-labelledby="general"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> General</h3><p>Please contact us if you find any documentation that requires updates. We aim to include information about tools, reference datasets, databases, and any other component used in our pipelines besides sensitive and protected personal information. We will not include README information about pipeline source code here - this information should be found in the individual project repositories. However, should any stable pipelines become routinely implemented, we are likely add a user guide here.</p><h3 id="content"> <a href="#content" class="anchor-heading" aria-labelledby="content"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Content</h3><p>This documentation is made to track the public progress of pipeline development. It includes information about the pipeline branches, for example DNA single variant detection, DNA CNV detection, RNA quantitative expression, etc. Many features are common to different branches (e.g. reference databases, reference datasets) are therefore listed separately form branch-specific content.</p><p>We rely on accurate reference data sources and therefore each component is tracked with identification information such as source location (URL, citation, etc), date, checksum, file size, contributor ID, etc. We maintain a database to ensure accurate tracking to reduce duplication or mislabelling. This database is to be used for automated variable generation, reporting, and cross-project consistency. We aim to maintain this database in our public git repository to promote data integrity and open science.</p><h3 id="data-protection"> <a href="#data-protection" class="anchor-heading" aria-labelledby="data-protection"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data protection</h3><p>No research data or personal information will be included in public documentation. Sensitive variables will only be stored within private offline project repositories. We use a public development repository to ensure a clear separation of infrastructure development and private patient data.</p><h2 id="contributors"> <a href="#contributors" class="anchor-heading" aria-labelledby="contributors"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Contributors</h2><ol><li>The switzerlandomics organisation on GitHub: <a href="https://github.com/switzerlandomics">github.com/switzerlandomics</a>.<li>This documentation repository is hosted on our GitHub organisation under <code class="language-plaintext highlighter-rouge">docs</code>: <a href="https://github.com/docs-switzerlandomics.github.io">github.com/docs-switzerlandomics.github.io</a>.</ol><ul><li>You can request to become a member of the organisation via GitHub.<li>You can make pull requests to the <code class="language-plaintext highlighter-rouge">docs</code> repository.<li>Alternatively, you can email us with comments directly.</ul><h2 id="documentation-style"> <a href="#documentation-style" class="anchor-heading" aria-labelledby="documentation-style"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Documentation style</h2><p>This site is built using the <em>bare-minimum</em> template from the “Just the Docs” theme. It uses Jekyll to build the static site which is then hosted on GitHub pages (or hosted from any other server).</p><p>If Jekyll is installed on your computer, you can also build and preview the created site <em>locally</em>. This lets you test changes before committing them, and avoids waiting for GitHub Pages.<sup id="fnref:1"><a href="#fn:1" class="footnote" rel="footnote" role="doc-noteref">1</a></sup> And you will be able to deploy your local build to a different platform than GitHub Pages.</p><p>We currently allow GitHub pages to rebuild the site using Jekyll. We have also tested the method to push the pre-built <code class="language-plaintext highlighter-rouge">_site</code>, should additional custom plugins be required. However, we aim to rely on minimum complexity.</p><p>More specifically, the created site:</p><ul><li>uses a gem-based approach, i.e. uses a <code class="language-plaintext highlighter-rouge">Gemfile</code> and loads the <code class="language-plaintext highlighter-rouge">just-the-docs</code> gem<li>uses the <a href="https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/">GitHub Pages / Actions workflow</a> to build and publish the site on GitHub Pages</ul><p><a href="https://just-the-docs.github.io/just-the-docs/">Browse the theme documentation</a> to learn more about how to use this theme.</p><p>You can read about how to maintain docs in the <code class="language-plaintext highlighter-rouge">docs</code> directory of an existing project repo, see <a href="https://github.com/just-the-docs/just-the-docs-template/blob/main/README.md#hosting-your-docs-from-an-existing-project-repo">Hosting your docs from an existing project repo</a> in the template README. However, this <code class="language-plaintext highlighter-rouge">docs</code> repo is a standalone repository within the SwitzerlandOmics github orgnaisation.</p><p class="note">We can use the special styles for <code class="language-plaintext highlighter-rouge">{: .note }</code> and <code class="language-plaintext highlighter-rouge">{: .warning }</code> by adding this code before a paragraph.</p><h2 id="change-log"> <a href="#change-log" class="anchor-heading" aria-labelledby="change-log"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Change log</h2><p>The change log and (intermittently) the page count is tracked in the <a href="pages/documentation_log.html">documentation log page</a>.</p><ul><li>Git log for 2024 contains 90 entries and is saved to <code class="language-plaintext highlighter-rouge">gitlog_2024.txt</code><li>Git log for 2023 contains 49 entries and is saved to <code class="language-plaintext highlighter-rouge">gitlog_2023.txt</code></ul><h3 id="growth"> <a href="#growth" class="anchor-heading" aria-labelledby="growth"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Growth</h3><p><img src="/assets/images/file_count_plot-1.png" alt="img" /></p><p><img src="/assets/images/logos/SwissPedHealth_Pipeline_Devs_wide_2.png" alt="img" /></p><hr /><div class="footnotes" role="doc-endnotes"><ol><li id="fn:1"><p><a href="https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll#creating-your-site">It can take up to 10 minutes for changes to your site to publish after you push the changes to GitHub</a>. <a href="#fnref:1" class="reversefootnote" role="doc-backlink">&#8617;</a></p></ol></div><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
