<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>ACMG criteria | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="ACMG criteria" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/acmg_criteria_table_main.html" /><meta property="og:url" content="http://localhost:4000/pages/acmg_criteria_table_main.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="ACMG criteria" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"ACMG criteria","url":"http://localhost:4000/pages/acmg_criteria_table_main.html"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/storage.html" class="nav-list-link">storage</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item active"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link active">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><p>Last update: 20220131 Last update: 20241205</p><details open="" class="no_toc"> <summary class="text-delta">Table of contents</summary><ul id="markdown-toc"><li><a href="#interpretation-of-variants-by-acmg-standards-and-guidelines" id="markdown-toc-interpretation-of-variants-by-acmg-standards-and-guidelines">Interpretation of variants by ACMG standards and guidelines</a><ul><li><a href="#classification-criteria" id="markdown-toc-classification-criteria">Classification criteria</a><ul><li><a href="#main-criteria-for-classifications" id="markdown-toc-main-criteria-for-classifications">Main criteria for classifications</a><li><a href="#caveats-implementing-filters" id="markdown-toc-caveats-implementing-filters">Caveats implementing filters</a></ul><li><a href="#scoring-point-system" id="markdown-toc-scoring-point-system">Scoring point system</a><li><a href="#acmguru-code-example" id="markdown-toc-acmguru-code-example">ACMGuru code example</a><li><a href="#references" id="markdown-toc-references">References</a></ul></ul></details><hr /><h1 id="interpretation-of-variants-by-acmg-standards-and-guidelines"> <a href="#interpretation-of-variants-by-acmg-standards-and-guidelines" class="anchor-heading" aria-labelledby="interpretation-of-variants-by-acmg-standards-and-guidelines"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Interpretation of variants by ACMG standards and guidelines</h1><h2 id="classification-criteria"> <a href="#classification-criteria" class="anchor-heading" aria-labelledby="classification-criteria"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Classification criteria</h2><p>Extensive annotation is applied during our genomics analysis. Interpretation of genetic determinants of disease is based on many evidence sources. One important source of interpretation comes from the <em>Standards and guidelines for the interpretation of sequence variants: a joint consensus recommendation of the American College of Medical Genetics and Genomics and the Association for Molecular Pathology</em>, Richards et al. <sup id="fnref:richards2015standards"><a href="#fn:richards2015standards" class="footnote" rel="footnote" role="doc-noteref">1</a></sup>. See also Li et al., 2017 <sup id="fnref:li2017standards"><a href="#fn:li2017standards" class="footnote" rel="footnote" role="doc-noteref">2</a></sup> and Riggs et al., 2020 <sup id="fnref:riggs2020technical"><a href="#fn:riggs2020technical" class="footnote" rel="footnote" role="doc-noteref">3</a></sup>. The following tables are provided as they appear in the initial steps of our filtering protocol for the addition of ACMG-standardised labels to candidate causal variants.</p><p>For reference, alternative public implementations of ACMG guidelines can be found in Li et al., 2017 <sup id="fnref:li2017intervar"><a href="#fn:li2017intervar" class="footnote" rel="footnote" role="doc-noteref">4</a></sup> and Xavier et al., 2019 <sup id="fnref:xavier2019tapes"><a href="#fn:xavier2019tapes" class="footnote" rel="footnote" role="doc-noteref">5</a></sup>; please note these tools have not implemented here nor is any assertion of their quality offered. Examples of effective variant filtering and expected candidate variant yield in studies of rare human disease are provided by Pedersen et al., 2021 <sup id="fnref:pedersen2021effective"><a href="#fn:pedersen2021effective" class="footnote" rel="footnote" role="doc-noteref">6</a></sup>.</p><h3 id="main-criteria-for-classifications"> <a href="#main-criteria-for-classifications" class="anchor-heading" aria-labelledby="main-criteria-for-classifications"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Main criteria for classifications</h3><div class="table-wrapper"><table><thead><tr><th style="text-align: center">Evidence type<th style="text-align: center">label<th style="text-align: center">Evidence<th style="text-align: center">Manual adjustment<th style="text-align: center">ACGM label<th style="text-align: center">Caveat checks<th style="text-align: left">Criteria<tbody><tr><td style="text-align: center">pathogenicity<td style="text-align: center">VS1<td style="text-align: center">very_strong<td style="text-align: center"> <td style="text-align: center">PVS1<td style="text-align: center">4<td style="text-align: left">null variant (nonsense, frameshift, canonical +- 2 splice sites, initiation codon, single or multiexon deletion) in a gene where LOF is a known mechanism of disease<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S1<td style="text-align: center">strong<td style="text-align: center"> <td style="text-align: center">PS1<td style="text-align: center">1<td style="text-align: left">Same amino acid change as a previously established pathogenic variant regardless of nucleotide change<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S2<td style="text-align: center">strong<td style="text-align: center">manual<td style="text-align: center">PS2<td style="text-align: center">1<td style="text-align: left">De novo (both maternity and paternity confirmed) in a patient with the disease and no family history<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S3<td style="text-align: center">strong<td style="text-align: center">manual<td style="text-align: center">PS3<td style="text-align: center">1<td style="text-align: left">Well-established in vitro or in vivo functional studies supportive of a damaging effect on the gene or gene product<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S4<td style="text-align: center">strong<td style="text-align: center"> <td style="text-align: center">PS4<td style="text-align: center">2<td style="text-align: left">The prevalence of the variant in affected individuals is significantly increased compared with the prevalence in controls<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S5<td style="text-align: center">other<td style="text-align: center"> <td style="text-align: center">PS5<td style="text-align: center"> <td style="text-align: left">The user has additional (value) strong pathogenic evidence<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M1<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM1<td style="text-align: center"> <td style="text-align: left">Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M2<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM2<td style="text-align: center">1<td style="text-align: left">Absent from controls (or at extremely low frequency if recessive) in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M3<td style="text-align: center">moderate<td style="text-align: center">manual<td style="text-align: center">PM3<td style="text-align: center">1<td style="text-align: left">For recessive disorders, detected in trans with a pathogenic variant<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M4<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM4<td style="text-align: center"> <td style="text-align: left">Protein length changes as a result of in-frame deletions/insertions in a nonrepeat region or stop-loss variants<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M5<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM5<td style="text-align: center">1<td style="text-align: left">Novel missense change at an amino acid residue where a different missense change determined to be pathogenic has been seen before<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M6<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM6<td style="text-align: center"> <td style="text-align: left">Assumed de novo, but without confirmation of paternity and maternity<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M7<td style="text-align: center">moderate<td style="text-align: center">other<td style="text-align: center">PM7<td style="text-align: center"> <td style="text-align: left">The user has additional (value) moderate pathogenic evidence<tr><td style="text-align: center">pathogenicity<td style="text-align: center">P1<td style="text-align: center">supporting<td style="text-align: center">manual<td style="text-align: center">PP1<td style="text-align: center"> <td style="text-align: left">Cosegregation with disease in multiple affected family members in a gene definitively known to cause the disease<tr><td style="text-align: center">pathogenicity<td style="text-align: center">P2<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">PP2<td style="text-align: center"> <td style="text-align: left">Missense variant in a gene that has a low rate of benign missense variation and in which missense variants are a common mechanism of disease<tr><td style="text-align: center">pathogenicity<td style="text-align: center">P3<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">PP3<td style="text-align: center">1<td style="text-align: left">Multiple lines of computational evidence support a deleterious effect on the gene or gene product (conservation, evolutionary, splicing impact, etc.)<tr><td style="text-align: center">pathogenicity<td style="text-align: center">P4<td style="text-align: center">supporting<td style="text-align: center">manual<td style="text-align: center">PP4<td style="text-align: center"> <td style="text-align: left">Patient’s phenotype or family history is highly specific for a disease with a single genetic etiology<tr><td style="text-align: center">pathogenicity<td style="text-align: center">P5<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">PP5<td style="text-align: center"> <td style="text-align: left">Reputable source recently reports variant as pathogenic, but the evidence is not available to the laboratory to perform an independent evaluation<tr><td style="text-align: center">pathogenicity<td style="text-align: center">P6<td style="text-align: center">supporting<td style="text-align: center">other<td style="text-align: center">PP6<td style="text-align: center"> <td style="text-align: left">The user has additional (value) supporting pathogenic evidence<tr><td style="text-align: center">benign<td style="text-align: center">A1<td style="text-align: center">stand_alone<td style="text-align: center"> <td style="text-align: center">BA1<td style="text-align: center"> <td style="text-align: left">Allele frequency is &gt;5% in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium<tr><td style="text-align: center">benign<td style="text-align: center">S1<td style="text-align: center">strong<td style="text-align: center"> <td style="text-align: center">BS1<td style="text-align: center"> <td style="text-align: left">Allele frequency is greater than expected for disorder<tr><td style="text-align: center">benign<td style="text-align: center">S2<td style="text-align: center">strong<td style="text-align: center"> <td style="text-align: center">BS2<td style="text-align: center"> <td style="text-align: left">Observed in a healthy adult individual for a recessive (homozygous), dominant (heterozygous), or X-linked (hemizygous) disorder, with full penetrance expected at an early age<tr><td style="text-align: center">benign<td style="text-align: center">S3<td style="text-align: center">strong<td style="text-align: center">manual<td style="text-align: center">BS3<td style="text-align: center"> <td style="text-align: left">Well-established in vitro or in vivo functional studies show no damaging effect on protein function or splicing<tr><td style="text-align: center">benign<td style="text-align: center">S4<td style="text-align: center">strong<td style="text-align: center">manual<td style="text-align: center">BS4<td style="text-align: center">1<td style="text-align: left">Lack of segregation in affected members of a family<tr><td style="text-align: center">benign<td style="text-align: center">S5<td style="text-align: center">strong<td style="text-align: center">other<td style="text-align: center">BS5<td style="text-align: center"> <td style="text-align: left">The user has additional (value) strong benign evidence<tr><td style="text-align: center">benign<td style="text-align: center">P1<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP1<td style="text-align: center"> <td style="text-align: left">Missense variant in a gene for which primarily truncating variants are known to cause disease<tr><td style="text-align: center">benign<td style="text-align: center">P2<td style="text-align: center">supporting<td style="text-align: center">manual<td style="text-align: center">BP2<td style="text-align: center"> <td style="text-align: left">Observed in trans with a pathogenic variant for a fully penetrant dominant gene/disorder or observed in cis with a pathogenic variant in any inheritance pattern<tr><td style="text-align: center">benign<td style="text-align: center">P3<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP3<td style="text-align: center"> <td style="text-align: left">In-frame deletions/insertions in a repetitive region without a known function<tr><td style="text-align: center">benign<td style="text-align: center">P4<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP4<td style="text-align: center">1<td style="text-align: left">Multiple lines of computational evidence suggest no impact on gene or gene product (conservation, evolutionary, splicing impact, etc.)<tr><td style="text-align: center">benign<td style="text-align: center">P5<td style="text-align: center">supporting<td style="text-align: center">manual<td style="text-align: center">BP5<td style="text-align: center"> <td style="text-align: left">Variant found in a case with an alternate molecular basis for disease<tr><td style="text-align: center">benign<td style="text-align: center">P6<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP6<td style="text-align: center"> <td style="text-align: left">Reputable source recently reports variant as benign, but the evidence is not available to the laboratory to perform an independent evaluation<tr><td style="text-align: center">benign<td style="text-align: center">P7<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP7<td style="text-align: center"> <td style="text-align: left">A synonymous (silent) variant for which splicing prediction algorithms predict no impact to the splice consensus sequence nor the creation of a new splice site AND the nucleotide is not highly conserved<tr><td style="text-align: center">benign<td style="text-align: center">P8<td style="text-align: center">supporting<td style="text-align: center">other<td style="text-align: center">BP8<td style="text-align: center"> <td style="text-align: left">The user has additional (value) supporting benign evidence</table></div><h3 id="caveats-implementing-filters"> <a href="#caveats-implementing-filters" class="anchor-heading" aria-labelledby="caveats-implementing-filters"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Caveats implementing filters</h3><p>Implementing the guidelines for interpretation of annotation requires multiple programmatic steps. The number of individual caveat checks indicate the number of bioinformatic filter functions used. Unnumbered caveat checks indicate that only a single filter function is required during reference to annotation databases. However, each function depends on reference to either one or several evidence source databases (approximately 150 sources) which are not shown here.</p><div class="table-wrapper"><table><thead><tr><th style="text-align: center">Evidence type<th style="text-align: center">label<th style="text-align: center">Evidence<th style="text-align: center">Manual adjustment<th style="text-align: center">ACGM label<th style="text-align: center">Caveat number<th style="text-align: left">Caveat<tbody><tr><td style="text-align: center">pathogenicity<td style="text-align: center">VS1<td style="text-align: center">very_strong<td style="text-align: center"> <td style="text-align: center">PVS1<td style="text-align: center">1<td style="text-align: left">LoF not disease causing for gene<tr><td style="text-align: center">pathogenicity<td style="text-align: center">VS1<td style="text-align: center">very_strong<td style="text-align: center"> <td style="text-align: center">PVS1<td style="text-align: center">2<td style="text-align: left">LoF at 3prime end (loftee)<tr><td style="text-align: center">pathogenicity<td style="text-align: center">VS1<td style="text-align: center">very_strong<td style="text-align: center"> <td style="text-align: center">PVS1<td style="text-align: center">3<td style="text-align: left">exon skipping splce that leave functional protein<tr><td style="text-align: center">pathogenicity<td style="text-align: center">VS1<td style="text-align: center">very_strong<td style="text-align: center"> <td style="text-align: center">PVS1<td style="text-align: center">4<td style="text-align: left">multiple transcript check<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S1<td style="text-align: center">strong<td style="text-align: center"> <td style="text-align: center">PS1<td style="text-align: center">1<td style="text-align: left">Assess for splicing vs amino acid change<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S2<td style="text-align: center">strong<td style="text-align: center">manual<td style="text-align: center">PS2<td style="text-align: center">1<td style="text-align: left">“Do not assume maternity or paternity i.e. egg donor, surrogate”<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S3<td style="text-align: center">strong<td style="text-align: center">manual<td style="text-align: center">PS3<td style="text-align: center">1<td style="text-align: left">Quality of functional evidence<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S4<td style="text-align: center">strong<td style="text-align: center"> <td style="text-align: center">PS4<td style="text-align: center">1<td style="text-align: left">“Assess RR, OR, CI for case/control evidence”<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S4<td style="text-align: center">strong<td style="text-align: center"> <td style="text-align: center">PS4<td style="text-align: center">2<td style="text-align: left">Rare variant in multiple indipendent cases can guide<tr><td style="text-align: center">pathogenicity<td style="text-align: center">S5<td style="text-align: center">other<td style="text-align: center"> <td style="text-align: center">PS5<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">pathogenicity<td style="text-align: center">M1<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM1<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">pathogenicity<td style="text-align: center">M2<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM2<td style="text-align: center">1<td style="text-align: left">Indels in population reference may not match your protocol<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M3<td style="text-align: center">moderate<td style="text-align: center">manual<td style="text-align: center">PM3<td style="text-align: center">1<td style="text-align: left">Pedigree sequencing is ideal. Phasing (GATK) may guide. Ldlink may guide<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M4<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM4<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">pathogenicity<td style="text-align: center">M5<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM5<td style="text-align: center">1<td style="text-align: left">Assess for splicing vs amino acid change<tr><td style="text-align: center">pathogenicity<td style="text-align: center">M6<td style="text-align: center">moderate<td style="text-align: center"> <td style="text-align: center">PM6<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">pathogenicity<td style="text-align: center">M7<td style="text-align: center">moderate<td style="text-align: center">other<td style="text-align: center">PM7<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">pathogenicity<td style="text-align: center">P1<td style="text-align: center">supporting<td style="text-align: center">manual<td style="text-align: center">PP1<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">pathogenicity<td style="text-align: center">P2<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">PP2<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">pathogenicity<td style="text-align: center">P3<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">PP3<td style="text-align: center">1<td style="text-align: left">“Prediction methods may have used the same data sources, do not assume each as independent evidence”<tr><td style="text-align: center">pathogenicity<td style="text-align: center">P4<td style="text-align: center">supporting<td style="text-align: center">manual<td style="text-align: center">PP4<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">pathogenicity<td style="text-align: center">P5<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">PP5<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">pathogenicity<td style="text-align: center">P6<td style="text-align: center">supporting<td style="text-align: center">other<td style="text-align: center">PP6<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">A1<td style="text-align: center">stand_alone<td style="text-align: center"> <td style="text-align: center">BA1<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">S1<td style="text-align: center">strong<td style="text-align: center"> <td style="text-align: center">BS1<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">S2<td style="text-align: center">strong<td style="text-align: center"> <td style="text-align: center">BS2<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">S3<td style="text-align: center">strong<td style="text-align: center">manual<td style="text-align: center">BS3<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">S4<td style="text-align: center">strong<td style="text-align: center">manual<td style="text-align: center">BS4<td style="text-align: center">1<td style="text-align: left">Question the accuracy of segregation. Presence of phenocopies can mimic lack of segregation. Additional pathogenic variants may produce autosomal dominant pattern.<tr><td style="text-align: center">benign<td style="text-align: center">S5<td style="text-align: center">strong<td style="text-align: center">other<td style="text-align: center">BS5<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">P1<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP1<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">P2<td style="text-align: center">supporting<td style="text-align: center">manual<td style="text-align: center">BP2<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">P3<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP3<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">P4<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP4<td style="text-align: center">1<td style="text-align: left">“Prediction methods may have used the same data sources, do not assume each as independent evidence”<tr><td style="text-align: center">benign<td style="text-align: center">P5<td style="text-align: center">supporting<td style="text-align: center">manual<td style="text-align: center">BP5<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">P6<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP6<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">P7<td style="text-align: center">supporting<td style="text-align: center"> <td style="text-align: center">BP7<td style="text-align: center"> <td style="text-align: left"> <tr><td style="text-align: center">benign<td style="text-align: center">P8<td style="text-align: center">supporting<td style="text-align: center">other<td style="text-align: center">BP8<td style="text-align: center"> <td style="text-align: left"> </table></div><h2 id="scoring-point-system"> <a href="#scoring-point-system" class="anchor-heading" aria-labelledby="scoring-point-system"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Scoring point system</h2><p><strong>Table 3</strong> from Tavtigian et al. 2020 table 2 <sup id="fnref:tavtigian2020fitting"><a href="#fn:tavtigian2020fitting" class="footnote" rel="footnote" role="doc-noteref">7</a></sup>: Point values for ACMG/AMP strength of evidence categories</p><div class="table-wrapper"><table><thead><tr><th>Evidence<th>Point scale<th>Pathogenic<th>Benign<tbody><tr><td>Indeterminate<td>0<td>0<td>0a<tr><td>Supporting<td>1<td>1<td>-1<tr><td>Moderate<td>2<td>2<td>-2b<tr><td>Strong<td>4<td>4<td>-4<tr><td>Very strong<td>8<td>8<td>-8b</table></div><p>a Note is made that <a href="#ref1">Richards et al.</a> did not specifically recognize indeterminate evidence. Nonetheless, if one thinks of the odds in favor of pathogenicity as a continuous variable, there exists a range that falls between Supporting Benign and Supporting Pathogenic. This is Indeterminate.</p><p>b Note is also made that <a href="#ref1">Richards et al.</a> did not specify benign evidence at the moderate or very strong levels. Nevertheless, the point system would readily support the addition of such criteria.</p><p><strong>Table 4</strong>. from Tavtigian et al. 2020 table 2 <sup id="fnref:tavtigian2020fitting:1"><a href="#fn:tavtigian2020fitting" class="footnote" rel="footnote" role="doc-noteref">7</a></sup>: Point-based variant classification categories</p><div class="table-wrapper"><table><thead><tr><th>Category<th>Point ranges<th>Notes<tbody><tr><td>Pathogenic<td>≥10<td> <tr><td>Likely Pathogenic<td>6 to 9a<td> <tr><td>Uncertain<td>0 to 5<td> <tr><td>Likely Benign<td>−1 to −6a<td> <tr><td>Benign<td>≤ −7<td> </table></div><p>a Operationally, the prior probability should be understood to be infinitesimally &gt;0.10. This has two effects. First, it makes the posterior probability of the American College of Medical Genetics (ACMG) Likely Pathogenic combining rules infinitesimally greater than 0.90, so that the Likely Pathogenic rules work properly. A specific value of 0.102 would have the added benefit that seven points would meet the IARC (International Agency for Research on Cancer) Likely Pathogenic threshold of 0.95. Second, it enforces a requirement for some evidence of benign effect for sequence variants to be classified as Likely Benign. One could also argue that the point threshold for Likely Benign should really be −2. This would match the ACMG rule “Likely Benign (ii)” rather than the simple numerical requirement that the posterior probability be &lt;0.10.</p><p>For reference, alternative public implementations of ACMG guidelines can be found in <a href="#ref3">Li &amp; Wang, 2017</a> and <a href="#ref4">Xavier et al., 2019</a>; please note these tools have not been implemented here nor is any assertion of their quality offered. Examples of effective variant filtering and expected candidate variant yield in studies of rare human disease are provided by <a href="#ref5">Pedersen et al., 2021</a>.</p><p>We use the evaluation of in silico predictors from Varsome. However, this paper discusses it also Wilcox et al., 2022 <sup id="fnref:wilcox2022evaluating"><a href="#fn:wilcox2022evaluating" class="footnote" rel="footnote" role="doc-noteref">8</a></sup>.</p><h2 id="acmguru-code-example"> <a href="#acmguru-code-example" class="anchor-heading" aria-labelledby="acmguru-code-example"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> ACMGuru code example</h2><p>The code from ACMGuru runs these runs using a range of functions and then performs a final tally. Some excerts from the code are shown here:</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># acmg_filters ----
# PVS1 ----
# PVS1 are null variants where IMPACT=="HIGH" and inheritance match, 
# in gene where LoF cause disease.
df$ACMG_PVS1 &lt;- NA
df &lt;- df %&gt;% dplyr::select(ACMG_PVS1, everything())
df$ACMG_PVS1 &lt;- ifelse(df$IMPACT == "HIGH" &amp; 
                        df$genotype == 2, "PVS1", NA) # homozygous
df$ACMG_PVS1 &lt;- ifelse(df$IMPACT == "HIGH" &amp; 
                        df$Inheritance == "AD", "PVS1", df$ACMG_PVS1) # dominant

# All functions for classification ...

# acmg tally  ----
# List of all ACMG labels
# acmg_labels &lt;- c("ACMG_PVS1", "ACMG_PS1", "ACMG_PS2", "ACMG_PS3", 
"ACMG_PS4", "ACMG_PS5", "ACMG_PM1", "ACMG_PM2", 
"ACMG_PM3", "ACMG_PM4", "ACMG_PM5", "ACMG_PM6", 
"ACMG_PM7", "ACMG_PP1", "ACMG_PP2", "ACMG_PP3", 
"ACMG_PP4")

# Transform 'Evidence_type' to 'P' for pathogenic and 'B' for benign
df_acmg$code_prefix &lt;- ifelse(df_acmg$Evidence_type == "pathogenicity", "P", "B")

# Create the ACMG code by combining the new prefix and the label, prepending 'ACMG_'
df_acmg$ACMG_code &lt;- paste0("ACMG_", df_acmg$code_prefix, df_acmg$label)

acmg_labels &lt;- df_acmg$ACMG_code
print(acmg_labels)

# df_acmg$ACMG_label
names(df) |&gt; head(30) |&gt; as.character()

# Check if each ACMG column exists, if not create it and fill with NA
for (acmg_label in acmg_labels) {
  if (!acmg_label %in% names(df)) {
      print("missing label")
          df[[acmg_label]] &lt;- NA
            }
            }

# Then use coalesce to find the first non-NA ACMG label
df$ACMG_highest &lt;- dplyr::coalesce(!!!df[acmg_labels])
df &lt;- df %&gt;% dplyr::select(ACMG_highest, everything())

# Count the number of non-NA values across the columns
df$ACMG_count &lt;- rowSums(!is.na(df[, acmg_labels ]))
df &lt;- df %&gt;% dplyr::select(ACMG_count, everything())
# df$ACMG_count[df$ACMG_count == 0] &lt;- NA

# ACMG Verdict----
# Define scores for Pathogenic criteria
pathogenic_scores &lt;- c(
 "PVS1" = 8,
  setNames(rep(4, 5), paste0("PS", 1:5)),
    setNames(rep(2, 7), paste0("PM", 1:7)),
      "PP3" = 1
      )

      # Define scores for Benign criteria
      benign_scores &lt;- c(
        "BA1" = -8,
          setNames(rep(-4, 5), paste0("BS", 1:5)),
            setNames(rep(-1, 8), paste0("BP", 1:8))
            )

            # Combine both scoring systems into one vector
            acmg_scores &lt;- c(pathogenic_scores, benign_scores)

            # Print the complete ACMG scoring system
            print(acmg_scores)

            # Create ACMG_score column by looking up ACMG_highest in acmg_scores
            df$ACMG_score &lt;- acmg_scores[df$ACMG_highest]

# If there are any ACMG labels that don't have a corresponding score, 
# these will be NA. You may want to set these to 0.
df$ACMG_score[is.na(df$ACMG_score)] &lt;- 0
df &lt;- df |&gt; dplyr::select(ACMG_score, everything())

# Total ACMG score ----

# Mutate all ACMG columns
df &lt;- df %&gt;% 
  mutate_at(acmg_labels, function(x) acmg_scores[x])

# Replace NAs with 0 in ACMG columns only
df[acmg_labels] &lt;- lapply(df[acmg_labels], function(x) ifelse(is.na(x), 0, x))

# Calculate total ACMG score
df$ACMG_total_score &lt;- rowSums(df[acmg_labels])

df &lt;- df |&gt; dplyr::select(ACMG_total_score, everything())
</code></pre></div></div><h2 id="references"> <a href="#references" class="anchor-heading" aria-labelledby="references"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> References</h2><div class="footnotes" role="doc-endnotes"><ol><li id="fn:richards2015standards"><p>Richards, S. et al., 2015. Standards and guidelines for the interpretation of sequence variants: a joint consensus recommendation of the American College of Medical Genetics and Genomics and the Association for Molecular Pathology. <em>Genetics in Medicine</em>, 17(5), pp.405–423. DOI: <a href="https://www.gimjournal.org/article/S1098-3600(21)03031-8/fulltext">10.1038/gim2015.30</a>. <a href="#fnref:richards2015standards" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:li2017standards"><p>Li, M.M. et al., 2017. Standards and guidelines for the interpretation and reporting of sequence variants in cancer: a joint consensus recommendation of the Association for Molecular Pathology, American Society of Clinical Oncology, and College of American Pathologists. <em>The Journal of Molecular Diagnostics</em>, 19(1), pp.4–23. DOI: <a href="https://doi.org/10.1016/j.jmoldx.2016.10.002">10.1016/j.jmoldx.2016.10.002</a>. <a href="#fnref:li2017standards" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:riggs2020technical"><p>Riggs, E.R. et al., 2020. Technical standards for the interpretation and reporting of constitutional copy-number variants: a joint consensus recommendation of the American College of Medical Genetics and Genomics (ACMGe and the Clinical Genome Resource (ClinGen). <em>Genetics in Medicine</em>, 22(2), pp.245–257. DOI: <a href="https://doi.org/10.1038/s41436-019-0686-8">10.1038/s41436-019-0686-8</a>. <a href="#fnref:riggs2020technical" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:li2017intervar"><p>Li, Q. and Wang, K., 2017. InterVar: clinical interpretation of genetic variants by the 2015 ACMG-AMP guidelines. <em>The American Journal of Human Genetics</em>, 100(2), pp.267–280. DOI: <a href="https://doi.org/10.1016/j.ajhg.2016.12.00://doi.org/10.1016/j.ajhg.2017.01.004">10.1016/j.ajhg.2017.01.004</a>. <a href="#fnref:li2017intervar" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:xavier2019tapes"><p>Xavier, A. et al., 2019. TAPES: A tool for assessment and prioritisation in exome studies. <em>PLoS Computational Biology</em>, 15(10), e1007453. DOI: <a href="https://doi.org/10.1371/journal.pcbi.1007453">10.1371/journal.pcbi.1007453</a>. <a href="#fnref:xavier2019tapes" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:pedersen2021effective"><p>Pedersen, B.S. et al., 2021. Effective variant filtering and expected candidate variant yield in studies of rare human disease. <em>NPJ Genomic Medicine</em>, 6(1), pp.1–8. DOI: <a href="https://doi.org/10.1038/s41525-021-00227-3">10.1038/s41525-021-00227-3</a>. <a href="#fnref:pedersen2021effective" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:tavtigian2020fitting"><p>Tavtigian SV, Harrison SM, Boucher KM, Biesecker LG. (2020). Fitting a naturally scaled point system to the ACMG/AMP variant classification guidelines. <em>Human Mutation</em>, 41, 1734–1737. <a href="https://doi.org/10.1002/humu.24088">https://doi.org/10.1002/humu.24088</a> <a href="#fnref:tavtigian2020fitting" class="reversefootnote" role="doc-backlink">&#8617;</a> <a href="#fnref:tavtigian2020fitting:1" class="reversefootnote" role="doc-backlink">&#8617;<sup>2</sup></a></p><li id="fn:wilcox2022evaluating"><p>Emma H. Wilcox, Mahdi Sarmady, Bryan Wulf, Matt W. Wright, Heidi L. Rehm, Leslie G. Biesecker, Ahmad N. Abou Tayoun. (2022). Evaluating the impact of in silico predictors on clinical variant classification. <em>Genetics in Medicine</em>, Volume 24, Issue 4, Pages 924-930, ISSN 1098-3600, <a href="https://doi.org/10.1016/j.gim.2021.11.018">https://doi.org/10.1016/j.gim.2021.11.018</a>. <a href="#fnref:wilcox2022evaluating" class="reversefootnote" role="doc-backlink">&#8617;</a></p></ol></div><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
