<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Stats Correlation, regression and repeated data | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Stats Correlation, regression and repeated data" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/altman_bland_correlation.html" /><meta property="og:url" content="http://localhost:4000/pages/altman_bland_correlation.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Stats Correlation, regression and repeated data" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Stats Correlation, regression and repeated data","url":"http://localhost:4000/pages/altman_bland_correlation.html"}</script> <script type="text/javascript" defer src="/assets/js/mathjax-script-type.js"> </script> <script type="text/javascript" id="MathJax-script" defer src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"> </script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item active"><a href="/pages/altman_bland_correlation.html" class="nav-list-link active">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/storage.html" class="nav-list-link">storage</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><h1 id="stats-correlation-regression-and-repeated-data"> <a href="#stats-correlation-regression-and-repeated-data" class="anchor-heading" aria-labelledby="stats-correlation-regression-and-repeated-data"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Stats Correlation, regression and repeated data</h1><p>Last update: 20210629</p><ul id="markdown-toc"><li><a href="#stats-correlation-regression-and-repeated-data" id="markdown-toc-stats-correlation-regression-and-repeated-data">Stats Correlation, regression and repeated data</a><ul><li><a href="#introduction" id="markdown-toc-introduction">Introduction</a><li><a href="#correlation-within-subjects-part-1" id="markdown-toc-correlation-within-subjects-part-1">Correlation within subjects, part 1</a><li><a href="#correlation-within-subjects-part-2" id="markdown-toc-correlation-within-subjects-part-2">Correlation within subjects, part 2</a><li><a href="#references" id="markdown-toc-references">References</a></ul></ul><hr /><h2 id="introduction"> <a href="#introduction" class="anchor-heading" aria-labelledby="introduction"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Introduction</h2><p>This topic is introduced as the first paper bland1994correlation in a series of BMJ statistical notes by Altman &amp; Bland (<a href="https://github.com/DylanLawless/notes/blob/main/202106291417-lit-altman_bland.md"> lit-altman_bland.md </a>): 1. Bland JM, Altman DG. (1994) <a href="http://www.bmj.com/cgi/content/full/308/6933/896"> Correlation, regression and repeated data.</a> <b>308</b>, 896. <sup id="a1"><a href="#f1">1</a></sup></p><p>It concerns the analysis of paired data where there is more than one observation per subject. They point out that it could be highly misleading to analyse such data by combining repeated observations from several subjects and then calculating the correlation coefficient as if the data were a simple sample.</p><p>Many researchers would assume that it is acceptable to gather repeated measurements for individuals and put all the data together.</p><p>They use simulated data showing five pairs of measurements of two uncorrelated variables X and Y for subjects 1, 2, 3, 4, and 5. Using each subject’s mean values, they show correlation coefficient r=-0.67, df=3, P=0.22. However, when they put all 25 observations together they get r=-0.47, df=23, P=0.02. When the calculation is performed as if they have 25 subjects, the number of degrees of freedom for the significance test is increased incorrectly and a spurious significant difference is produced. Thus demonstrating that one should not mix observations from different subjects indiscriminately, whether using correlation or the closely related regression analysis.</p><h2 id="correlation-within-subjects-part-1"> <a href="#correlation-within-subjects-part-1" class="anchor-heading" aria-labelledby="correlation-within-subjects-part-1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Correlation within subjects, part 1</h2><p>The methods to use in these circumstances are later discussed in another note for the BMJ series bland1995statistics, number 11 on the list <a href="https://github.com/DylanLawless/notes/blob/main/202106291417-lit-altman_bland.md"> lit-altman_bland.md </a>: 11. Bland JM, Altman DG. (1995) <a href="http://www.bmj.com/cgi/content/full/310/6977/446">Calculating correlation coefficients with repeated observations: Part 1, correlation within subjects.</a> <b>310</b>, 446.</p><p><em>Notes: I am replacing their terms for my notes:</em></p><ul><li><em>X = Paco2</em><li><em>Y = pHi</em></ul><p>In this note they show an example table using 8 subjects with <em>4-8 observations</em> for X and Y (<strong>table I</strong>):</p><div class="table-wrapper"><table><thead><tr><th style="text-align: left">Subject<th style="text-align: center">Y<th style="text-align: center">X<tbody><tr><td style="text-align: left">1<td style="text-align: center">6.68<td style="text-align: center">3.97<tr><td style="text-align: left">1<td style="text-align: center">6.53<td style="text-align: center">4.12<tr><td style="text-align: left">…<td style="text-align: center">…<td style="text-align: center">…</table></div><ul><li>If subject’s average Y is related to the subject’s average X<ul><li>We can use the correlation between the subject means, which they shall describe in a subsequent note.</ul><li>If an increase in Y within the individual was associated with an increase in X<ul><li>We want to remove the differences between subjects and look only at changes within.</ul></ul><p>To look at variation within the subject we can use multiple regression.</p><ul><li>Make one variable, X or Y, the outcome variable and the other variable and the subject the predictor variables.<li><p>The subject is treated as a categorical factor using dummy variables and so has seven degrees of freedom.</p><li>Using an analysis of variance table for the regression (table II) shows how the variability in Y can be partitioned into components due to different sources.<ul><li>Also known as analysis of covariance<li>Equivalent to fitting parallel lines through each subject’s data (<strong>Figure 1.</strong>)</ul></ul><div class="table-wrapper"><div class="table-wrapper"><table><thead><tr><th style="text-align: left">Source of variation<th style="text-align: right">Degrees of freedom<th style="text-align: right">Sum of squares<th style="text-align: right">Mean square<th style="text-align: right">Variance ratio (F)<th style="text-align: right">Probability<tbody><tr><td style="text-align: left">Subjects<td style="text-align: right">7<td style="text-align: right">2.9661<td style="text-align: right">0.4237<td style="text-align: right">48.3<td style="text-align: right">\(&lt;\)0.0001<tr><td style="text-align: left">X<td style="text-align: right">1<td style="text-align: right">0.1153<td style="text-align: right">0.1153<td style="text-align: right">13.1<td style="text-align: right">0.0008<tr><td style="text-align: left">Residual<td style="text-align: right">38<td style="text-align: right">0.3337<td style="text-align: right">0.0088<td style="text-align: right"> <td style="text-align: right"> <tr><td style="text-align: left">Total<td style="text-align: right">46<td style="text-align: right">3.3139<td style="text-align: right">0.0720<td style="text-align: right"> <td style="text-align: right"> </table></div></div><p>Table II. Analysis of variance for the data in table I (as shown in Altman &amp; Bland). At the end of this page, this table is reproduced based on the original data and new R code, as shown. (Note that there is a slight varition between the published version and my replicated version of table II and Figure 1 below; probably due to a minor data entry error by the publisher or authors).</p><ul><li>The residual sum of squares in <strong>table II</strong> represents the variation about regression lines.<li>This removes the variation due to subjects (and any other nuisance variables which might be present) and express the variation in Y due to X as a proportion of what’s left:<ul><li>(Sum of squares for X)/(Sum of squares for X + residual sum of squares).</ul><li>The magnitude of the correlation coefficient within subjects is the square root of this proportion.<ul><li>For <strong>table II</strong> this is: \(\sqrt{ \frac{0.1153}{0.1153+0.3337} } = 0.51\)<li>The sign of the correlation coefficient is given by the sign of the regression coefficient for X.</ul><li>Regression slope is -0.108<li>So the correlation coefficient within subjects is -0.51.<li>The P value is found either from:<ul><li>F test in the associated analysis of variance table,<li>t test for the regression slope.<li>It doesn’t matter which variable we regress on which;<li>we get the same correlation coefficient and P value either way.</ul></ul><p>Incorrectly calculating the correlation coefficient by ignoring the fact that we have 47 observations on only 8 subjects, would produce -0.07, P=0.7.</p><p><img src="/assets/images/altman_bland_repeated_observations.png" width="100%" /> <strong>Figure 1.</strong> Recreation of “(Y) pH against (X) PaCO2 for eight subjects, with parallel lines fitted for each subject” as used in bland1995statistics. Interestingly, replotting this data shows that their figure was not fully accurate (forgivable before the days of Rstudio in 1995, and not important for this example).</p><h2 id="correlation-within-subjects-part-2"> <a href="#correlation-within-subjects-part-2" class="anchor-heading" aria-labelledby="correlation-within-subjects-part-2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Correlation within subjects, part 2</h2><p>The second part shows how to find the correlation between the subject means bland1995calculating, number 12 on the list <a href="https://github.com/DylanLawless/notes/blob/main/202106291417-lit-altman_bland.md"> lit-altman_bland.md </a>: 12. Bland JM, Altman DG. (1995) <a href="http://www.bmj.com/cgi/content/full/310/6980/633">Calculating correlation coefficients with repeated observations: Part 2, correlation between subjects.</a> <b>310</b>, 633.</p><p>In this note they show the example table using the same 8 subjects with <em>one mean observation</em> for X and Y:</p><div class="table-wrapper"><table><thead><tr><th style="text-align: center">Subject<th style="text-align: center">Y<th style="text-align: center">X<th style="text-align: center">Number<tbody><tr><td style="text-align: center">1<td style="text-align: center">6.49<td style="text-align: center">4.04<td style="text-align: center">4<tr><td style="text-align: center">2<td style="text-align: center">7.05<td style="text-align: center">5.37<td style="text-align: center">4<tr><td style="text-align: center">3<td style="text-align: center">7.36<td style="text-align: center">4.83<td style="text-align: center">9<tr><td style="text-align: center">4<td style="text-align: center">7.33<td style="text-align: center">5.31<td style="text-align: center">5<tr><td style="text-align: center">5<td style="text-align: center">7.31<td style="text-align: center">4.40<td style="text-align: center">8<tr><td style="text-align: center">6<td style="text-align: center">7.32<td style="text-align: center">4.92<td style="text-align: center">6<tr><td style="text-align: center">7<td style="text-align: center">6.91<td style="text-align: center">6.60<td style="text-align: center">3<tr><td style="text-align: center">8<td style="text-align: center">7.12<td style="text-align: center">4.78<td style="text-align: center">8</table></div><ul><li>If subject’s average Y is related to the subject’s average X<ul><li>We can use the correlation between the subject means.</ul></ul><p>They calculate the usual correlation coefficient for the mean Y and mean X; r=0.09, P=0.8. Does not take into account the different numbers of measurements on each subject.</p><ul><li>Does this matter?:<ul><li>Depends on how different the numbers of observations are<li>whether the measurements within subjects vary much compared with the means between subjects</ul></ul><p>We can calculate a weighted correlation coefficient using the number of observations as weights. Many computer programs will calculate this, but it is not difficult to do by hand.</p><ul><li>They denote the mean Y and X for subject i by \(\bar{x}_i\) and \(\bar{y}_i\),<li>the number of observations for subject i by \(m_i\),<li>and the number of subjects by \(n\).<li>The weighted mean of the \(\bar{x}_i\) is \(\frac{ \sum{ m_i \bar{x}_i } }{ \sum{ m_i } }\)</ul><p>In the usual case, where there is one observation per subject, the \(m_i\) are all one and this formula gives the usual mean \(\frac{ \sum{\bar{x}_i} }{n}\).</p><p>An easy way to calculate the weighted correlation coefficient is to replace each individual observation by its subject mean. Thus the table would yield 47 pairs of observations, the first four of which would each be pH=6.49 and Paco2=4.04, and so on.</p><p>If we use the usual formula for the correlation coefficient on the expanded data we will get the weighted correlation coefficient. However, we must be careful when it comes to the P value. We have only 8 observations (n in general), not 47. We should ignore any P value printed by our computer program, and use a statistical table instead.</p><p>The formula for a weighted correlation coefficient is:</p><div class="math-container"> $$ \frac{ \sum{m_i \bar{x}_i \bar{y}_i} - \sum{m_i \bar{x}_i} \sum{m_i \bar{y}_i} \mathbin{/} \sum{m_i} }{ \sqrt{ ( \sum{m_i \bar{y}_i^2} - (\sum{m_i \bar{y}_i})^2 \mathbin{/} \sum{m_i} ) ( \sum{m_i \bar{y}_i^2} - (\sum{m_i \bar{y}_i})^2 \mathbin{/} \sum{m_i} )} } $$</div><p>where all summations are from \(i=1\) to \(n\). When all the \(m_i\) are equal they cancel out, giving the usual formula for a correlation coefficient.</p><p>For the data in the table the weighted correlation coefficient is r=0.08, P=0.9. There is no evidence that subjects with a high Y also have a high X. However, as they have already shown in part 1, within the subject a rise in Y was associated with a fall in X.</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## Code and raw data for Table I, Analysis of variance table II, and Figure 1</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="w"> </span><span class="p">(</span><span class="w">
  </span><span class="n">Subject</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"1"</span><span class="p">,</span><span class="w"> </span><span class="s2">"2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"2"</span><span class="p">,</span><span class="w"> </span><span class="s2">"3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"3"</span><span class="p">,</span><span class="w"> </span><span class="s2">"4"</span><span class="p">,</span><span class="w"> </span><span class="s2">"4"</span><span class="p">,</span><span class="w"> </span><span class="s2">"4"</span><span class="p">,</span><span class="w"> </span><span class="s2">"4"</span><span class="p">,</span><span class="w"> </span><span class="s2">"4"</span><span class="p">,</span><span class="w"> </span><span class="s2">"5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"5"</span><span class="p">,</span><span class="w"> </span><span class="s2">"6"</span><span class="p">,</span><span class="w"> </span><span class="s2">"6"</span><span class="p">,</span><span class="w"> </span><span class="s2">"6"</span><span class="p">,</span><span class="w"> </span><span class="s2">"6"</span><span class="p">,</span><span class="w"> </span><span class="s2">"6"</span><span class="p">,</span><span class="w"> </span><span class="s2">"6"</span><span class="p">,</span><span class="w"> </span><span class="s2">"7"</span><span class="p">,</span><span class="w"> </span><span class="s2">"7"</span><span class="p">,</span><span class="w"> </span><span class="s2">"7"</span><span class="p">,</span><span class="w"> </span><span class="s2">"8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"8"</span><span class="p">,</span><span class="w"> </span><span class="s2">"8"</span><span class="p">),</span><span class="w">

  </span><span class="n">Y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">6.68</span><span class="p">,</span><span class="w"> </span><span class="m">6.53</span><span class="p">,</span><span class="w"> </span><span class="m">6.43</span><span class="p">,</span><span class="w"> </span><span class="m">6.33</span><span class="p">,</span><span class="w"> </span><span class="m">6.85</span><span class="p">,</span><span class="w"> </span><span class="m">7.06</span><span class="p">,</span><span class="w"> </span><span class="m">7.13</span><span class="p">,</span><span class="w"> </span><span class="m">7.17</span><span class="p">,</span><span class="w"> </span><span class="m">7.4</span><span class="p">,</span><span class="w"> </span><span class="m">7.42</span><span class="p">,</span><span class="w"> </span><span class="m">7.41</span><span class="p">,</span><span class="w"> </span><span class="m">7.37</span><span class="p">,</span><span class="w"> </span><span class="m">7.34</span><span class="p">,</span><span class="w"> </span><span class="m">7.35</span><span class="p">,</span><span class="w"> </span><span class="m">7.28</span><span class="p">,</span><span class="w"> </span><span class="m">7.3</span><span class="p">,</span><span class="w"> </span><span class="m">7.34</span><span class="p">,</span><span class="w"> </span><span class="m">7.36</span><span class="p">,</span><span class="w"> </span><span class="m">7.33</span><span class="p">,</span><span class="w"> </span><span class="m">7.29</span><span class="p">,</span><span class="w"> </span><span class="m">7.3</span><span class="p">,</span><span class="w"> </span><span class="m">7.35</span><span class="p">,</span><span class="w"> </span><span class="m">7.35</span><span class="p">,</span><span class="w"> </span><span class="m">7.3</span><span class="p">,</span><span class="w"> </span><span class="m">7.3</span><span class="p">,</span><span class="w"> </span><span class="m">7.37</span><span class="p">,</span><span class="w"> </span><span class="m">7.27</span><span class="p">,</span><span class="w"> </span><span class="m">7.28</span><span class="p">,</span><span class="w"> </span><span class="m">7.32</span><span class="p">,</span><span class="w"> </span><span class="m">7.32</span><span class="p">,</span><span class="w"> </span><span class="m">7.38</span><span class="p">,</span><span class="w"> </span><span class="m">7.3</span><span class="p">,</span><span class="w"> </span><span class="m">7.29</span><span class="p">,</span><span class="w"> </span><span class="m">7.33</span><span class="p">,</span><span class="w"> </span><span class="m">7.31</span><span class="p">,</span><span class="w"> </span><span class="m">7.33</span><span class="p">,</span><span class="w"> </span><span class="m">6.86</span><span class="p">,</span><span class="w"> </span><span class="m">6.94</span><span class="p">,</span><span class="w"> </span><span class="m">6.92</span><span class="p">,</span><span class="w"> </span><span class="m">7.19</span><span class="p">,</span><span class="w"> </span><span class="m">7.29</span><span class="p">,</span><span class="w"> </span><span class="m">7.21</span><span class="p">,</span><span class="w"> </span><span class="m">7.25</span><span class="p">,</span><span class="w"> </span><span class="m">7.2</span><span class="p">,</span><span class="w"> </span><span class="m">7.19</span><span class="p">,</span><span class="w"> </span><span class="m">6.77</span><span class="p">,</span><span class="w"> </span><span class="m">6.82</span><span class="p">),</span><span class="w">

  </span><span class="n">X</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">3.97</span><span class="p">,</span><span class="w"> </span><span class="m">4.12</span><span class="p">,</span><span class="w"> </span><span class="m">4.09</span><span class="p">,</span><span class="w"> </span><span class="m">3.97</span><span class="p">,</span><span class="w"> </span><span class="m">5.27</span><span class="p">,</span><span class="w"> </span><span class="m">5.37</span><span class="p">,</span><span class="w"> </span><span class="m">5.41</span><span class="p">,</span><span class="w"> </span><span class="m">5.44</span><span class="p">,</span><span class="w"> </span><span class="m">5.67</span><span class="p">,</span><span class="w"> </span><span class="m">3.64</span><span class="p">,</span><span class="w"> </span><span class="m">4.32</span><span class="p">,</span><span class="w"> </span><span class="m">4.73</span><span class="p">,</span><span class="w"> </span><span class="m">4.96</span><span class="p">,</span><span class="w"> </span><span class="m">5.04</span><span class="p">,</span><span class="w"> </span><span class="m">5.22</span><span class="p">,</span><span class="w"> </span><span class="m">4.82</span><span class="p">,</span><span class="w"> </span><span class="m">5.07</span><span class="p">,</span><span class="w"> </span><span class="m">5.67</span><span class="p">,</span><span class="w"> </span><span class="m">5.1</span><span class="p">,</span><span class="w"> </span><span class="m">5.53</span><span class="p">,</span><span class="w"> </span><span class="m">4.75</span><span class="p">,</span><span class="w"> </span><span class="m">5.51</span><span class="p">,</span><span class="w"> </span><span class="m">4.28</span><span class="p">,</span><span class="w"> </span><span class="m">4.44</span><span class="p">,</span><span class="w"> </span><span class="m">4.32</span><span class="p">,</span><span class="w"> </span><span class="m">3.23</span><span class="p">,</span><span class="w"> </span><span class="m">4.46</span><span class="p">,</span><span class="w"> </span><span class="m">4.72</span><span class="p">,</span><span class="w"> </span><span class="m">4.75</span><span class="p">,</span><span class="w"> </span><span class="m">4.99</span><span class="p">,</span><span class="w"> </span><span class="m">4.78</span><span class="p">,</span><span class="w"> </span><span class="m">4.73</span><span class="p">,</span><span class="w"> </span><span class="m">5.12</span><span class="p">,</span><span class="w"> </span><span class="m">4.93</span><span class="p">,</span><span class="w"> </span><span class="m">5.03</span><span class="p">,</span><span class="w"> </span><span class="m">4.93</span><span class="p">,</span><span class="w"> </span><span class="m">6.85</span><span class="p">,</span><span class="w"> </span><span class="m">6.44</span><span class="p">,</span><span class="w"> </span><span class="m">6.52</span><span class="p">,</span><span class="w"> </span><span class="m">5.28</span><span class="p">,</span><span class="w"> </span><span class="m">4.56</span><span class="p">,</span><span class="w"> </span><span class="m">4.34</span><span class="p">,</span><span class="w"> </span><span class="m">4.32</span><span class="p">,</span><span class="w"> </span><span class="m">4.41</span><span class="p">,</span><span class="w"> </span><span class="m">3.69</span><span class="p">,</span><span class="w"> </span><span class="m">6.09</span><span class="p">,</span><span class="w"> </span><span class="m">5.58</span><span class="p">)</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Run the Analysis of Variance with mutiple variable</span><span class="w">
</span><span class="n">name</span><span class="o">=</span><span class="n">aov</span><span class="p">(</span><span class="n">Y</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Subject</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">X</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">df</span><span class="p">)</span><span class="w"> </span><span class="c1">#runs the ANOVA test</span><span class="w">
</span><span class="n">ls</span><span class="p">(</span><span class="n">name</span><span class="p">)</span><span class="w"> </span><span class="c1">#lists the items stored by the test.</span><span class="w">
</span><span class="n">summary</span><span class="p">(</span><span class="n">name</span><span class="p">)</span><span class="w"> </span><span class="c1">#give the basic ANOVA output.</span><span class="w">

</span><span class="c1"># Output the column totals to match Altman &amp; Bland table</span><span class="w">
</span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">unlist</span><span class="p">(</span><span class="w"> </span><span class="n">summary</span><span class="p">(</span><span class="n">name</span><span class="p">)</span><span class="w"> </span><span class="p">))</span><span class="w">

</span><span class="nf">sum</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">,])</span><span class="w"> </span><span class="c1"># Total Degrees of freedom</span><span class="w">
</span><span class="nf">sum</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="m">4</span><span class="o">:</span><span class="m">6</span><span class="p">,])</span><span class="w"> </span><span class="c1"># Total Sum Sq</span><span class="w">
</span><span class="nf">sum</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="m">7</span><span class="o">:</span><span class="m">9</span><span class="p">,])</span><span class="w"> </span><span class="c1"># Total Mean Sq</span><span class="w">

</span></code></pre></div></div><div class="table-wrapper"><div class="table-wrapper"><table><thead><tr><th> <th>Degrees of freedom<th>Sum of squares<th>Mean Square<th>Variance ratio (F)<th>Probability<tbody><tr><td>df$Subject<td>7<td>2.8648<td>0.4093<td>46.60<td>&lt; 2e-16<tr><td>df$X<td>1<td>0.1153<td>0.1153<td>13.13<td>0.000847<tr><td>Residuals<td>38<td>0.3337<td>0.0088<td> <td> </table></div></div><p>Replicated version of Table II. Analysis of variance for the data in table I. Default R output headings modified: Degrees of freedom (Df), Sum of squares (Sum Sq), Mean square (Mean Sq), Variance ratio F (F value), Probability (Pr(&gt;F)). (Repeated note: there is a slight varition between the published version and my replicated version of table II and Figure 1; probably due to a minor data entry error by the publisher or authors).</p><div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># code used to produce Figure 1. </span><span class="w">

</span><span class="n">require</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">X</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Y</span><span class="p">,</span><span class="w"> </span><span class="n">group</span><span class="o">=</span><span class="n">Subject</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Subject</span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_smooth</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Subject</span><span class="p">),</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"lm"</span><span class="p">,</span><span class="w"> </span><span class="n">formula</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">se</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w">

</span><span class="c1"># The dataset is cited by Bland &amp; Altman 1995 as: "Boyd O, Mackay CJ, Lamb G, Bland JM, Grounds RM, Bennett ED.Comparison of clinical information gained from routine blood-gas analysis and from gastric tonometry for intramural pH.Lancet1993;341:142–6."</span><span class="w">

</span></code></pre></div></div><h2 id="references"> <a href="#references" class="anchor-heading" aria-labelledby="references"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> References</h2><p><strong>Footnote</strong> <b id="f1">1</b> This article is almost identical to the original version in acknowledgment to Altman and Bland. It is adapted here as part of a set of curated, consistent, and minimal examples of statistics required for human genomic analysis. <a href="#a1">↩</a></p><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
