<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>BeviMed | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="BeviMed" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/bevimed.html" /><meta property="og:url" content="http://localhost:4000/pages/bevimed.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="BeviMed" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"BeviMed","url":"http://localhost:4000/pages/bevimed.html"}</script> <script type="text/javascript" defer src="/assets/js/mathjax-script-type.js"> </script> <script type="text/javascript" id="MathJax-script" defer src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"> </script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item active"><a href="/pages/bevimed.html" class="nav-list-link active">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metadata_users.html" class="nav-list-link">WGS metadata users</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><p>Last update: 20241104</p><h1 id="bevimed"> <a href="#bevimed" class="anchor-heading" aria-labelledby="bevimed"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> BeviMed</h1><ul id="markdown-toc"><li><a href="#bevimed" id="markdown-toc-bevimed">BeviMed</a><ul><li><a href="#talks-on-the-topics" id="markdown-toc-talks-on-the-topics">Talks on the topics</a><li><a href="#introduction" id="markdown-toc-introduction">Introduction</a><li><a href="#material-and-methods" id="markdown-toc-material-and-methods">Material and Methods</a><ul><li><a href="#model-specification" id="markdown-toc-model-specification">Model Specification</a><li><a href="#inference" id="markdown-toc-inference">Inference</a></ul></ul><li><a href="#references" id="markdown-toc-references">References</a></ul><p class="note">This page is a work in progress and thus contains content quoted directly from: Greene, Daniel et al. “A Fast Association Test for Identifying Pathogenic Variants Involved in Rare Diseases.” <em>The American Journal of Human Genetics</em>, Volume 101, Issue 1, pages 104 - 114. <a href="http://dx.doi.org/10.1016/j.ajhg.2017.05.015">http://dx.doi.org/10.1016/j.ajhg.2017.05.015</a>.</p><h2 id="talks-on-the-topics"> <a href="#talks-on-the-topics" class="anchor-heading" aria-labelledby="talks-on-the-topics"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Talks on the topics</h2><ul><li><a href="https://www.youtube.com/watch?v=0SFGjWeGI0I">https://www.youtube.com/watch?v=0SFGjWeGI0I</a><li><a href="https://www.youtube.com/watch?v=9f4BOQvN_E4">https://www.youtube.com/watch?v=9f4BOQvN_E4</a></ul><h2 id="introduction"> <a href="#introduction" class="anchor-heading" aria-labelledby="introduction"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Introduction</h2><p>In principle, Bayesian inference lends itself well to rare variant association analysis because it provides a coherent framework for sharing information across variants and provides a natural way of incorporating prior information on variant pathogenicity. The variational Bayes discrete mixture method (vbdm),<sup id="fnref:6"><a href="#fn:6" class="footnote" rel="footnote" role="doc-noteref">1</a></sup> the Bayesian risk index,<sup id="fnref:7"><a href="#fn:7" class="footnote" rel="footnote" role="doc-noteref">2</a></sup> and the Bayesian rare variant detector (BRVD)<sup id="fnref:8"><a href="#fn:8" class="footnote" rel="footnote" role="doc-noteref">3</a></sup> all model a mixture of pathogenic and non-pathogenic variants in a locus, but they employ additive models of disease risk or severity more suited to complex rather than rare diseases caused by dominant or recessive inheritance of one or two pathogenic alleles. Here we present a Bayesian model in which disease risk depends on the genotypes at rare variants in a locus, a latent mode of inheritance, and a latent partition of variants into pathogenic and non-pathogenic subsets. Different modes of inheritance are modeled by conditioning the probability of case status on the number of pathogenic alleles and the ploidy for each individual at the variants. Thus, disease risk due to compound heterozygosity or X-linked inheritance is explicitly accommodated. Prior knowledge concerning variant pathogenicity can be incorporated in the form of shifts in the log odds of pathogenicity relative to a global mean. By placing a vague prior distribution on the scale of these shifts, the usefulness or otherwise of these co-data are accounted for flexibly to maximize power. For a given set of variants, inference is performed by comparing the model described above with a baseline model in which disease risk is independent of the genotypes. The mode of inheritance and the pathogenicity of each variant, conditional on an association, can be inferred through the posterior distributions of parameters in the model. Particular classes of variants in a locus may be the only ones that confer disease risk. For example, only variants in the 50 UTR region or only high-impact coding variants may be involved. Our method can compare models fitted to different classes of variants in order to infer which ones are responsible for disease. Typically the inference process would be repeated over many sets of variants selected from different loci throughout the genome. The procedures are implemented in an efficient R package called BeviMed, which stands for Bayesian evaluation of variant involvement in Mendelian disease.</p><h2 id="material-and-methods"> <a href="#material-and-methods" class="anchor-heading" aria-labelledby="material-and-methods"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Material and Methods</h2><h3 id="model-specification"> <a href="#model-specification" class="anchor-heading" aria-labelledby="model-specification"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Model Specification</h3><p>Let \(y\) be a binary vector of length \(N\) indicating whether individual \(i\) is a case (\(y_i = 1\)) or a control (\(y_i = 0\)) subject with respect to a particular disease. Suppose \(k\) rare variants are under consideration (typically in a particular genomic region) and the genotype for individual \(i\) at variant \(j\) is coded in the \(i\)th row and \(j\)th column of the genotype matrix \(G\). A genotype of 0 or 2 denotes homozygosity for the common or minor allele, respectively, and a genotype of 1 denotes heterozygosity. Under a baseline model, labeled \(\gamma = 0\), \(y\) is independent of \(G\) and all individuals have a probability of being a case \(\tau_0\). Under the association model, labeled \(\gamma = 1\), individuals either have or do not have a pathogenic configuration of alleles and have probabilities of being a case subject \(\pi\) and \(\tau\), respectively. Whether or not an individual has a pathogenic configuration of alleles depends on a function \(f\) of the genotypes \(G_i\) of that individual, a latent binary vector \(z\) indicating which of the \(k\) variants are pathogenic, a value \(s_i\) equal to the ploidy of the individual at the variant sites, and a variable \(m\) representing the mode of inheritance governing the disease etiology though the \(k\) variants:</p>\[\gamma = 0 : \mathbb{P}(y_i = 1) = \tau_0,\] \[\gamma = 1 : \mathbb{P}(y_i = 1) = \begin{cases} \tau &amp; \text{if } f(G_{i \cdot}, z, s_i, m) = 2, \\ \pi &amp; \text{if } f(G_{i \cdot}, z, s_i, m) = 1. \end{cases}\]<p>The function \(f\) can represent a dominant inheritance model or a recessive inheritance model that accounts for sex-dependent differences in ploidy on the X chromosome (i.e., X-linked recessive inheritance), depending on variable \(m \in \{m_{\text{dom}}, m_{\text{rec}}\}\):</p>\[f(G_{i \cdot}, z, s_i, m_{\text{dom}}) = 1 \left( \sum_j G_{ij} z_j \geq 1 \right),\] \[f(G_{i \cdot}, z, s_i, m_{\text{rec}}) = 1 \left( \sum_j G_{ij} z_j \geq s_i \right).\]<p>Thus, the interpretation of \(z\) depends on the mode of inheritance. In order to have a pathogenic allele configuration, individual \(i\) requires at least one allele at a variant for which \(z_j = 1\) under a dominant model, but \(s_i\) alleles under a recessive model. If genotypes are phased, then a requirement that the \(s_i\) pathogenic alleles are on different haplotypes can be imposed. Recent relatedness is a potential confounder because it is correlated with both case/control status and genotype and, therefore, only unrelated individuals should be included in the model.</p><p>We place beta priors on all three parameters representing risk of disease:</p>\[\tau_0 \sim \text{Beta}(\alpha_0, \beta_0),\] \[\tau \sim \text{Beta}(\alpha_\tau, \beta_\tau),\] \[\pi \sim \text{Beta}(\alpha_\pi, \beta_\pi).\]<p>The mean risk of disease for individuals without a pathogenic combination of alleles in the variants under consideration is uncertain under both models, and thus we place uniform priors on \(\tau_0\) and \(\tau\) by default. However, as pathogenic combinations of alleles typically confer a high disease risk, we suggest setting the hyperparameters for \(\pi\) to \(\alpha_\pi = 6\) and \(\beta_\pi = 1\) (i.e., with a mean of \(\frac{6}{7}\)). However, the prior mean could be adapted, for example, to reflect the consistency with which the disease manifests within families.</p><p>We adopt a logistic regression framework for the prior probability that the variants are pathogenic. The logit of the prior probabilities are shrunk toward a common mean, \(\omega\). If prior information that discriminates between the likely pathogenicity of variants is available, it can be incorporated in the form of a covariate \(c\) with regression coefficient \(\phi\) in the regression equation:</p>\[z_j \sim \text{Bernoulli}(p_j),\] \[\text{logit}(p_j) = \omega + \phi c_j.\]<p>One would typically place a Gaussian prior on the intercept \(\omega\) but, for computational purposes, we prefer to use a logit-beta prior with hyperparameters \(\alpha_\omega\) and \(\beta_\omega\) (see <a href="#app-1">Appendix A</a>). The prior mean of \(\omega\) should reflect the expected proportion of variants that are pathogenic, conditional on an association, and may depend on the filtering procedures used to select the variants to include in the model. By default,</p>\[p(\omega) = 0.20\]<p>reflects a prior expectation that 20% of variants are pathogenic and a prior probability of only 0.01 that the proportion of pathogenic variants exceeds 0.54. This prior is well suited to missense variants but a distribution with a higher mean should be specified if most variants are expected to be pathogenic. This would be the case if the variants under consideration are all protein truncating and thought to be functionally equivalent to each other. To ensure that \(\omega\) can be interpreted as the global mean log odds of pathogenicity, the \(c\) are required to sum to zero. Thus, any user-supplied weights,</p>\[c_j = \tilde{c}_j - \left(\frac{1}{k}\right) \sum_{l} \tilde{c}_l\]<p>are centered such that. We place a log-normal prior on the regression coefficient \(\phi\) to force the effects of the \(c_j\) to be the same as their signs. The prior mean of \(\phi\) is set to 1 so that the \(c_j\) are interpretable as prior shifts in the log odds of pathogenicity relative to the mean. A prior variance on \(\phi\) of 0.35 ensures that the effect of the co-data can be diminished if the co-data are not informative and increased if they improve the model fit.</p><p>Finally, the prior probability on the mode of inheritance parameter \(m\) and the model indicator parameter \(\gamma\) need to be specified. By default, we set the prior probabilities for each mode of inheritance given an association to be the same, i.e.,</p>\[\mathbb{P}(m = m_{\text{dom}} | \gamma = 1) = 0.5,\]<p>and we assume that there is only a 1% chance a priori of an association, i.e.,</p>\[\mathbb{P}(\gamma = 1) = 0.01.\]<p>However, for a particular set of variants, the choice of values for these parameters could be based on the scientific literature or reference variant databases, for example.</p><h3 id="inference"> <a href="#inference" class="anchor-heading" aria-labelledby="inference"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Inference</h3><p>The principal quantity of interest is the posterior probability of the model indicator \(\gamma\), which can be derived from a Bayes factor comparing the two models and \(\mathbb{P}(\gamma)\). The Bayes factor has two components: the evidence under \(\gamma = 0\) and the evidence under \(\gamma = 1\). A closed-form expression exists for the evidence under either model and can be computed rapidly under \(\gamma = 0\), irrespective of \(y\). However, the expression for the evidence under \(\gamma = 1\) contains a sum over every possible value of \(z\), of which there are \(2^k\), and \(k\) is usually large enough to render this sum computationally intractable.</p><p>To tackle this problem, we reviewed various methods for estimating the evidence of a model<sup id="fnref:9"><a href="#fn:9" class="footnote" rel="footnote" role="doc-noteref">4</a></sup> and chose the method of power posteriors<sup id="fnref:10"><a href="#fn:10" class="footnote" rel="footnote" role="doc-noteref">5</a></sup>, which enables the evidence to be estimated by Markov chain Monte Carlo (MCMC) sampling. In this method, the MCMC is tempered, which is helpful in a variable selection setting such as ours because it makes exploration of the space of sets of pathogenic variants more efficient. Samples are drawn from a series of related distributions called power posteriors. Each power posterior has a temperature \(t\) between 0 and 1 and is proportional to the likelihood of the parameters to the power of \(t\) times the prior. These samples can be combined to obtain an estimate of the integrated likelihood (see <a href="#app-1">Appendix A</a>).</p><p>Sampling for our model can be done very efficiently because an MCMC update to \(z_j\) entails changes only in \(f(G_{i \cdot}, z, s_i, m)\) for individuals for whom \(G_{ij} &gt; 0\). For convenience, we estimate the evidence conditional on \(m\) but we can integrate over it through simple summation. Once the MCMC samples have been collected, the marginal posterior probability of \(z\) given \(\gamma\) and \(m\) can be obtained directly and used for ranking variants by their likely pathogenicity. The estimated number of pathogenic variants and the expected posterior number of case subjects explained by the pathogenic variants, given \(\gamma = 1\), can also be computed (see <a href="#app-1">Appendix A</a>). The posterior probability of \(\gamma\) provides a natural means of ranking sets of variants from different loci across the genome.</p><p>The model above assumes that the prior probabilities of variant pathogenicity are conditionally independent. However, particular classes of variants in a locus may confer disease risk, while others may be benign. We can impose a prior correlation structure on the \(z\) reflecting these competing hypotheses by fitting a different association model for each class of variant. If one of the hypotheses matches the true etiology of disease, then this modeling approach can improve model fit and thus increase power. Let \(\gamma \in \{1, 2, \ldots, g\}\) index the association models and let \(I_{uv}\) indicate whether variant \(v\) is included in association model \(u\). Then, we can compute the probability of association across the competing models as:</p>\[\mathbb{P}(\gamma &gt; 0 | y, G, c, I) = \frac{\sum_{u=1}^g \mathbb{P}(y | \gamma = u, G^{(u)}, c^{(u)}) \mathbb{P}(\gamma = u)}{\sum_{u=0}^g \mathbb{P}(y | \gamma = u, G^{(u)}, c^{(u)}) \mathbb{P}(\gamma = u)},\]<p>where \(G^{(u)} = G \cdot \{v : I_{uv} = 1\}\) and \(c^{(u)} = c \{v : I_{uv} = 1\}\). The prior on the model indicator, \(\mathbb{P}(\gamma)\), can be informed by external data. For example, if a gene has a high probability of loss-of-function intolerance<sup id="fnref:11"><a href="#fn:11" class="footnote" rel="footnote" role="doc-noteref">6</a></sup>, then the prior corresponding to a model of high-impact variants in that gene could be up-weighted relative to competing models. We can also compute the posterior probability of variant pathogenicity averaged over all association models using the following expression:</p>\[\mathbb{P}(z | \gamma &gt; 0, y, G, c, I) = \frac{\sum_{u=1}^g \mathbb{P}(z | \gamma = u, y, G^{(u)}, c^{(u)}) \mathbb{P}(\gamma = u | y, G^{(u)}, c^{(u)})}{\sum_{u=1}^g \mathbb{P}(\gamma = u | y, G^{(u)}, c^{(u)})}.\]<p>Other quantities of interest, such as the expected posterior number of cases explained by pathogenic variants, can be averaged over models in the same way.</p><h1 id="references"> <a href="#references" class="anchor-heading" aria-labelledby="references"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> References</h1><p><a href="https://cran.r-project.org/web/packages/BeviMed/">https://cran.r-project.org/web/packages/BeviMed/</a></p><div class="footnotes" role="doc-endnotes"><ol><li id="fn:6"><p>Logsdon, B.A., Dai, J.Y., Auer, P.L., et al., NHLBI GO Exome Sequencing Project. A variational Bayes discrete mixture test for rare variant association. Genet. Epidemiol. 2014; 38:21-30. <a href="#fnref:6" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:7"><p>Quintana, M.A., Berstein, J.L., Thomas, D.C., et al. Incorporating model uncertainty in detecting rare variants: the Bayesian risk index. Genet. Epidemiol. 2011; 35:638-649. <a href="#fnref:7" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:8"><p>Liang, F., Xiong, M. Bayesian detection of causal rare variants under posterior consistency. PLoS ONE. 2013; 8:e69633. <a href="#fnref:8" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:9"><p>Friel, N., Wyse, J. Estimating the evidence–a review. Stat. Neerl. 2012; 66:288-308. <a href="#fnref:9" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:10"><p>Friel, N., Pettitt, A.N. Marginal likelihood estimation via power posteriors. J. R. Stat. Soc. Series B Stat. Methodol. 2008; 70:589-607. <a href="#fnref:10" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:11"><p>Lek, M., Karczewski, K.J., Minikel, E.V., et al., Exome Aggregation Consortium. Analysis of protein-coding genetic variation in 60,706 humans. Nature. 2016; 536:285-291. <a href="#fnref:11" class="reversefootnote" role="doc-backlink">&#8617;</a></p></ol></div><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
