<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Causal inference stats | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Causal inference stats" /><meta name="author" content="Dylan Lawless" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/causal_inference_stats.html" /><meta property="og:url" content="http://localhost:4000/pages/causal_inference_stats.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Causal inference stats" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","author":{"@type":"Person","name":"Dylan Lawless"},"description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Causal inference stats","url":"http://localhost:4000/pages/causal_inference_stats.html"}</script> <script type="text/javascript" defer src="/assets/js/mathjax-script-type.js"> </script> <script type="text/javascript" id="MathJax-script" defer src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"> </script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item active"><a href="/pages/causal_inference_stats.html" class="nav-list-link active">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metadata_users.html" class="nav-list-link">WGS metadata users</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><h1 id="causal-inference-stats-pearl-review"> <a href="#causal-inference-stats-pearl-review" class="anchor-heading" aria-labelledby="causal-inference-stats-pearl-review"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Causal inference stats Pearl review</h1><p>Last update: 20250101</p><p>This document is based on <strong>[1]</strong> “<strong>J. Pearl, “Causal inference in statistics: An overview</strong>,” <em>Statistics Surveys</em>, 2009, <a href="https://ftp.cs.ucla.edu/pub/stat_ser/r350-reprint.pdf">link to pdf</a>. Here is a <a href="https://bayes.cs.ucla.edu/jp_home.html">link to the author’s homepage</a> where you will find the key work highlighted. The complete textbook is <strong>[2]</strong> “<strong>Causality - Models, Reasoning, and Inference</strong>”, Second edition, Judea Pearl, <a href="https://bayes.cs.ucla.edu/BOOK-09/causality2-excerpts.htm">link to author page</a>. The same topics are introduced for a general audience in <strong>[3]</strong> “<strong>The Book of Why</strong>”, Judea Pearl and Dana Mackenzie, 2018, <a href="https://en.wikipedia.org/wiki/The_Book_of_Why">link to Wikipedia</a>.</p><p>This topic is widely ignored. The techincal papers are sometimes verbose and opinionated and thus are difficult to read. An alternative approach is to read the general audience introduction (link <strong>3</strong> above) followed by hands-on work as demonstrated in the tutorial page on <a href="causal_inference_whole_game.html">causal inference in R</a>.</p><h2 id="introduction"> <a href="#introduction" class="anchor-heading" aria-labelledby="introduction"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Introduction</h2><p><strong>Background</strong>: The field of causal inference has advanced significantly, driven by developments in:</p><ul><li>Nonparametric structural equations.<li>Graphical models.<li>Integration of counterfactual and graphical methods.</ul><p><strong>Importance</strong>: These advances are crucial for empirical sciences to address causal questions in health, social, and behavioral sciences.</p><p><strong>Objective</strong>: To summarize key advances in causal inference and illustrate their application in empirical research.</p><h2 id="from-associational-to-causal-analysis-distinctions-and-barriers"> <a href="#from-associational-to-causal-analysis-distinctions-and-barriers" class="anchor-heading" aria-labelledby="from-associational-to-causal-analysis-distinctions-and-barriers"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> From Associational to Causal Analysis: Distinctions and Barriers</h2><p><strong>2.1 Basic Distinction</strong>:</p><ul><li><strong>Statistical Analysis</strong>: Infers parameters from a distribution to describe static conditions.<li><strong>Causal Analysis</strong>: Aims to understand data generation processes, enabling predictions about changes due to interventions or external factors.<li><strong>Key Principle</strong>: Associations cannot substantiate causal claims; causal analysis requires specific causal assumptions.</ul><p><strong>2.2 Formulating the Distinction</strong>:</p><ul><li><strong>Associational Concepts</strong>: Defined by relationships within a joint distribution (e.g., correlation, regression).<li><strong>Causal Concepts</strong>: Involve relationships that extend beyond observable distributions (e.g., randomization, effect, confounding).</ul><p><strong>2.3 Ramifications</strong>:</p><ul><li><strong>Associational vs. Causal</strong>: Confounding, often misconstrued as associational, is inherently causal as it involves discrepancies between observed associations and those under experimental conditions.<li><strong>Notational Challenges</strong>: Traditional statistical methods and language are insufficient for expressing causal relationships.</ul><p><strong>2.4 Barrier of Untested Assumptions</strong>:</p><ul><li>Associational assumptions are potentially testable; causal assumptions are not without experimental control.<li>The clarity of causal assumptions invites scrutiny and requires a shift in how assumptions are conceptualized and communicated.</ul><p><strong>2.5 New Notation</strong>:</p><ul><li>Introduction of specific causal notation is necessary to clearly distinguish between statistical associations and causal relationships.<li>Counterfactual notation (e.g., \(Y_x(u)\) or \(P(Y=y \mid \text{do}(X=x))\)) helps express causal concepts that cannot be captured by traditional probability calculus.</ul><h2 id="the-language-of-diagrams-and-structural-equations"> <a href="#the-language-of-diagrams-and-structural-equations" class="anchor-heading" aria-labelledby="the-language-of-diagrams-and-structural-equations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> The Language of Diagrams and Structural Equations</h2><h3 id="31-linear-structural-equation-models"> <a href="#31-linear-structural-equation-models" class="anchor-heading" aria-labelledby="31-linear-structural-equation-models"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3.1 Linear Structural Equation Models</h3><p><strong>Historical context</strong>: Wright (1921) combined equations and diagrams to illustrate causal relationships, e.g., disease (X) causing a symptom (Y): \(y = \beta x + u\) Equation (3.1) illustrates the level of disease (x) influencing the level of symptom (y) with u representing external factors.</p><p><strong>Path diagrams</strong>: Diagrams clarify causal directions, preventing misinterpretations of equations like reversing (3.1) to suggest symptoms cause diseases.</p><p><strong>Model interpretation</strong>: In path diagrams, causal influence is indicated by arrows, with their absence denoting no direct causal effect. The model includes exogenous variables (U, V) representing unexplained factors influencing endogenous variables (X, Y).</p><p><img src="/assets/images/pearl_causal_inference_1.png" width="80%" /></p><p><img src="/assets/images/pearl_causal_inference_2.png" width="80%" /></p><h3 id="32-from-linear-to-nonparametric-models"> <a href="#32-from-linear-to-nonparametric-models" class="anchor-heading" aria-labelledby="32-from-linear-to-nonparametric-models"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3.2 From Linear to Nonparametric Models</h3><p><strong>Evolution of Structural Equation Modeling (SEM)</strong>: Originally developed for linear relationships, SEM has been extended to nonparametric and nonlinear models.</p><p><strong>Reconceptualizing effect</strong>: Effects in SEM are redefined as general capacities to transmit changes among variables, beyond algebraic coefficients.</p><p><strong>Nonparametric SEM</strong>: The new approach uses structural equations to simulate hypothetical interventions: \(z = f_Z(w)\) \(x = f_X(z, v)\) \(y = f_Y(x, u)\) These functions assume that variables are structurally autonomous, invariant to changes in other equations’ forms.</p><p><strong>Intervention simulation</strong>: The “do” operator (do(x)) represents interventions by setting variables to constants in models, affecting the distribution of other variables: \(x = x_0\) This changes the joint distribution to reflect controlled conditions, providing insights into treatment effects: \(E(Y|do(x'0)) - E(Y|do(x0))\) Equation (3.7) calculates the difference in expected outcomes under different treatment conditions.</p><p><strong>Identifiability and Markovian models</strong>: Identification in causal analysis is feasible under Markovian conditions, where: \(P(v1, v2, ..., vn) = \prod P(vi|pai)\) Equation (3.9) expresses the joint distribution as a product of conditional distributions determined by the graph structure.</p><p><strong>Truncated factorization</strong>: After interventions, distributions are re-factorized to exclude affected variables: \(P(v1, v2, ..., vk|do(x0)) = \prod_{i|Vi \notin X} P(vi|pai)|_{x=x0}\) Equation (3.11) reflects distributions post-intervention, essential for deriving causal effects from observational data.</p><p>These sections outline how causal relationships are modeled using equations and diagrams, and how these models adapt to non-linear and nonparametric forms to estimate effects of interventions in complex systems.</p><p><img src="/assets/images/pearl_causal_inference_3.png" width="80%" /></p><p><img src="/assets/images/pearl_causal_inference_4.png" width="80%" /></p><h2 id="33-counterfactual-analysis-in-structural-models"> <a href="#33-counterfactual-analysis-in-structural-models" class="anchor-heading" aria-labelledby="33-counterfactual-analysis-in-structural-models"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3.3 Counterfactual Analysis in Structural Models</h2><p><strong>Conceptual Framework</strong>: Counterfactual analysis explores causal relationships not expressible by experimental setups or \(P(y|do(x))\), using counterfactual notation \(Y_x(u) = y\) to examine potential outcomes in hypothetical scenarios.</p><p><strong>Interpretation of Counterfactuals</strong>: The approach modifies the original model by replacing variables with constants to reflect hypothetical conditions, thus formalizing counterfactual queries within structural equation models.</p><p>For example, in a modified model: \(Y_{x_0}(u,v,w) = Y_{x_0}(u) = f_Y(x_0,u)\) This represents the potential response \(Y\) to a hypothetical treatment \(x_0\).</p><p><strong>Attribution and Susceptibility</strong>: Counterfactuals allow calculation of effects like change due to treatment, expressed as: \(Y_{x_1} = \beta(x_1 - x_0) + y_0\) This signifies how the outcome \(Y\) would change if \(X\) were \(x_1\) instead of \(x_0\).</p><h2 id="34-an-example-non-compliance-in-clinical-trials"> <a href="#34-an-example-non-compliance-in-clinical-trials" class="anchor-heading" aria-labelledby="34-an-example-non-compliance-in-clinical-trials"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3.4 An Example: Non-Compliance in Clinical Trials</h2><h3 id="formulating-the-assumptions"> <a href="#formulating-the-assumptions" class="anchor-heading" aria-labelledby="formulating-the-assumptions"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Formulating the Assumptions</h3><p><strong>Model Setup</strong>: Considers a clinical trial model with variables \(Z\) (randomized treatment assignment), \(X\) (treatment received), and \(Y\) (response), influenced by unobserved factors \(U\) and \(V\).</p><p><strong>Diagram of a Clinical Trial with Imperfect Compliance</strong>:</p><ul><li><strong>Figure 5(a)</strong>: Shows the causal relationships, including compliance and response pathways.<li><strong>Exclusion Restriction</strong>: Assumes \(Z\) influences \(Y\) only through \(X\), not directly.<li><strong>Randomization Assumption</strong>: \(Z\) is independent of \(U\) and \(V\), ensuring no common cause affecting both.</ul><p><img src="/assets/images/pearl_causal_inference_5.png" width="80%" /></p><h3 id="estimating-causal-effects"> <a href="#estimating-causal-effects" class="anchor-heading" aria-labelledby="estimating-causal-effects"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Estimating Causal Effects</h3><p><strong>Non-identifiability</strong>: The causal effect of treatment on outcome is not identifiable directly due to unobserved confounders \(V\) and \(U\), without further assumptions.</p><p><strong>Instrumental Variables Approach</strong>: Linear case: Uses correlation to determine causal effect, with: \(\beta = \frac{Cov(Z, Y)}{Cov(Z, X)}\) Equation (3.22) shows the use of instrumental variables to estimate causal effects when direct measurement or control of confounders is not possible.</p><h3 id="bounding-causal-effects"> <a href="#bounding-causal-effects" class="anchor-heading" aria-labelledby="bounding-causal-effects"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Bounding Causal Effects</h3><p><strong>Bounds Determination</strong>: When exact causal effects can’t be identified, the analysis seeks to establish bounds on these effects based on observed data.</p><p><strong>Instrumental Inequality</strong>: Ensures any variable \(Z\) serves as an instrument only if it meets specific conditions, reflected by the inequality: \(\max \left[ \max P(x, y|z) \right] \leq 1\) Equation (3.25) imposes constraints on the observed distribution to maintain the validity of the instrumental assumptions.</p><h3 id="testable-implications"> <a href="#testable-implications" class="anchor-heading" aria-labelledby="testable-implications"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Testable Implications</h3><p><strong>Detection of Violations</strong>: Specific inequalities derived from the model help detect violations of assumptions regarding the absence of direct effects of \(Z\) on \(Y\) or hidden biases.</p><p><strong>Non-compliance and Side Effects</strong>: The model assesses non-compliance impacts and side effects through observational inequalities, aiding in understanding the limitations and robustness of causal inferences in clinical trials with partial compliance.</p><h2 id="4-the-language-of-potential-outcomes-and-counterfactuals"> <a href="#4-the-language-of-potential-outcomes-and-counterfactuals" class="anchor-heading" aria-labelledby="4-the-language-of-potential-outcomes-and-counterfactuals"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4 The Language of Potential Outcomes and Counterfactuals</h2><h3 id="41-formulating-assumptions"> <a href="#41-formulating-assumptions" class="anchor-heading" aria-labelledby="41-formulating-assumptions"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4.1 Formulating Assumptions</h3><ul><li><strong>Concept</strong>: The potential outcome framework utilises Yx(u), representing the outcome Y for unit u if X were x, handling this as a random variable Yx.<li><strong>Axiomatic Framework</strong>: The framework postulates a comprehensive probability function covering both real and hypothetical events.<li><strong>Consistency Constraints</strong>: It asserts that if X equals x, then Yx must match the observed Y. \(X = x \Rightarrow Yx = Y\) Equation (4.1) aligns the potential outcome with the observed outcome when the intervention aligns with the actual treatment.</ul><h3 id="42-performing-inferences"> <a href="#42-performing-inferences" class="anchor-heading" aria-labelledby="42-performing-inferences"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4.2 Performing Inferences</h3><p><strong>Conditional Ignorability</strong>: Assumes conditional independence of Yx and X given Z (Yx ⊥ X | Z), allowing for straightforward calculation of causal effects using potential outcomes: \(P(Yx = y) = \sum_z P(Y = y | X = x, z)P(z)\) Equation (4.5) effectively aligns with standard covariate adjustment formulas, leveraging observed data to infer counterfactual scenarios.</p><h2 id="43-combining-graphs-and-algebra"> <a href="#43-combining-graphs-and-algebra" class="anchor-heading" aria-labelledby="43-combining-graphs-and-algebra"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4.3 Combining Graphs and Algebra</h2><ul><li><strong>Translation of Assumptions</strong>: Graphs convey causal assumptions, translating into counterfactual notation to express and verify causal relationships.<li><strong>Parent Sets and Exclusion Restrictions</strong>: Define relationships and independence among variables, using the structure to ensure interventions only affect relevant pathways.<li><strong>Graph-Assisted Analysis</strong>: Graphs help validate and explore dependencies and independencies essential for deriving causal inferences, using criteria like d-separation for testing conditional independencies.</ul><h2 id="5-conclusions"> <a href="#5-conclusions" class="anchor-heading" aria-labelledby="5-conclusions"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 5 Conclusions</h2><p><strong>Structural Equation Models (SEMs)</strong>: Presented as a formal language for articulating causal assumptions, enabling clear representation of complex causal relationships.</p><p><strong>Integration of Methods</strong>: Combining SEMs with potential outcomes enriches the statistical toolkit, allowing for robust empirical research through a unified approach that encapsulates randomization, intervention, effects, and confounding in a coherent framework.</p><p>This synthesis of graphical models and algebraic potential outcomes facilitates a deeper understanding and application of causal inference techniques in statistical research, enhancing both theoretical and practical approaches to exploring causality in complex systems.</p><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
