<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Data concepts | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Data concepts" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/data_concepts.html" /><meta property="og:url" content="http://localhost:4000/pages/data_concepts.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Data concepts" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Data concepts","url":"http://localhost:4000/pages/data_concepts.html"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item active"><a href="/pages/data_concepts.html" class="nav-list-link active">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metadata_users.html" class="nav-list-link">WGS metadata users</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/my_voice.html" class="nav-list-link">QV - My voice</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><h1 id="data-concepts"> <a href="#data-concepts" class="anchor-heading" aria-labelledby="data-concepts"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data concepts</h1><p>Last update: 20230619</p><details open="" class="no_toc"> <summary class="text-delta">Table of contents</summary><ul id="markdown-toc"><li><a href="#data-concepts" id="markdown-toc-data-concepts">Data concepts</a><ul><li><a href="#how-data-is-collected-and-stored" id="markdown-toc-how-data-is-collected-and-stored">How data is collected and stored</a><li><a href="#database-concepts" id="markdown-toc-database-concepts">Database concepts</a><li><a href="#availability" id="markdown-toc-availability">Availability</a><li><a href="#specific-genetic-examples" id="markdown-toc-specific-genetic-examples">Specific genetic examples</a><li><a href="#potential-concepts" id="markdown-toc-potential-concepts">Potential concepts</a><li><a href="#examples" id="markdown-toc-examples">Examples</a><ul><li><a href="#genomics-england-ge" id="markdown-toc-genomics-england-ge">Genomics England (GE)</a><li><a href="#opencb" id="markdown-toc-opencb">OpenCB</a><li><a href="#ontobee" id="markdown-toc-ontobee">Ontobee</a></ul></ul></ul></details><hr /><h2 id="how-data-is-collected-and-stored"> <a href="#how-data-is-collected-and-stored" class="anchor-heading" aria-labelledby="how-data-is-collected-and-stored"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> How data is collected and stored</h2><ul><li>The Swiss Personalised Health Network (SPHN): collects and hosts omic data that we will ultimately use<li>Branch of SPHN: <a href="https://sphn.ch/network/data-coordination-center/">Data Coordination Center (DCC)</a><li>DCC is also a part of <a href="https://www.sib.swiss/personalized-health-informatics">Swiss Institute of Bioinformatics (SIB)</a><li>The data is to be stored and processed on <a href="https://www.biomedit.ch">BioMedIT network</a><li>Our tenant on BioMedIT is called MOMIC and this is on the <a href="https://scicore.ch">sciCORE infrastructure</a>.<li>The database structure requires Resource Description Framework <a href="https://www.biomedit.ch/rdf/sphn-ontology/sphn">(RDF) Schema</a></ul><h2 id="database-concepts"> <a href="#database-concepts" class="anchor-heading" aria-labelledby="database-concepts"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Database concepts</h2><p>The DCC is responsible for making research data suitable for database management. To do this they design the concepts. Training <a href="https://sphn.ch/training/">material available here</a>, but is a DCC responsibility.</p><p>Several concepts for the types of data that we use alredy exist. e.g. genetic data concepts.</p><p>We also assist in generating new omics-related concepts for future expansion. This page will be updated to summarise the progress as these new concepts are defined.</p><p><img title="Illustration Semantic-Interoperability-Framework" alt="Alt text" src="https://www.sib.swiss/images/sib/7-about-us/media/news_2021/Semantic-Interoperability-Framework.png" /></p><h2 id="availability"> <a href="#availability" class="anchor-heading" aria-labelledby="availability"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Availability</h2><p>The following information is copied from <a href="https://sphn.ch/2023/03/20/sphn-dataset-rdf-schema-2023-release/">https://sphn.ch/2023/03/20/sphn-dataset-rdf-schema-2023-release/</a></p><ul><li>The SPHN Dataset is openly available here: <a href="https://sphn.ch/document/sphn-dataset/">https://sphn.ch/document/sphn-dataset/</a><li>The SPHN RDF Schema is browsable here: <a href="https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2">https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2</a><li>The external terminologies in RDF are accessible on BioMedIT Portal: <a href="https://portal.dcc.sib.swiss/">https://portal.dcc.sib.swiss/</a><li>The Quality Assurance Framework is available here: <a href="https://git.dcc.sib.swiss/sphn-semantic-framework/sphn-ontology/-/tree/master/quality_assurance">https://git.dcc.sib.swiss/sphn-semantic-framework/sphn-ontology/-/tree/master/quality_assurance</a><li>Project templates are available here: <a href="https://git.dcc.sib.swiss/sphn-semantic-framework/sphn-ontology/-/tree/master/templates">https://git.dcc.sib.swiss/sphn-semantic-framework/sphn-ontology/-/tree/master/templates</a><li>A comprehensive documentation is openly accessible here: <a href="https://sphn-semantic-framework.readthedocs.io/en/latest/">https://sphn-semantic-framework.readthedocs.io/en/latest/</a></ul><h2 id="specific-genetic-examples"> <a href="#specific-genetic-examples" class="anchor-heading" aria-labelledby="specific-genetic-examples"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Specific genetic examples</h2><ul><li><a href="https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2#GeneticVariation">https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2#GeneticVariation</a><li><a href="https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2#SingleNucleotideVariation">https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2#SingleNucleotideVariation</a><li><a href="https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2#VariantDescriptor">https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2#VariantDescriptor</a><li><a href="https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2#VariantNotation">https://www.biomedit.ch/rdf/sphn-ontology/sphn/2023/2#VariantNotation</a></ul><h2 id="potential-concepts"> <a href="#potential-concepts" class="anchor-heading" aria-labelledby="potential-concepts"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Potential concepts</h2><hr /><p><img src="/assets/images/concept_ideas_1.png" width="100%" /></p><hr /><p><img src="/assets/images/concept_ideas_2.png" width="100%" /></p><hr /><p><img src="/assets/images/concept_ideas_3.png" width="100%" /></p><hr /><p><img src="/assets/images/concept_ideas_4.png" width="100%" /></p><h2 id="examples"> <a href="#examples" class="anchor-heading" aria-labelledby="examples"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Examples</h2><h3 id="genomics-england-ge"> <a href="#genomics-england-ge" class="anchor-heading" aria-labelledby="genomics-england-ge"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Genomics England (GE)</h3><p>GE represents one of the best national platforms so far and they have complete analysis of ~200’000 genomes for clinical use.</p><ul><li><a href="https://www.genomicsengland.co.uk">Homepage</a><li><a href="https://www.genomicsengland.co.uk/bioinformatics">Bioinformatics page</a></ul><p>GE model documentation shows the concepts in use: <a href="http://gelreportmodels.genomicsengland.co.uk/models.html#">http://gelreportmodels.genomicsengland.co.uk/models.html#</a>. Concept development could start with the GE 1.3.0-SNAPSHOT. For example, <code class="language-plaintext highlighter-rouge">VariantMetadata</code> is a good starting place since it includes “individual” (i.e. subject), “sample type”, “experiment”, etc., which are required in most genomic data scenarios. We could work through a prioritised list of concepts over several months-years for any new concept that matches user requirements. They have derived some logical structures derived from Ontobee, OpenCB, and probably other common sources.</p><p><strong>Example: metadata</strong></p><p>Here are some examples which most users will require, from <code class="language-plaintext highlighter-rouge">VariantMetadata</code>:</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>|--VariantMetadata:
   |-- Cohort
      |-- Experiment
         |--- center, date, molecule, technique, library, libraryLayout, platform, description
   |-- Individual
       |-- id, family, father, mother, sex, phenotype, samples
   |-- Program
       |-- ...
   |-- Sample
       |-- ...
   |-- SampleSetType
	    |-- CASE_CONTROL, CASE_SET, CONTROL_SET, PAIRED, TIME_SERIES, FAMILY, TRIO, MISCELLANEOUS, UNKNOWN
   |--Species
       |-- ...
</code></pre></div></div><p><strong>Example: variant</strong> Some of the most important concepts for our genomics needs are listed on the <a href="http://gelreportmodels.genomicsengland.co.uk/html_schemas/org.opencb.biodata.models.variant.avro/1.3.0-SNAPSHOT/variant.html#/schema/org.opencb.biodata.models.variant.avro.Variants">variant procol page</a>, which come from org.opencb.biodata.models.variant.avro. These are beyond our current needs but this list of 55 entries likely cover most of the conceivable needs:</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AdditionalAttribute, AlleleOrigin Enum, AllelesCode Enum, AlternateCoordinate, ClinVar, ClinicalSignificance Enum, Confidence Enum, ConsequenceType, ConsistencyStatus Enum, Cosmic, Cytoband, Drug, DrugResponseClassification Enum, EthnicCategory Enum, EvidenceEntry, EvidenceImpact Enum, EvidenceSource, EvidenceSubmission, ExonOverlap, Expression, ExpressionCall Enum, FeatureTypes Enum, FileEntry, GeneDrugInteraction, GeneTraitAssociation, GenomicFeature, Genotype, Gwas, HeritableTrait, ModeOfInheritance Enum, Penetrance Enum, PopulationFrequency, Property, ProteinFeature, ProteinVariantAnnotation, Repeat, Score, SequenceOntologyTerm, SomaticInformation, StructuralVariantType Enum, StructuralVariation, StudyEntry, TraitAssociation Enum, TumorigenesisClassification Enum, VariantAnnotation, VariantAvro, VariantClassification, VariantFunctionalEffect Enum, VariantHardyWeinbergStats, VariantStats, VariantTraitAssociation, VariantType Enum, Xref,
</code></pre></div></div><p>Let’s pick one example from that list which would come under the heading “variant interpretation” - <code class="language-plaintext highlighter-rouge">ClinicalSignificance</code>:</p><p><strong>Example: variant interpretation - <code class="language-plaintext highlighter-rouge">ClinicalSignificance</code></strong></p><p>We use this exact example for our clinical genetics work: <a href="http://gelreportmodels.genomicsengland.co.uk/html_schemas/org.opencb.biodata.models.variant.avro/1.3.0-SNAPSHOT/variant.html#/schema/org.opencb.biodata.models.variant.avro.ClinicalSignificance">ClinicalSignificance</a>.</p><p>Mendelian variants classification with ACMG terminology as defined in Richards, S. et al. (2015). Standards and guidelines for the interpretation of sequence variants: a joint consensus recommendation of the American College of Medical Genetics and Genomics and the Association for Molecular Pathology. Genetics in Medicine, 17(5), 405?423. <a href="https://doi.org/10.1038/gim.2015.30.">https://doi.org/10.1038/gim.2015.30.</a></p><p>Classification for variants associated with disease, etc., based on the ACMG recommendations and ClinVar classification (<a href="https://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/">https://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/</a>).</p><ul><li><code class="language-plaintext highlighter-rouge">benign_variant</code> : Benign variants interpreted for Mendelian disorders<li><code class="language-plaintext highlighter-rouge">likely_benign_variant</code> : Likely benign variants interpreted for Mendelian disorders with a certainty of at least 90%<li><code class="language-plaintext highlighter-rouge">pathogenic_variant</code> : Pathogenic variants interpreted for Mendelian disorders<li><code class="language-plaintext highlighter-rouge">likely_pathogenic_variant</code> : Likely pathogenic variants interpreted for Mendelian disorders with a certainty of at least 90%<li><code class="language-plaintext highlighter-rouge">uncertain_significance</code> : Uncertain significance variants interpreted for Mendelian disorders. Variants with conflicting evidences should be classified as uncertain_significance<li>Enum symbols:<ul><li><code class="language-plaintext highlighter-rouge">benign</code>, <code class="language-plaintext highlighter-rouge">likely_benign</code>, <code class="language-plaintext highlighter-rouge">VUS</code>, <code class="language-plaintext highlighter-rouge">likely_pathogenic</code>, <code class="language-plaintext highlighter-rouge">pathogenic</code>, <code class="language-plaintext highlighter-rouge">uncertain_significance</code></ul></ul><p>However, we do not restrict our use of ACMG standard for variant interpretation using only <code class="language-plaintext highlighter-rouge">ClinicalSignificance</code>, since there are a large number of other variant interpretation datasets which can be used to make the final determination. One method we use is the ACMG scoring sytem to score all variants based on the ACMG evidence categorisation method. Therefore, <code class="language-plaintext highlighter-rouge">ACMG_score</code> could be derived from <code class="language-plaintext highlighter-rouge">ClinicalSignificance</code>, but also from other sources. Perhaps <a href="http://gelreportmodels.genomicsengland.co.uk/html_schemas/org.opencb.biodata.models.variant.avro/1.3.0-SNAPSHOT/variant.html#/schema/org.opencb.biodata.models.variant.avro.EvidenceEntry">EvidenceEntry</a> is a major entry which includes most of these subtypes for interpretation evidence.</p><h3 id="opencb"> <a href="#opencb" class="anchor-heading" aria-labelledby="opencb"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> OpenCB</h3><ul><li>Used for some Genomics England concepts<li>OpenCB which provides scalable a storage engine framework with data and metadata catalogue.<li><a href="https://github.com/opencb/opencga">https://github.com/opencb/opencga</a><li><a href="http://docs.opencb.org">http://docs.opencb.org</a></ul><h3 id="ontobee"> <a href="#ontobee" class="anchor-heading" aria-labelledby="ontobee"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Ontobee</h3><ul><li>Used for some Genomics England concepts<li>Ontology data server with RDF source code.<li>Example GE concept for “Variants / AlleleOrigin / germline_variant” comes from :<li><a href="https://ontobee.org/ontology/SO?iri=http://purl.obolibrary.org/obo/SO_0001762">https://ontobee.org/ontology/SO?iri=http://purl.obolibrary.org/obo/SO_0001762</a><li><a href="https://ontobee.org/ontology/rdf/SO?iri=http://purl.obolibrary.org/obo/SO_0001778">germline_variant RDF source code</a></ul><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>|-- sequence_attribute
   |-- variant_quality
   |-- variant_origin
      |-- maternal_variant
      |-- paternal_variant
      |-- somatic_variant
      |-- pedigree_specific_variant
      |-- population_specific_variant
      |-- de_novo_variant
      |-- germline_variant
         |-- RDF sourcode 
</code></pre></div></div><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
