<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Design QV SNV INDEL v1 | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Design QV SNV INDEL v1" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/design_qv_snvindel_v1.html" /><meta property="og:url" content="http://localhost:4000/pages/design_qv_snvindel_v1.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Design QV SNV INDEL v1" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Design QV SNV INDEL v1","url":"http://localhost:4000/pages/design_qv_snvindel_v1.html"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item active"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item active"><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link active">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/pages/design_doc.html">Design documents</a><li class="breadcrumb-nav-list-item"><span>Design QV SNV INDEL v1</span></ol></nav><div id="main-content" class="main-content" role="main"><p>Last update: 20241217</p><h1 id="design-qv-snv-indel-v1"> <a href="#design-qv-snv-indel-v1" class="anchor-heading" aria-labelledby="design-qv-snv-indel-v1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Design QV SNV INDEL v1</h1><h2 id="qualifying-variant-protocol-for-snv-indel-v1"> <a href="#qualifying-variant-protocol-for-snv-indel-v1" class="anchor-heading" aria-labelledby="qualifying-variant-protocol-for-snv-indel-v1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Qualifying variant protocol for SNV INDEL v1</h2><hr /><ul id="markdown-toc"><li><a href="#design-qv-snv-indel-v1" id="markdown-toc-design-qv-snv-indel-v1">Design QV SNV INDEL v1</a><ul><li><a href="#qualifying-variant-protocol-for-snv-indel-v1" id="markdown-toc-qualifying-variant-protocol-for-snv-indel-v1">Qualifying variant protocol for SNV INDEL v1</a><li><a href="#introduction" id="markdown-toc-introduction">Introduction</a><li><a href="#variables-and-thresholds" id="markdown-toc-variables-and-thresholds">Variables and thresholds</a><li><a href="#variables-file" id="markdown-toc-variables-file">Variables file</a><li><a href="#the-use-of-qv-in-analysis-plan" id="markdown-toc-the-use-of-qv-in-analysis-plan">The use of QV in analysis plan</a></ul><li><a href="#references" id="markdown-toc-references">References</a></ul><hr /><p><strong>Protocol name</strong>: <code class="language-plaintext highlighter-rouge">design_qv_snvindel_v1</code></p><h2 id="introduction"> <a href="#introduction" class="anchor-heading" aria-labelledby="introduction"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Introduction</h2><p>This pages summarises the qualifying variant (QV) settings used for <code class="language-plaintext highlighter-rouge">QV SNV INDEL v1</code>. QV defines those variants which remain after filtering for use in downstream analysis. In general, most choices are based on well established best practices. For any subjective choices of thesholds, we will rely on evidence-based reasoning. However, it is important for users to understand whether their QV is suited to a particular context. For example, a GWAS analysis typically relies on the presence of common shared variants while a clinical genetics report might focus on rare novel variants. Conversely, cohort-level a genetic relationship matrix for use in statistical analysis might rely on some but not all of the QV steps (such as dropping some filters so that haplotypes can be correcly represented). You can read about QV in this review <sup id="fnref:povysil2019rare"><a href="#fn:povysil2019rare" class="footnote" rel="footnote" role="doc-noteref">1</a></sup> <a href="https://www.nature.com/articles/s41576-019-0177-4/">https://www.nature.com/articles/s41576-019-0177-4/</a>.</p><p>Examples of QV sets are: (1) <strong>QC-only QV</strong> resulting in a very large remaining variant dataset, e.g. &gt;500’000 variants per subject. (2) <strong>Rare disease QV</strong> resulting in a small dataset with strict filters, e.g. 10’000 variants per subject. (3) <strong>Flexible QV</strong> resulting in disease-causing candidate variants with modest filtering to balance QC and false positives, e.g. &lt;100’000 variants per subject.</p><p>Several QV protocols can be piped together to create increasingly filtered datasets to match the needs at a certain stage of analysis. It is also typical that different analysis from QVs sets are used and the final results from each step are merged to cover multiple scenarios. For example, <code class="language-plaintext highlighter-rouge">SNV/INDEL QV</code> + <code class="language-plaintext highlighter-rouge">CNV QV</code> + <code class="language-plaintext highlighter-rouge">rare disease known QV</code> + <code class="language-plaintext highlighter-rouge">statitcal assossiation QC</code> may be merged to reach the final combined analysis of (1) single case-level known disease causing results with (2) newly identified cohort-level genes associated with disease.</p><p>The following <strong>figure 1</strong> from Hillman et. al. <sup id="fnref:hillman2020identification"><a href="#fn:hillman2020identification" class="footnote" rel="footnote" role="doc-noteref">2</a></sup> is a typical illustration of QV workflow. We do not use this reference for any analysis. The figure is simply shown for illustration. <strong>Figure 2</strong> from Povysil et al. <sup id="fnref:povysil2019rare:1"><a href="#fn:povysil2019rare" class="footnote" rel="footnote" role="doc-noteref">1</a></sup> provides an example of the application of QV in an analysis.</p><p><img src="/pages/design_doc/images/Variant-filtering-and-annotation-workflow-with-summative-case-and-control-variant.png" width="100%" /> Figure 1: From Hillman et. al. <sup id="fnref:hillman2020identification:1"><a href="#fn:hillman2020identification" class="footnote" rel="footnote" role="doc-noteref">2</a></sup>. Variant filtering and annotation workflow with summative case and control variant and gene counts. Top. General workflow and filtering parameters applied to data. Bottom. Table summarizing total qualifying variant and gene counts by ethnicity and in total for cases and controls. Variant counts represent the number of unique qualifying variant loci identified in each ethnicity and network. Gene counts represent the number of genes containing at least one qualifying variant in each network and ethnicity. AMR, American Latino; CADD, combined annotation dependent depletion; dbNSFP, database of non-synonymous single-nucleotide variant functional predictions; EA, European Americans; FOCM, Folate and One-Carbon Metabolism; GATK, Genome Analysis Toolkit; GHOS, Glucose Homeostasis and Oxidative Stress; gnomAD, genome aggregation database; MA, Mexican Americans; NFE, Non-Finnish Europeans; VCF, variant call format.</p><h2 id="variables-and-thresholds"> <a href="#variables-and-thresholds" class="anchor-heading" aria-labelledby="variables-and-thresholds"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Variables and thresholds</h2><p>This protocol shown here (<code class="language-plaintext highlighter-rouge">SNV INDEL v1</code>) is considered as the flexible format.</p><p><strong>QC-only QV</strong></p><ol><li><strong>[QC]</strong> <code class="language-plaintext highlighter-rouge">01_fastp.sh</code> The tool fastp is used for QC. FASTQ that fail are investigated or removed. See <a href="fastp.html">fastp</a> for more.<li><strong>[QC]</strong> <code class="language-plaintext highlighter-rouge">03b_collectwgsmetrics.sh</code> BAMs that fail are investigated or removed. See <a href="metrics_CollectWgsMetrics.html">metrics_CollectWgsMetrics</a> for more.<li><strong>[QC]</strong> <code class="language-plaintext highlighter-rouge">05_rmdup_merge.sh</code> is used to mark optical duplicates. See <a href="design_doc/gatk_duplicates.html">GATK Duplicates</a> for more.<li><strong>[QC]</strong> <code class="language-plaintext highlighter-rouge">07_haplotype_caller.sh</code> used <code class="language-plaintext highlighter-rouge">-ERC GVCF</code> mode. This does not remove variants but unlike <code class="language-plaintext highlighter-rouge">BP_RESOLUTION</code>, <code class="language-plaintext highlighter-rouge">GVCF</code> mode condenses non-variant blocks which could be misunderstood later as missing if not recongnised by the user, for example in a genotype matrix which has been merged with other cohorts. See <a href="vcf.html">VCF</a> and <a href="vcf_gvcf.html">VCF and gVCF</a> for more.<li><strong>[QC]</strong> <code class="language-plaintext highlighter-rouge">07c_qc_summary_stats.sh</code> is used to log QC. This implements <code class="language-plaintext highlighter-rouge">bcftools stats</code> (<a href="https://samtools.github.io/bcftools/bcftools.html#stats">https://samtools.github.io/bcftools/bcftools.html#stats</a>) and subsequently the bcftools <code class="language-plaintext highlighter-rouge">plot-vcfstats</code> (<a href="https://samtools.github.io/bcftools/bcftools.html#plot-vcfstats">https://samtools.github.io/bcftools/bcftools.html#plot-vcfstats</a>) using <code class="language-plaintext highlighter-rouge">python -m venv envQCplot</code> with matplotlib. Subjects fail are investigated or removed. See <a href="metrics_bcftoolsstats.html">metrics_bcftoolsstats</a>.<li><strong>[QC]</strong> <strong>!Not used</strong> Large cohort hard-filter. The following step is generllay useful before VQSR (shown later), however since our cohort is likely to contain some large populations of unrelated patients with the <strong>same causal genetic variants</strong>, we skip it to avoid loosing them. (<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531112--How-to-Filter-variants-either-with-VQSR-or-by-hard-filtering">https://gatk.broadinstitute.org/hc/en-us/articles/360035531112--How-to-Filter-variants-either-with-VQSR-or-by-hard-filtering</a>) [A] Hard-filtter a large cohort callset on ExcessHet (<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036726851-ExcessHet">https://gatk.broadinstitute.org/hc/en-us/articles/360036726851-ExcessHet</a>) using VariantFiltration (<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036350452-VariantFiltration">https://gatk.broadinstitute.org/hc/en-us/articles/360036350452-VariantFiltration</a>). ExcessHet filtering applies only to callsets with a large number of samples, e.g. hundreds of unrelated samples. Small cohorts should not trigger ExcessHet filtering as values should remain small. Note cohorts of consanguinous samples will inflate ExcessHet, and it is possible to limit the annotation to founders for such cohorts by providing a pedigree file during variant calling. For example, <code class="language-plaintext highlighter-rouge">"ExcessHet &gt; 54.69" </code> produces a VCF callset where any record with ExcessHet greater than 54.69 is filtered with the ExcessHet label in the FILTER column. The phred-scaled 54.69 corresponds to a z-score of -4.5. If a record lacks the ExcessHet annotation, it will pass filters. If run, this step would be followed by <code class="language-plaintext highlighter-rouge">MakeSitesOnlyVcf</code> (<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036346712-MakeSitesOnlyVcf-Picard-">https://gatk.broadinstitute.org/hc/en-us/articles/360036346712-MakeSitesOnlyVcf-Picard-</a>) to speed up analysis.<li><strong>[QC]</strong> <code class="language-plaintext highlighter-rouge">10_vqsr.sh</code> applies the Variant Quality Score Recalibration (VQSR) method in GATK, a sophisticated approach used to assess variant quality and filter out sequencing and data processing artifacts. Our settings are shown in table below; they include priors for SNPs from <code class="language-plaintext highlighter-rouge">HapMap=15.0</code>, <code class="language-plaintext highlighter-rouge">Omni=12.0</code>, <code class="language-plaintext highlighter-rouge">1000G=10.0</code>, and <code class="language-plaintext highlighter-rouge">Truth Sensitivity = 99.7</code>, for INDELS <code class="language-plaintext highlighter-rouge">Mills 1KG=12.0</code>, <code class="language-plaintext highlighter-rouge">dbSNP=2.0</code> and <code class="language-plaintext highlighter-rouge">Truth Sensitivity=95</code> and across all, the annotations QD, MQRankSum, ReadPosRankSum, FS, SOR. This two-step procedure begins with building a recalibration model that evaluates the relationship between variant annotations, such as Quality by Depth (QD), Mapping Quality (MQ), and Read Position Rank Sum Test (ReadPosRankSum), against the likelihood that a variant is genuine. Key resources like HapMap and Omni 2.5M SNP chip array data are leveraged to train this model adaptively. By employing an adaptive error model and a Gaussian mixture model, VQSR provides a score known as VQSLOD (Variant Quality Score Log Odds), which indicates the likelihood of each variant being true. The VQSLOD scores are incorporated into the variant call file (VCF), allowing for highly precise and flexible filtering. The overall aim of VQSR is to allocate a calibrated probability to each variant, surpassing traditional methods that rely strictly on fixed annotation value thresholds. This technique is especially crucial in contexts where accuracy in variant filtering significantly impacts subsequent analyses and conclusions, particularly in medical genomics research. Population resource files are publically available at <a href="https://console.cloud.google.com/storage/browser/broad-references/hg38/v0/">https://console.cloud.google.com/storage/browser/broad-references/hg38/v0/</a>. Further reading about techniques here <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531112--How-to-Filter-variants-either-with-VQSR-or-by-hard-filtering">https://gatk.broadinstitute.org/hc/en-us/articles/360035531112--How-to-Filter-variants-either-with-VQSR-or-by-hard-filtering</a> and here <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531612-Variant-Quality-Score-Recalibration-VQSR?id=1259">https://gatk.broadinstitute.org/hc/en-us/articles/360035531612-Variant-Quality-Score-Recalibration-VQSR?id=1259</a>.<li><strong>[QC]</strong> see <code class="language-plaintext highlighter-rouge">10b_qc_summary_stats</code> for plink logs <code class="language-plaintext highlighter-rouge">${QC_SUMMARY_STATS}/vqsr/${QV_MODEL}_chr${INDEX}_summary_stat.log</code><li><strong>[QC]</strong> <strong>Optional</strong> CollectVariantCallingMetrics. An example is shown here <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531112--How-to-Filter-variants-either-with-VQSR-or-by-hard-filtering">https://gatk.broadinstitute.org/hc/en-us/articles/360035531112--How-to-Filter-variants-either-with-VQSR-or-by-hard-filtering</a> and documentation here <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036363592-CollectVariantCallingMetrics-Picard">https://gatk.broadinstitute.org/hc/en-us/articles/360036363592-CollectVariantCallingMetrics-Picard</a>.<li><strong>[QC]</strong> <strong>Optional</strong> Other Picard metrics which are not in use: <a href="https://broadinstitute.github.io/picard/picard-metric-definitions.html">https://broadinstitute.github.io/picard/picard-metric-definitions.html</a>.<li><strong>[QC]</strong> <code class="language-plaintext highlighter-rouge">11_genotype_refinement.sh</code> Genotype Refinement workflow is a collection of steps <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531432-Genotype-Refinement-workflow-for-germline-short-variants">https://gatk.broadinstitute.org/hc/en-us/articles/360035531432-Genotype-Refinement-workflow-for-germline-short-variants</a>. We will continue to add steps to our current implementation. For example, when we have <strong>trio studies</strong> we can provide more nuanced variant confidence methods. We currently use (1) <code class="language-plaintext highlighter-rouge">CalculateGenotypePosteriors</code>: This tool refines genotype probabilities using family data or population allele frequencies. We use gnomAD <code class="language-plaintext highlighter-rouge">af-only-gnomad.hg38.vcf.gz</code>. It enhances the initial genotype likelihoods from variant callers, ensuring accuracy especially when incorporating trio and population data. This step reduces false positives and improves genotype precision. <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037226592-CalculateGenotypePosteriors">https://gatk.broadinstitute.org/hc/en-us/articles/360037226592-CalculateGenotypePosteriors</a>. (2) <code class="language-plaintext highlighter-rouge">VariantFiltration</code>: This stage applies filters. We use <code class="language-plaintext highlighter-rouge">GQ &lt; 20</code> on genotype quality scores to flag lower confidence genotypes. It refines variant call quality by annotating individual genotypes based on specified criteria, helping to isolate high-confidence calls from potential errors. <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360041850471-VariantFiltration">https://gatk.broadinstitute.org/hc/en-us/articles/360041850471-VariantFiltration</a>.<li><strong>[QC]</strong> The same stats as <code class="language-plaintext highlighter-rouge">10b_qc_summary_stats</code> for plink logs are run in step <code class="language-plaintext highlighter-rouge">11_genotype_refinement.sh</code>.<li><strong>[Filter]</strong> <code class="language-plaintext highlighter-rouge">12_pre_annotation_processing.sh</code>. We apply the following <code class="language-plaintext highlighter-rouge">bcftools filter</code> with (1) QUAL&gt;=30: Filters for variants where the quality score is 30 or higher. The quality score (QUAL) represents the confidence in the correctness of the variant call. (2) INFO/DP&gt;=20: Filters for variants where the total depth of quality base calls (DP in the INFO field of a VCF file) is 20 or more. This is a measure of the total amount of sequencing data that supports the variants in the file. (3) FORMAT/DP&gt;=10: Applies a filter to each individual sample in the VCF file, ensuring that the depth of reads for each genotype call is 10 or more. This ensures that each genotype call in every sample has sufficient data supporting it. (4) FORMAT/GQ&gt;=20: Ensures that the genotype quality (GQ) for each sample’s call is 20 or above. The genotype quality score expresses the confidence in the assignment of the genotype at that specific locus. (5) Then GATK <code class="language-plaintext highlighter-rouge">SelectVariants</code> is subsequently applied with <code class="language-plaintext highlighter-rouge">--exclude-filtered</code>, <code class="language-plaintext highlighter-rouge">--exclude-non-variants</code>, and <code class="language-plaintext highlighter-rouge">--remove-unused-alternates</code>.<li><strong>[QC]</strong> <code class="language-plaintext highlighter-rouge">12_pre_annotation_processing.sh</code> also contains the steps using vt (variant tool) <a href="https://genome.sph.umich.edu/wiki/Vt">https://genome.sph.umich.edu/wiki/Vt</a> to run (1) <code class="language-plaintext highlighter-rouge">variant decompose</code> <a href="https://genome.sph.umich.edu/wiki/Vt#Decompose">https://genome.sph.umich.edu/wiki/Vt#Decompose</a> a complex operation to ensure that multiallelic variants (a feature of VCF format) are split into distinct observations to remove errors in downstream analysis and (2) <code class="language-plaintext highlighter-rouge">variant normalisation</code> <a href="https://genome.sph.umich.edu/wiki/Variant_Normalization">https://genome.sph.umich.edu/wiki/Variant_Normalization</a> a proof-bacjed procedure to ensure that variant representations conform to a consensus format. See Tan et. al <sup id="fnref:tan2015unified"><a href="#fn:tan2015unified" class="footnote" rel="footnote" role="doc-noteref">3</a></sup>. A variant is parsimonious if and only if it is represented in as few nucleotides as possible without an allele of length 0. Futher reading here <a href="https://genome.sph.umich.edu/wiki/Vt#Normalization">https://genome.sph.umich.edu/wiki/Vt#Normalization</a><li><strong>[Filter]</strong> 13_pre_annotation_MAF.sh runs data preparation steps including a filter with <code class="language-plaintext highlighter-rouge">vcftools</code> <code class="language-plaintext highlighter-rouge">--max-maf</code> using the source variable <code class="language-plaintext highlighter-rouge">MAF_value="0.4"</code>. This step is simply for reducing the data size to remove highly common variants. We do not expect more than 40% of the cohort to share a relavent variant.</ol><p><strong>Rare disease QV</strong> (run after <strong>QC-only QV</strong>)</p><ol><li>ACMG filtering <a href="acmg_criteria_table_main.md">ACMG criteria and scoring</a><li>Variant interpretation in disease. This final step is used in a case-by-case basis to assess whether automated analysis agrees with the consensus clinical genetics approach.</ol><p>Table 1. VQSR settings</p><div class="table-wrapper"><table><thead><tr><th><strong>VQSR Settings</strong><th><strong>Explanation</strong><tbody><tr><td><strong>SNP Mode</strong><td> <tr><td>- <strong>HapMap:</strong> known=false, training=true, truth=true, prior=15.0<td>Used as a high-confidence reference set for training the recalibration model.<tr><td>- <strong>Omni:</strong> known=false, training=true, truth=false, prior=12.0<td>Provides additional training data derived from Omni genotyping arrays.<tr><td>- <strong>1000G:</strong> known=false, training=true, truth=false, prior=10.0<td>Utilizes data from the 1000 Genomes Project to inform the model on common SNP variations.<tr><td>- <strong>Annotations:</strong> QD, MQ, MQRankSum, ReadPosRankSum, FS, SOR<td>Annotations are metrics used to predict the likelihood of a variant being a true genetic variation versus a sequencing artifact. They include quality by depth, mapping quality, mapping quality rank sum test, read position rank sum test, Fisher’s exact test for strand bias, and symmetric odds ratio of strand bias.<tr><td>- <strong>Truth Sensitivity Filter Level:</strong> 99.7<td>Specifies the percentage of true variants to retain at a given VQSLOD score threshold, set here to capture 99.7% of true variants.<tr><td> <td> <tr><td><strong>INDEL Mode</strong><td> <tr><td>- <strong>Mills:</strong> known=false, training=true, truth=true, prior=12.0<td>Utilizes the Mills and 1000G gold standard indel dataset for high-accuracy recalibration of indels.<tr><td>- <strong>dbSNP:</strong> known=true, training=false, truth=false, prior=2.0<td>Includes known indel sites from the dbSNP database to enhance the detection and filtering process.<tr><td>- <strong>Annotations:</strong> QD, MQRankSum, ReadPosRankSum, FS, SOR<td>Same as for SNPs, these annotations are critical for assessing the likelihood of indels being true genetic variations rather than errors.<tr><td>- <strong>Truth Sensitivity Filter Level:</strong> 95<td>This setting defines the percentage of true indels to retain, aiming to capture 95% of true indels at the specified VQSLOD threshold.</table></div><h2 id="variables-file"> <a href="#variables-file" class="anchor-heading" aria-labelledby="variables-file"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Variables file</h2><p>During analysis we source the following <code class="language-plaintext highlighter-rouge">variables.sh</code> file. We will continue to update this so that all QV variables and thresholds are included here. In future releases, we plan to have each version of the variables file indentified with sensible names like <code class="language-plaintext highlighter-rouge">variables_dna_snvindel_v1</code> etc.</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># This script is meant to be sourced, not executed. To use the variables
# defined in this script, use the command `source variables.sh` in your
# own scripts.

# main paths
HOME="/project/home/lawless"
DATA="${HOME}/data/wgs"
PROD="/project/data/prod"

# processed data
STUDYBOOK_DIR="${DATA}/study_book_data"
FASTP_DIR="${DATA}/fastp"
FASTP_METRICS_DIR="${DATA}/fastp_reports"
BAM_DIR="${DATA}/bam"
SAMSORT_DIR="${DATA}/samsort"
RMDUP_DIR="${DATA}/rmdup"
RMDUP_MERGE_DIR="${DATA}/rmdup_merge"
BQSR_DIR="${DATA}/bqsr"
BQSR_DIR_HOLD="${DATA}/bqsr_hold"
KNOWN_SITES="${HOME}/data/db/known_sites"
HC_DIR="${DATA}/hc"
HC_RENAMED_DIR="${DATA}/hc_renamed"
GENOMICS_DB_IMPORT_DIR="${DATA}/genomics_db_import"
GENOTYPE_GVCF_DIR="${DATA}/genotype_gvcf"
VQSR_DIR="${DATA}/vqsr"
GENOTYPE_REFINEMENT_DIR="${DATA}/genotype_refine"
PRE_ANNOTATION_DIR="${DATA}/pre_annotation"
PRE_ANNOTATION_MAF_DIR="${DATA}/pre_annotation_maf"
ANNOTATION_DIR="${DATA}/annotation"
ANNOTATION_GNOMAD_DIR="${DATA}/annotation_gnomad"
POSTANNOSPLIT="${DATA}/annotation_split"
POSTVCF2TABLE="${DATA}/vcf2table"
VIRTUAL_PANEL_DIR="${DATA}/virtual_panel"
VCFLIB_DIR="${DATA}/vcflib_out"
VARLEVEL_DIR="${DATA}/variant_level"
VAL_DIR="${DATA}/validation"
MAF_value="0.4"
GNOMAD_value="0.002"
DATABASE_DIR="${HOME}/data/db"
dbNSFP_grch38="${DATABASE_DIR}/dbnsfp/dbNSFP4.4a_grch38_v3.gz"
THOUSAND_GENOMES="/project/home/lawless/data/ref/1000genomes/"
PCA_BIPLOT_1KG="${DATA}/pca_biplot_1kg"
SMOOVE="${DATA}/smoove"
VSAT="${DATA}/plink_skat_vsat"
QC_SUMMARY_STATS="${DATA}/qc_summary_stats"

# src
PROJECT="${HOME}/wgs"
SRC="${PROJECT}/src"

# ref
REF="/project/home/lawless/data/ref/bgzip_fasta/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz"
REF_NONZIP="/project/home/lawless/data/ref/bgzip_fasta/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna"
REF_SHARED="/project/data/shared/processed_data/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna"
SV_EXCLUDE_BED="/project/home/lawless/data/ref/bed_annotaions/exclude.cnvnator_100bp.GRCh38.20170403.bed"
VEP_VERSION="95"
VEP_CACHE_DIR="${DATABASE_DIR}/vep${VEP_VERSION}/cache"
VEP_PLUGIN_DIR="${DATABASE_DIR}/vep${VEP_VERSION}/plugins"
</code></pre></div></div><h2 id="the-use-of-qv-in-analysis-plan"> <a href="#the-use-of-qv-in-analysis-plan" class="anchor-heading" aria-labelledby="the-use-of-qv-in-analysis-plan"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> The use of QV in analysis plan</h2><p>QV are a logical necessity of data cleaning and preperation. Labelling then the single umbrella of QV is useful for simplicity. In reality the steps of QV can be sepated and result from a mixture of different steps or sources. <strong>Figure 2</strong> from Povysil et al. <sup id="fnref:povysil2019rare:2"><a href="#fn:povysil2019rare" class="footnote" rel="footnote" role="doc-noteref">1</a></sup> provides an example of the application of QV in an analysis.</p><p><img src="/pages/design_doc/images/povysil2019qv.png.webp" width="100%" /></p><p>Figure 2. From Povysil et. al <sup id="fnref:povysil2019rare:3"><a href="#fn:povysil2019rare" class="footnote" rel="footnote" role="doc-noteref">1</a></sup>: Outline of the standard collapsing analysis approach. First, cases and matching controls are selected (part Aa), and the same sample-level quality control (QC) is performed for cases and controls (part Ab). The sample-pruning level comprises relatedness pruning (part Ba) and the removal of population outliers based on principal components (PCs) (part Bb). All the remaining samples are used to perform coverage harmonization (part C), in which sites and therefore variants that show coverage differences between cases and controls are pruned. All remaining variants are used for qualifying variant (QV) selection (part D), in which various filters, including internal and external minor allele frequency (MAF) filters, are applied. The selected QVs are used to build the gene-by-individual collapsing matrix (part E), which indicates the presence of at least one QV. Finally, each gene is tested for an association between QV status and the phenotype of interest (part Fa), and the results can be evaluated by means of a quantile–quantile (QQ) plot (part Fb).</p><h1 id="references"> <a href="#references" class="anchor-heading" aria-labelledby="references"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> References</h1><div class="footnotes" role="doc-endnotes"><ol><li id="fn:povysil2019rare"><p>Povysil, G. et al., 2019. Rare-variant collapsing analyses for complex traits: guidelines and applications. <em>Nature Reviews Genetics</em>, 20(12), pp.747-759. DOI: <a href="https://doi.org/10.1038/s41576-019-0177-4">10.1038/s41576-019-0177-4</a>. <a href="#fnref:povysil2019rare" class="reversefootnote" role="doc-backlink">&#8617;</a> <a href="#fnref:povysil2019rare:1" class="reversefootnote" role="doc-backlink">&#8617;<sup>2</sup></a> <a href="#fnref:povysil2019rare:2" class="reversefootnote" role="doc-backlink">&#8617;<sup>3</sup></a> <a href="#fnref:povysil2019rare:3" class="reversefootnote" role="doc-backlink">&#8617;<sup>4</sup></a></p><li id="fn:hillman2020identification"><p>Hillman, P., Baker, C., Hebert, L., Brown, M., Hixson, J., Ashley-Koch, A., Morrison, A.C., Northrup, H. and Au, K.S. (2020), Identification of novel candidate risk genes for myelomeningocele within the glucose homeostasis/oxidative stress and folate/one-carbon metabolism networks. Mol Genet Genomic Med, 8: e1495. https://doi.org/10.1002/mgg3.1495 <a href="#fnref:hillman2020identification" class="reversefootnote" role="doc-backlink">&#8617;</a> <a href="#fnref:hillman2020identification:1" class="reversefootnote" role="doc-backlink">&#8617;<sup>2</sup></a></p><li id="fn:tan2015unified"><p>Adrian Tan, Gonçalo R. Abecasis, Hyun Min Kang, Unified representation of genetic variants, Bioinformatics, Volume 31, Issue 13, July 2015, Pages 2202–2204, https://doi.org/10.1093/bioinformatics/btv112 <a href="#fnref:tan2015unified" class="reversefootnote" role="doc-backlink">&#8617;</a></p></ol></div><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
