<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Design release DNA SNV INDEL v1 | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Design release DNA SNV INDEL v1" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/design_dna_snvindel_v1_release.html" /><meta property="og:url" content="http://localhost:4000/pages/design_dna_snvindel_v1_release.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Design release DNA SNV INDEL v1" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Design release DNA SNV INDEL v1","url":"http://localhost:4000/pages/design_dna_snvindel_v1_release.html"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item active"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item active"><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link active">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_evidence_flag.html" class="nav-list-link">Project design QV-ES</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a><li class="nav-list-item "><a href="/pages/design_quant_es.html" class="nav-list-link">Project design Quant ES</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metadata_users.html" class="nav-list-link">WGS metadata users</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/my_voice.html" class="nav-list-link">QV - My voice</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/qv_design.html" class="nav-list-link">QV design principles</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/pages/design_doc.html">Design documents</a><li class="breadcrumb-nav-list-item"><span>Design release DNA SNV INDEL v1</span></ol></nav><div id="main-content" class="main-content" role="main"><p>Last update: 20250102</p><h1 id="design-release-dna-snv-indel-v1"> <a href="#design-release-dna-snv-indel-v1" class="anchor-heading" aria-labelledby="design-release-dna-snv-indel-v1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Design release DNA SNV INDEL v1</h1><p><strong>Protocol name</strong>: <code class="language-plaintext highlighter-rouge">design_dna_snv_indel_v1_relsease</code> (this document) for <code class="language-plaintext highlighter-rouge">design_dna_snvindel_v1</code> (see <a href="design_dna_snvindel_v1.html">design_dna_snvindel_v1</a>).</p><hr /><ul id="markdown-toc"><li><a href="#design-release-dna-snv-indel-v1" id="markdown-toc-design-release-dna-snv-indel-v1">Design release DNA SNV INDEL v1</a><ul><li><a href="#about" id="markdown-toc-about">About</a><li><a href="#contents" id="markdown-toc-contents">Contents</a><li><a href="#data-synchronisation-process" id="markdown-toc-data-synchronisation-process">Data synchronisation process</a><li><a href="#data-availability" id="markdown-toc-data-availability">Data Availability</a><li><a href="#current-version-script" id="markdown-toc-current-version-script">Current version script</a><li><a href="#checksums" id="markdown-toc-checksums">Checksums</a><li><a href="#stats" id="markdown-toc-stats">Stats</a></ul></ul><hr /><blockquote class="highlight-title"><p>Release information</p><p><strong>Current release version</strong>: v1<br /> <strong>Location temp</strong>: <code class="language-plaintext highlighter-rouge">/project/data/shared_all/release_dna_snv_indel_v1</code><br /> <strong>Location stable</strong>: <code class="language-plaintext highlighter-rouge">/project/data/shared/release_dna_snv_indel_v1</code></p></blockquote><blockquote class="new-title"><p>Status</p><p><strong>Status</strong>: <code class="language-plaintext highlighter-rouge">design_dna_snv_indel_v2_relsease</code> is currently <strong>recommended</strong>.<br /> We have added new annotation settings affecting which QV are reported. <br /> Any modifications due to error will be notified and a new release ID will be issued (e.g. <code class="language-plaintext highlighter-rouge">design_dna_snv_indel_v1.2_relsease</code>).</p></blockquote><h2 id="about"> <a href="#about" class="anchor-heading" aria-labelledby="about"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> About</h2><p>If you are reading this then you are interested in using the release data from the <code class="language-plaintext highlighter-rouge">design_dna_snvindel_v1</code> pipeline. This pipeline produces a multi-use dataset via the qualifying variants 1 (QV1) protocol. Incremental additions with new data for this release will be added. Potential uses are suggested as the end-points in figure 1 (e.g. statistical genomics, etc.)</p><p>The protocol used was:</p><ol><li><a href="design_dna_snvindel_v1.html">design_dna_snvindel_v1</a><ul><li><a href="design_PCA_SNV_INDEL_V1.html">design_PCA_SNV_INDEL_v1</a><li><a href="design_qv_snvindel_v1.html">design_qv_snvindel_v1</a></ul></ol><p><img src="/pages/design_doc/images/qv_pipeline_vcurrent.png" width="75%" /></p><p>Figure 1: Summary of design DNA SNV INDEL v1 pipeline plan.</p><h2 id="contents"> <a href="#contents" class="anchor-heading" aria-labelledby="contents"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Contents</h2><p>See the README.md for this release to find files:</p><p><a href="dna_snv_indel_v1_release.README.md">dna_snv_indel_v1_release.README.md</a></p><h2 id="data-synchronisation-process"> <a href="#data-synchronisation-process" class="anchor-heading" aria-labelledby="data-synchronisation-process"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data synchronisation process</h2><ol><li>Sync release data to <code class="language-plaintext highlighter-rouge">shared_all</code>.<li>Log and verify data during sync to <code class="language-plaintext highlighter-rouge">shared_all</code>.<li>Data manager syncs from <code class="language-plaintext highlighter-rouge">shared_all</code> to <code class="language-plaintext highlighter-rouge">shared</code>.<li>The final release is locked in <code class="language-plaintext highlighter-rouge">shared</code> but allows incremental additions.<li>Labelled as <code class="language-plaintext highlighter-rouge">release_dna_snv_indel_v1</code>.</ol><h2 id="data-availability"> <a href="#data-availability" class="anchor-heading" aria-labelledby="data-availability"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data Availability</h2><p>All raw and processed data are accessible to all internal researchers. Processed/intermediate data might change during projects, so released data is recommended for downstream projects. Release data should have detailed supporting documentation following the protocol. Essential data is curated to minimise storage use. Non-release data is accessible as usual from user directories but may be overwritten or updated without notice. Data paths are listed in the release script, generally read from a master list of variables.</p><h2 id="current-version-script"> <a href="#current-version-script" class="anchor-heading" aria-labelledby="current-version-script"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Current version script</h2><p><code class="language-plaintext highlighter-rouge">release_dna_snv_indel_v1.sh</code> is run from the user src directory.</p><h2 id="checksums"> <a href="#checksums" class="anchor-heading" aria-labelledby="checksums"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Checksums</h2><p>All files are logged with their md5 checksum. To verify integrity run: <code class="language-plaintext highlighter-rouge">md5sum -c checksums.md5</code></p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ cat checksums.md5
457dc93749669bed628575dfe551bdac  ./release_dna_snv_indel_v1.sh.log
d35670b5ac26302b19f07c4cf3a96977  ./release_dna_snv_indel_v1.out
</code></pre></div></div><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ md5sum -c checksums.md5
./release_dna_snv_indel_v1.sh.log: OK
./release_dna_snv_indel_v1.out: OK
</code></pre></div></div><h2 id="stats"> <a href="#stats" class="anchor-heading" aria-labelledby="stats"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Stats</h2><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Chromosome 1
Number of samples: 180
Number of SNPs:    1543167
Number of INDELs:  385213
Number of MNPs:    0
Number of others:  0
Number of sites:   1972696

Chromosome 2
Number of samples: 180
Number of SNPs:    1620394
Number of INDELs:  394968
Number of MNPs:    0
Number of others:  0
Number of sites:   2061648

Chromosome 3
Number of samples: 180
Number of SNPs:    1316780
Number of INDELs:  324975
Number of MNPs:    0
Number of others:  0
Number of sites:   1673128

Chromosome 4
Number of samples: 180
Number of SNPs:    1337064
Number of INDELs:  316959
Number of MNPs:    0
Number of others:  0
Number of sites:   1689336

Chromosome 5
Number of samples: 180
Number of SNPs:    1196261
Number of INDELs:  293793
Number of MNPs:    0
Number of others:  0
Number of sites:   1520849

Chromosome 6
Number of samples: 180
Number of SNPs:    1214390
Number of INDELs:  303459
Number of MNPs:    0
Number of others:  0
Number of sites:   1552036

Chromosome 7
Number of samples: 180
Number of SNPs:    1145244
Number of INDELs:  277881
Number of MNPs:    0
Number of others:  0
Number of sites:   1460406

Chromosome 8
Number of samples: 180
Number of SNPs:    1020218
Number of INDELs:  240781
Number of MNPs:    0
Number of others:  0
Number of sites:   1284467

Chromosome 9
Number of samples: 180
Number of SNPs:    858094
Number of INDELs:  198839
Number of MNPs:    0
Number of others:  0
Number of sites:   1080126

Chromosome 10
Number of samples: 180
Number of SNPs:    946539
Number of INDELs:  237079
Number of MNPs:    0
Number of others:  0
Number of sites:   1212448

Chromosome 11
Number of samples: 180
Number of SNPs:    910215
Number of INDELs:  221252
Number of MNPs:    0
Number of others:  0
Number of sites:   1154276

Chromosome 12
Number of samples: 180
Number of SNPs:    903797
Number of INDELs:  242211
Number of MNPs:    0
Number of others:  0
Number of sites:   1173553

Chromosome 13
Number of samples: 180
Number of SNPs:    683880
Number of INDELs:  169236
Number of MNPs:    0
Number of others:  0
Number of sites:   872669

Chromosome 14
Number of samples: 180
Number of SNPs:    618962
Number of INDELs:  156463
Number of MNPs:    0
Number of others:  0
Number of sites:   792576

Chromosome 15
Number of samples: 180
Number of SNPs:    609207
Number of INDELs:  141670
Number of MNPs:    0
Number of others:  0
Number of sites:   768672

Chromosome 16
Number of samples: 180
Number of SNPs:    653752
Number of INDELs:  151086
Number of MNPs:    0
Number of others:  0
Number of sites:   826293

Chromosome 17
Number of samples: 180
Number of SNPs:    521864
Number of INDELs:  161194
Number of MNPs:    0
Number of others:  0
Number of sites:   700410

Chromosome 18
Number of samples: 180
Number of SNPs:    534571
Number of INDELs:  130405
Number of MNPs:    0
Number of others:  0
Number of sites:   680157

Chromosome 19
Number of samples: 180
Number of SNPs:    455618
Number of INDELs:  136628
Number of MNPs:    0
Number of others:  0
Number of sites:   610238

Chromosome 20
Number of samples: 180
Number of SNPs:    427477
Number of INDELs:  108222
Number of MNPs:    0
Number of others:  0
Number of sites:   547588

Chromosome 21
Number of samples: 180
Number of SNPs:    333850
Number of INDELs:  67464
Number of MNPs:    0
Number of others:  0
Number of sites:   413496

Chromosome 22
Number of samples: 180
Number of SNPs:    332138
Number of INDELs:  74089
Number of MNPs:    0
Number of others:  0
Number of sites:   418913

Chromosome X
Number of samples: 180
Number of SNPs:    678392
Number of INDELs:  183618
Number of MNPs:    0
Number of others:  0
Number of sites:   881808

Chromosome Y
Number of samples: 180
Number of SNPs:    71640
Number of INDELs:  1289
Number of MNPs:    0
Number of others:  0
Number of sites:   77666
</code></pre></div></div><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
