<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Design DNA SNV INDEL v1 | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Design DNA SNV INDEL v1" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/design_dna_snvindel_v1.html" /><meta property="og:url" content="http://localhost:4000/pages/design_dna_snvindel_v1.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Design DNA SNV INDEL v1" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Design DNA SNV INDEL v1","url":"http://localhost:4000/pages/design_dna_snvindel_v1.html"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item active"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item active"><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link active">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/storage.html" class="nav-list-link">storage</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><nav aria-label="Breadcrumb" class="breadcrumb-nav"><ol class="breadcrumb-nav-list"><li class="breadcrumb-nav-list-item"><a href="/pages/design_doc.html">Design documents</a><li class="breadcrumb-nav-list-item"><span>Design DNA SNV INDEL v1</span></ol></nav><div id="main-content" class="main-content" role="main"><p>Last update: 20241217</p><h1 id="design-dna-snv-indel-v1"> <a href="#design-dna-snv-indel-v1" class="anchor-heading" aria-labelledby="design-dna-snv-indel-v1"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Design DNA SNV INDEL v1</h1><h2 id="germline-short-variant-discovery-snvs--indels-and-interpretation"> <a href="#germline-short-variant-discovery-snvs--indels-and-interpretation" class="anchor-heading" aria-labelledby="germline-short-variant-discovery-snvs--indels-and-interpretation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Germline short variant discovery (SNVs + Indels) and interpretation</h2><hr /><ul id="markdown-toc"><li><a href="#design-dna-snv-indel-v1" id="markdown-toc-design-dna-snv-indel-v1">Design DNA SNV INDEL v1</a><ul><li><a href="#germline-short-variant-discovery-snvs--indels-and-interpretation" id="markdown-toc-germline-short-variant-discovery-snvs--indels-and-interpretation">Germline short variant discovery (SNVs + Indels) and interpretation</a><li><a href="#aims" id="markdown-toc-aims">Aims</a><li><a href="#introduction" id="markdown-toc-introduction">Introduction</a><li><a href="#protocol-summary" id="markdown-toc-protocol-summary">Protocol summary</a><li><a href="#protocol-steps" id="markdown-toc-protocol-steps">Protocol steps</a><li><a href="#metrics" id="markdown-toc-metrics">Metrics</a><li><a href="#data-release" id="markdown-toc-data-release">Data release</a></ul></ul><hr /><p><strong>Protocol name</strong>: <code class="language-plaintext highlighter-rouge">design_dna_snvindel_v1</code></p><h2 id="aims"> <a href="#aims" class="anchor-heading" aria-labelledby="aims"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Aims</h2><div class="table-wrapper"><table><thead><tr><th style="text-align: center">Phase<th style="text-align: center">Aim<th>Status<th>Task<tbody><tr><td style="text-align: center">Phase 2<td style="text-align: center">(1)<td><strong>Complete</strong><td>Process all WGS from the study cohort to a consensus format.<tr><td style="text-align: center">Phase 2<td style="text-align: center">(2)<td><strong>2 of 2 complete</strong><td>Prepare qualifying variant (QV) sets <br />for each downstream aim.<tr><td style="text-align: center">Phase 2<td style="text-align: center">(3)<td><strong>v1 complete</strong> <br /> v2 in progress<td>Clinical genetics report per individual<br />(i.e., baseline benchmark of <br />known disease-causing).<tr><td style="text-align: center">Phase 2<td style="text-align: center">(4)<td><strong>2 experiment complete</strong><td>GWAS: Statistical genomics to find new <br />cohort-level associations with <br />disease.<tr><td style="text-align: center">Phase 2<td style="text-align: center">(5)<td><strong>2 experiment complete</strong><td>Gene-VSAT: Statistical genomics to find new <br />cohort-level associations with <br />disease.<tr><td style="text-align: center">Phase 2<td style="text-align: center">(6)<td><strong>1 experiment complete</strong><td>Proteome-VSAT: Statistical genomics to find new <br />cohort-level associations with <br />disease (Proteom-VSAT).<tr><td style="text-align: center">Phase 2<td style="text-align: center">(7)<td><strong>1 experiment complete</strong><td>ACAT: Statistical multiomics to find new <br />cohort-level associations with <br />disease..<tr><td style="text-align: center">Phase 2<td style="text-align: center">(8)<td>In progress<td>New methods (ML/DL, causal inference) <br />for individual and cohort-level<br />discovery.</table></div><p><img src="/pages/design_doc/images/qv_pipeline_vcurrent.png" width="75%" /></p><p>Figure 1: Summary of design DNA SNV INDEL v1 pipeline plan.</p><h2 id="introduction"> <a href="#introduction" class="anchor-heading" aria-labelledby="introduction"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Introduction</h2><p>This protocol is designed to process DNA WGS data in FASTQ format into qualifying qariants (QV) based on consensus variables and thresholds (figure 1). The QV can then be used in multiple applications such as ML/DL to find disease-related variants or gene functions. Additionally, in the clinical genetic protocol further standardised filtering criteria are used to reach a single genetic determinant in a clinical genetics report for each subject. The design name <code class="language-plaintext highlighter-rouge">Design DNA SNV INDEL v1</code> indicates that this protocol is tailored to single nucleotide variants (SNVs) and short insertion/deletions (INDELs) (e.g. GATK pipeline). We implement the genome analysis tool kit <a href="https://gatk.broadinstitute.org/hc/en-us">GATK</a> best practices workflow for <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels">germline short variant discovery</a> (open source licence <a href="https://github.com/broadinstitute/gatk/blob/master/LICENSE.TXT">here</a>). This GATK workflow is designed to operate on a set of samples constituting a study cohort; specifically, a set of per-sample BAM files that have been pre-processed as described in the GATK Best Practices for data pre-processing. Single-variant and genomics-only analysis will be followed up to confirm if causal effects are identified in RNA and protein layers. Joint-multiomic analysis will include all layers in a single statistical model.</p><h2 id="protocol-summary"> <a href="#protocol-summary" class="anchor-heading" aria-labelledby="protocol-summary"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Protocol summary</h2><ol><li>Process all raw WGS into an analysis-ready format - geonmic VCF (gVCF).<ul><li>The first goal is to process all raw whole genome sequencing (WGS) data into analysis-ready formats, specifically into joint cohort Variant Call Format (VCF) using the emit-ref-confidence (ERC) gVCF mode. This involves using a reference model to emit data with condensed non-variant blocks, adhering to the gVCF format. gVCF is split per chromosome.</ul><li>The joint cohort chromosome level gVCF are filtered into qualifying variants (QV).<li>The QV sets are used individually or mixed to produce the main analysis results:<ul><li>QV set 1 for clinical genetics (known disease-causing) report for each individual<li>QV set 2 for statistical genomics (new associations with established methods) for cohort level discovery<li>QV set 1 or 2 for other methods (ML/DL, causal inference) (new methods) for individual and cohort level discovery</ul><li>Release data.<li>If not already included in an analysis model, the candidate causal variants will be followed up to confirm if causal effects are identified in RNA and protein layers.</ol><h2 id="protocol-steps"> <a href="#protocol-steps" class="anchor-heading" aria-labelledby="protocol-steps"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Protocol steps</h2><p>The major processing steps in sequential order are:</p><ul><li><strong>Aim 1</strong><ul><li>FASTQ QC - <a href="dna_qc.html">see DNA QC</a><li><a href="fastp.html">FASTP</a>: QC, check adapters, trimming, filtering, splitting/merging.<li>Genome alignment with <a href="bwa.html">BWA</a> with <a href="ref.html">reference genome</a> GCA_000001405.15_GRCh38_no_alt_analysis_set<li><a href="gatk_duplicates.html">GATK Duplicates</a><li><a href="gatk_bsqr.html">GATK BQSR</a><li><a href="gatk_hc.html">GATK Haplotype caller</a><li><a href="gatk_dbimport.html">GATK Genomic db import</a><li><a href="gatk_genotypegvcf.html">GATK Genotyping gVCFs</a><li><a href="gatk_vqsr.html">GATK VQSR</a><li><a href="gatk_genotyperefine.html">GATK Genotype refine</a><li>VCF QC - <a href="dna_qc.html">see DNA QC</a></ul><li><strong>Aim 2</strong><ul><li><a href="design_qv_snvindel_v1.html">Design QV SNV INDEL V1</a> Qualifying variants (variables and thresholds) SNV INDEL v1<li><a href="pre_annoprocess.html">Pre-annotation processing</a>: data conversion for simpler handling<li><a href="pre_anno_maf.html">Pre-annotation MAF</a>: filtering to remove noise<li><a href="dna_annotation.html">DNA annotation</a>: annotate known effects, biological function, associations</ul><li><strong>Aim 3 use a selective mixture of the remaining methods depending on the QV set</strong><ul><li><a href="dna_interpretation.html">DNA interpretation</a><li><a href="acmg_criteria_table_main.html">ACMG criteria</a>: Standardised scoring for interpreting variant pathogenicity<li>Clinical genetics reports - not documented here</ul><li><strong>Aim 4-7 use a selective mixture of the remaining methods depending on the QV set</strong><ul><li><a href="design_statistical_genomics_v1.html">Design Statistical genomics v1</a><li>ML/DL projects - not documented here</ul><li><strong>Release</strong><ul><li>See <a href="design_dna_snvindel_v1_release.html">design_dna_snvindel_v1_release</a></ul></ul><p><img src="/pages/design_doc/images/variant_annotation_graph.png" width="75%" /></p><p>Figure 2: Extended methods of figure 1 DNA germline short variant discovery pipeline plan.</p><h2 id="metrics"> <a href="#metrics" class="anchor-heading" aria-labelledby="metrics"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Metrics</h2><p>Study book data:</p><ol><li><code class="language-plaintext highlighter-rouge">CollectWgsMetrics</code>: <code class="language-plaintext highlighter-rouge">03b_collectwgsmetrics.sh</code> -&gt; <code class="language-plaintext highlighter-rouge">study_book/qc_summary_stats</code> mapping, depth, and more. See <a href="metrics_collectwgsmetrics.html">metrics_collectwgsmetrics</a>.<li><code class="language-plaintext highlighter-rouge">bcftools stats</code> and <code class="language-plaintext highlighter-rouge">plot-vcfstats</code>: <code class="language-plaintext highlighter-rouge">07c_qc_summary_stats.sh</code> -&gt; <code class="language-plaintext highlighter-rouge">study_book/qc_summary_stats</code> gVCF summary after HC. See <a href="metrics_bcftoolsstats.html">metrics_bcftoolsstats</a>.</ol><h2 id="data-release"> <a href="#data-release" class="anchor-heading" aria-labelledby="data-release"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data release</h2><p>The private internal data release: <a href="design_dna_snvindel_v1_release.html">design_dna_snvindel_v1_release</a></p><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
