<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Exomiser | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Exomiser" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/exomiser.html" /><meta property="og:url" content="http://localhost:4000/pages/exomiser.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Exomiser" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Exomiser","url":"http://localhost:4000/pages/exomiser.html"}</script> <script type="text/javascript" defer src="/assets/js/mathjax-script-type.js"> </script> <script type="text/javascript" id="MathJax-script" defer src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"> </script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item active"><a href="/pages/exomiser.html" class="nav-list-link active">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><p>Last update: 20250809</p><h1 id="exomiser-scoring-and-ranking"> <a href="#exomiser-scoring-and-ranking" class="anchor-heading" aria-labelledby="exomiser-scoring-and-ranking"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Exomiser Scoring and Ranking</h1><ul id="markdown-toc"><li><a href="#exomiser-scoring-and-ranking" id="markdown-toc-exomiser-scoring-and-ranking">Exomiser Scoring and Ranking</a><ul><li><a href="#1-overview" id="markdown-toc-1-overview">1. Overview</a><li><a href="#2-algorithm" id="markdown-toc-2-algorithm">2. Algorithm</a><ul><li><a href="#21-notation" id="markdown-toc-21-notation">2.1 Notation</a><li><a href="#22-filtering-no-equations" id="markdown-toc-22-filtering-no-equations">2.2 Filtering (no equations)</a><li><a href="#23-variant-score" id="markdown-toc-23-variant-score">2.3 Variant score</a><li><a href="#24-phenotype-score" id="markdown-toc-24-phenotype-score">2.4 Phenotype score</a><li><a href="#25-gene-level-aggregation" id="markdown-toc-25-gene-level-aggregation">2.5 Gene-level aggregation</a><li><a href="#26-final-exomiser-score" id="markdown-toc-26-final-exomiser-score">2.6 Final Exomiser score</a><li><a href="#27-ranking" id="markdown-toc-27-ranking">2.7 Ranking</a></ul><li><a href="#3-summary" id="markdown-toc-3-summary">3. Summary</a><li><a href="#4-references" id="markdown-toc-4-references">4. References</a><li><a href="#appendix-yaml-to-equation-mapping" id="markdown-toc-appendix-yaml-to-equation-mapping">Appendix. YAML to equation mapping</a></ul></ul><hr /><p>See <a href="exomiser_phenodigm">exomiser phenodigm</a> for the phenotype similarity algorithm in context.</p><h2 id="1-overview"> <a href="#1-overview" class="anchor-heading" aria-labelledby="1-overview"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1. Overview</h2><p>Exomiser assigns each <strong>surviving variant</strong> a <strong>variant score</strong>. Those that passes pre-scoring filters for genotype–mode of inheritance compatibility, quality, population frequency, and consequence. Each gene containing one or more surviving variants also receives a <strong>phenotype score</strong>. The highest variant score per gene and its phenotype score are then combined using a logistic regression model, pre-trained by the authors on 10 000 pathogenic and 10 000 benign variants with 10-fold cross-validation. The model coefficients are fixed in Exomiser and are applied directly to the user’s data to produce the final <strong>Exomiser score</strong> in the range \([0,1]\), which determines the ranking of candidate variants.</p><hr /><h2 id="2-algorithm"> <a href="#2-algorithm" class="anchor-heading" aria-labelledby="2-algorithm"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2. Algorithm</h2><h3 id="21-notation"> <a href="#21-notation" class="anchor-heading" aria-labelledby="21-notation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.1 Notation</h3><p>Let:</p><ul><li>\(f\) = maximum MAF across selected frequency datasets (gnomAD, TOPMed, UK10K, 1000 Genomes, ExAC, ESP, plus any in-house set)<li>\(V\) = variant-level score<li>\(P\) = phenotype score for a gene<li>\(S\) = final Exomiser score for a gene</ul><p>All scores lie in \([0,1]\).</p><hr /><h3 id="22-filtering-no-equations"> <a href="#22-filtering-no-equations" class="anchor-heading" aria-labelledby="22-filtering-no-equations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.2 Filtering (no equations)</h3><p>A variant is “surviving” if it passes:</p><ul><li>MOI compatibility for at least one selected inheritance mode (AD, AR, XD, XR, MT)<li>Quality thresholds<li>Population allele frequency limits<li>Allowed predicted consequence types</ul><hr /><h3 id="23-variant-score"> <a href="#23-variant-score" class="anchor-heading" aria-labelledby="23-variant-score"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.3 Variant score</h3><p><strong>Frequency score</strong> (<strong>reported</strong> in Cipriani et al.):</p>\[\text{freqScore}(f) = \begin{cases} 1, &amp; f=0 \\[4pt] 1.13533 - 0.13533\, e^{100f}, &amp; 0 &lt; f \le 0.02 \\[4pt] 0, &amp; f &gt; 0.02 \end{cases}\]<blockquote><p>The constants \(1.13533\) and \(0.13533\) in the frequency score equation are fixed values calibrated once by the Exomiser authors using their reference dataset. They define the shape of the exponential decay from rare to common variants and are not recalculated for user runs. The only run-specific input is \(f\), the maximum MAF across the frequency datasets selected for that analysis, which is plugged into the fixed equation to yield the frequency score.</p></blockquote><p><strong>Pathogenicity score</strong> (<strong>reported</strong>):<br /> Let \(\text{normPred}_k \in [0,1]\) be the normalised score from predictor \(k\) (PolyPhen-2, SIFT, MutationTaster, CADD, REVEL, M-CAP, MPC, MVP, PrimateAI, etc.).<br /> High-impact consequences (frameshift, nonsense, canonical splice acceptor/donor, start-loss, stop-loss) are fixed at \(1.0\), splice-region at \(0.8\).<br /> If no predictor data exist, use a preset by consequence.</p>\[\text{pathScore} = \max_k \,\text{normPred}_k\]<p><strong>Variant score</strong> (<strong>reported</strong>):</p>\[V = \text{freqScore}(f) \cdot \text{pathScore}\]<p><strong>Compound heterozygote averaging for AR</strong> (<strong>reported</strong>):<br /> If \(V_1, V_2\) are the two alleles:</p>\[V_{\text{AR}} = \frac{V_1 + V_2}{2}\]<blockquote><p>The frequency score equation is applied identically to homozygous and heterozygous variants. In autosomal recessive mode, Exomiser averages the two variant scores for a compound heterozygote. If trio data are available, it uses inheritance patterns to confirm the variants are in trans; otherwise, it assumes potential compound heterozygosity when a gene contains two or more rare, surviving variants, which can produce false positives in unphased singleton data.</p></blockquote><hr /><h3 id="24-phenotype-score"> <a href="#24-phenotype-score" class="anchor-heading" aria-labelledby="24-phenotype-score"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.4 Phenotype score</h3><p>Computed per gene using <a href="exomiser_phenodigm">PhenoDigm</a> semantic similarity (<strong>reported</strong>).<br /> The patient’s HPO terms are compared against annotations from:</p><ul><li>Human diseases (OMIM, Orphanet)<li>Mouse models (MGI, IMPC)<li>Zebrafish models (ZFIN)<li>Protein–protein interaction neighbours (STRING)</ul><p>Because PhenoDigm uses ontology expansion, each HPO term is expanded to include all parent terms.</p><p>Let \(P \in [0,1]\) be the best phenotype match for the gene from any source.</p><p><strong>MOI consistency penalty</strong> (<strong>reported</strong>):</p>\[P' = \begin{cases} \tfrac{1}{2}P, &amp; \text{OMIM MOI conflict and omimPrioritiser enabled} \\ P, &amp; \text{otherwise} \end{cases}\]<hr /><h3 id="25-gene-level-aggregation"> <a href="#25-gene-level-aggregation" class="anchor-heading" aria-labelledby="25-gene-level-aggregation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.5 Gene-level aggregation</h3><p>For gene \(g\) with surviving variants \(\{v\}\) and their scores \(V_v\):</p><p>\(V_g^\ast = \max_{v \in g} V_v\) (<strong>reported</strong>)</p><hr /><h3 id="26-final-exomiser-score"> <a href="#26-final-exomiser-score" class="anchor-heading" aria-labelledby="26-final-exomiser-score"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.6 Final Exomiser score</h3><p><strong>Logistic combination</strong>: form <strong>inferred</strong>, training setup <strong>reported</strong>. Model trained once on reference data; coefficients \(\beta_0,\beta_1,\beta_2\) are fixed per Exomiser release.</p>\[S_g = \sigma\!\big( \beta_0 + \beta_1 V_g^\ast + \beta_2 P'_g \big) \qquad\text{with}\qquad \sigma(x) = \frac{1}{1 + e^{-x}}\]<p>This step combines the variant score \(V_g^\ast\) and phenotype score \(P'_g\) into a single probability-like value using a logistic regression model. The logistic function \(\sigma(x)\) maps any real value to the range \(0 \le S_g \le 1\). The coefficients \(\beta_0, \beta_1, \beta_2\) set the relative weight of the two scores.</p><p>\(\beta_0\) is the intercept, \(\beta_1\) is the weight for the variant score \(V_g^\ast\), and \(\beta_2\) is the weight for the phenotype score \(P'_g\).</p><p>The logistic regression uses these three parameters together to balance baseline probability with the contributions of variant and phenotype evidence.</p><p>The model was trained once on 10 000 pathogenic and 10 000 benign variants with 10-fold cross-validation. The resulting coefficients are fixed in each release and applied directly to user data. Variants not in the training set are scored in the same way as those that were, so novel variants are fully supported.</p><hr /><h3 id="27-ranking"> <a href="#27-ranking" class="anchor-heading" aria-labelledby="27-ranking"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.7 Ranking</h3><p>Genes are ranked in descending \(S_g\) (<strong>reported</strong>).</p><hr /><h2 id="3-summary"> <a href="#3-summary" class="anchor-heading" aria-labelledby="3-summary"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 3. Summary</h2><p>Exomiser integrates variant rarity, predicted functional impact, ontology-aware phenotype–gene similarity across multiple species and sources, and MOI compatibility into a single ranking metric. Even minimal phenotype input (e.g. one HPO term) can influence ranking due to ontology expansion, though richer descriptions yield stronger matches.</p><hr /><h2 id="4-references"> <a href="#4-references" class="anchor-heading" aria-labelledby="4-references"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 4. References</h2><blockquote><p>An Improved Phenotype-Driven Tool for Rare Mendelian Variant Prioritization: Benchmarking Exomiser on Real Patient Whole-Exome Data. Cipriani V, Pontikos N, Arno G, Sergouniotis PI, Lenassi E, Thawong P, Danis D, Michaelides M, Webster AR, Moore AT, Robinson PN, Jacobsen JOB, Smedley D. Genes (Basel). 2020 Apr 23;11(4):460<br /> PMID:32340307 DOI:<a href="https://doi.org/10.3390/genes11040460">https://doi.org/10.3390/genes11040460</a></p></blockquote><blockquote><p>100,000 Genomes Pilot on Rare-Disease Diagnosis in Health Care – Preliminary Report. 100,000 Genomes Project Pilot Investigators; Smedley D, … Caulfield M. N Engl J Med. 2021 Nov 11;385(20):1868-1880. PMID:34758253 DOI:<a href="https://doi.org/10.1056/NEJMoa2035790">https://doi.org/10.1056/NEJMoa2035790</a></p></blockquote><p>See <a href="exomiser_phenodigm">exomiser phenodigm</a> for the phenotype similarity algorithm in context.</p><blockquote><p>Damian Smedley, Anika Oellrich, Sebastian Köhler, Barbara Ruef, Sanger Mouse Genetics Project, Monte Westerfield, Peter Robinson, Suzanna Lewis, Christopher Mungall, PhenoDigm: analyzing curated annotations to associate animal models with human diseases, Database, Volume 2013, 2013, bat025, <a href="https://doi.org/10.1093/database/bat025">https://doi.org/10.1093/database/bat025</a></p></blockquote><h2 id="appendix-yaml-to-equation-mapping"> <a href="#appendix-yaml-to-equation-mapping" class="anchor-heading" aria-labelledby="appendix-yaml-to-equation-mapping"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Appendix. YAML to equation mapping</h2><p>Example config:</p><div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">inheritanceModes</span><span class="pi">:</span> <span class="pi">[</span><span class="nv">AUTOSOMAL_DOMINANT</span><span class="pi">,</span> <span class="nv">AUTOSOMAL_RECESSIVE</span><span class="pi">,</span> <span class="nv">X_LINKED_RECESSIVE</span><span class="pi">,</span> <span class="nv">X_LINKED_DOMINANT</span><span class="pi">,</span> <span class="nv">MITOCHONDRIAL</span><span class="pi">]</span>

<span class="na">frequencySources</span><span class="pi">:</span> <span class="pi">[</span><span class="nv">GNOMAD</span><span class="pi">,</span> <span class="nv">TOPMED</span><span class="pi">,</span> <span class="nv">UK10K</span><span class="pi">,</span> <span class="nv">THOUSAND_GENOMES</span><span class="pi">,</span> <span class="nv">EXAC</span><span class="pi">,</span> <span class="nv">ESP</span><span class="pi">,</span> <span class="nv">LOCAL</span><span class="pi">]</span>
<span class="na">frequencyFilter</span><span class="pi">:</span>
  <span class="na">maxFreq</span><span class="pi">:</span>
    <span class="na">AD</span><span class="pi">:</span> <span class="m">0.001</span>
    <span class="na">AR</span><span class="pi">:</span> <span class="m">0.01</span>
    <span class="na">XR</span><span class="pi">:</span> <span class="m">0.01</span>
    <span class="na">XD</span><span class="pi">:</span> <span class="m">0.001</span>
    <span class="na">MT</span><span class="pi">:</span> <span class="m">0.01</span>

<span class="na">pathogenicitySources</span><span class="pi">:</span> <span class="pi">[</span><span class="nv">CADD</span><span class="pi">,</span> <span class="nv">REVEL</span><span class="pi">,</span> <span class="nv">SIFT</span><span class="pi">,</span> <span class="nv">POLYPHEN</span><span class="pi">,</span> <span class="nv">MUTATION_TASTER</span><span class="pi">,</span> <span class="nv">MVP</span><span class="pi">,</span> <span class="nv">PRIMATE_AI</span><span class="pi">,</span> <span class="nv">M_CAP</span><span class="pi">,</span> <span class="nv">MPC</span><span class="pi">]</span>

<span class="na">hiPhivePrioritiser</span><span class="pi">:</span> <span class="pi">{}</span>
<span class="na">omimPrioritiser</span><span class="pi">:</span> <span class="kc">true</span>

<span class="na">qualityFilter</span><span class="pi">:</span> <span class="m">20</span>
<span class="na">variantEffectFilter</span><span class="pi">:</span>
  <span class="na">keep</span><span class="pi">:</span>
    <span class="pi">-</span> <span class="s">frameshift_variant</span>
    <span class="pi">-</span> <span class="s">stop_gained</span>
    <span class="pi">-</span> <span class="s">splice_acceptor_variant</span>
    <span class="pi">-</span> <span class="s">splice_donor_variant</span>
    <span class="pi">-</span> <span class="s">start_lost</span>
    <span class="pi">-</span> <span class="s">stop_lost</span>
    <span class="pi">-</span> <span class="s">missense_variant</span>
    <span class="pi">-</span> <span class="s">splice_region_variant</span>

<span class="na">whitelist</span><span class="pi">:</span> <span class="s">clinvar_path_lpath.tsv.gz</span>
</code></pre></div></div><div class="table-wrapper"><table><thead><tr><th>YAML key<th>Symbol in text<th>How it is used<tbody><tr><td><code class="language-plaintext highlighter-rouge">frequencySources</code><td>\(f\)<td>\(f = \max_{\text{sources}} \mathrm{MAF}\) for the variant in the selected datasets. Feeds \(\text{freqScore}(f)\).<tr><td><code class="language-plaintext highlighter-rouge">frequencyFilter.maxFreq.*</code><td>survival gate on \(f\)<td>Pre-filter by MOI. If variant fails, it is not scored. Also aligns with piecewise caps in \(\text{freqScore}(f)\).<tr><td><code class="language-plaintext highlighter-rouge">pathogenicitySources</code><td>\(\text{normPred}_k\)<td>Predictors whose normalised scores enter \(\text{pathScore} = \max_k \text{normPred}_k\).<tr><td><code class="language-plaintext highlighter-rouge">variantEffectFilter.keep</code><td>consequence policy<td>Determines which consequences survive filtering. High-impact consequences map to fixed scores inside \(\text{pathScore}\): frameshift, nonsense, canonical splice, start-loss, stop-loss set to \(1.0\); splice-region set to \(0.8\).<tr><td><code class="language-plaintext highlighter-rouge">qualityFilter</code><td>survival gate<td>Variants must meet quality to be considered surviving before any scoring.<tr><td><code class="language-plaintext highlighter-rouge">inheritanceModes</code><td>MOI filter and AR combine<td>Controls survival and whether compound heterozygote averaging \(V_{\mathrm{AR}} = \frac{V_1 + V_2}{2}\) applies.<tr><td><code class="language-plaintext highlighter-rouge">hiPhivePrioritiser</code><td>\(P\)<td>Enables PhenoDigm to compute the phenotype score \(P \in [0,1]\) for each gene.<tr><td><code class="language-plaintext highlighter-rouge">omimPrioritiser</code><td>\(P'\)<td>When true and OMIM MOI conflicts, apply \(P' = \frac{1}{2} P\), else $$P’ = P$.<tr><td><code class="language-plaintext highlighter-rouge">whitelist</code><td>survival bypass<td>Listed variants bypass frequency and consequence filters and thus can contribute to $$V_g^\ast$.</table></div><p>Referenced equations:</p>\[\text{freqScore}(f) = \begin{cases} 1, &amp; f=0\\ 1.13533 - 0.13533\, e^{100f}, &amp; 0 &lt; f \le 0.02\\ 0, &amp; f&gt;0.02 \end{cases}\] \[\text{pathScore}=\max_k \text{normPred}_k \qquad V=\text{freqScore}(f)\cdot \text{pathScore}\] \[V_{\text{AR}}=\frac{V_1+V_2}{2} \qquad V_g^\ast=\max_{v\in g} V_v \qquad P'=\begin{cases} \tfrac{1}{2}P,&amp; \text{OMIM MOI conflict}\\ P,&amp; \text{otherwise} \end{cases}\] \[S_g=\sigma\!\big(\beta_0+\beta_1 V_g^\ast+\beta_2 P'_g\big) \qquad \sigma(x)=\frac{1}{1+e^{-x}}\]<p>Provenance: frequency formula, high-impact fixed scores, AR averaging, MOI penalty, use of a fixed pre-trained logistic model are <strong>reported</strong>. The explicit logistic form with symbolic coefficients is <strong>inferred</strong>.</p><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
