<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Inference of causal metabolite networks | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Inference of causal metabolite networks" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/inf_causal_metab_sem.html" /><meta property="og:url" content="http://localhost:4000/pages/inf_causal_metab_sem.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Inference of causal metabolite networks" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Inference of causal metabolite networks","url":"http://localhost:4000/pages/inf_causal_metab_sem.html"}</script> <script type="text/javascript" defer src="/assets/js/mathjax-script-type.js"> </script> <script type="text/javascript" id="MathJax-script" defer src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"> </script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_evidence_flag.html" class="nav-list-link">Project design QV-ES</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a><li class="nav-list-item "><a href="/pages/design_quant_es.html" class="nav-list-link">Project design Quant ES</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item active"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link active">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/journal_ranking.html" class="nav-list-link">Journal ranking</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metadata_users.html" class="nav-list-link">WGS metadata users</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/my_voice.html" class="nav-list-link">QV - My voice</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/qv_design.html" class="nav-list-link">QV design principles</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><p class="no_toc">Last update: 20230813</p><details open=""> <summary class="text-delta">Table of contents</summary><ul id="markdown-toc"><li><a href="#inference-of-causal-metabolite-networks" id="markdown-toc-inference-of-causal-metabolite-networks">Inference of causal metabolite networks</a><ul><li><a href="#abstract-summary" id="markdown-toc-abstract-summary">Abstract summary</a><li><a href="#key-methods-summary" id="markdown-toc-key-methods-summary">Key methods summary</a><ul><li><a href="#21-one-sample-data" id="markdown-toc-21-one-sample-data">2.1 One-sample data</a><li><a href="#22-sem-with-individual-level-data" id="markdown-toc-22-sem-with-individual-level-data">2.2 SEM with individual-level data</a><li><a href="#23-sem-with-some-invalid-ivs" id="markdown-toc-23-sem-with-some-invalid-ivs">2.3 SEM with some invalid IVs</a></ul><li><a href="#code" id="markdown-toc-code">Code</a><ul><li><a href="#onesample-mvstepiv" id="markdown-toc-onesample-mvstepiv">onesample mvstepIV</a><li><a href="#onesample-mvstepiv-ind" id="markdown-toc-onesample-mvstepiv-ind">onesample mvstepIV ind</a></ul><li><a href="#in-context" id="markdown-toc-in-context">In context</a><ul><li><a href="#background" id="markdown-toc-background">Background</a><li><a href="#what-is-an-instrumental-variable-iv" id="markdown-toc-what-is-an-instrumental-variable-iv">What is an Instrumental Variable (IV)?</a><li><a href="#key-assumptions" id="markdown-toc-key-assumptions">Key Assumptions</a><li><a href="#how-does-it-work" id="markdown-toc-how-does-it-work">How Does It Work?</a><li><a href="#why-is-it-necessary-for-causal-inference" id="markdown-toc-why-is-it-necessary-for-causal-inference">Why is it Necessary for Causal Inference?</a><li><a href="#usage" id="markdown-toc-usage">Usage</a><li><a href="#limitations" id="markdown-toc-limitations">Limitations</a><li><a href="#summary" id="markdown-toc-summary">Summary</a></ul><li><a href="#references" id="markdown-toc-references">References</a></ul></ul></details><h1 id="inference-of-causal-metabolite-networks"> <a href="#inference-of-causal-metabolite-networks" class="anchor-heading" aria-labelledby="inference-of-causal-metabolite-networks"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Inference of causal metabolite networks</h1><p>A summary of: Chen, S., Lin, Z., Shen, X., Li, L., &amp; Pan, W. (2023). Inference of causal metabolite networks in the presence of invalid instrumental variables with GWAS summary data. Genetic Epidemiology, 1–15, <a href="https://doi.org/10.1002/gepi.22535">https://doi.org/10.1002/gepi.22535</a> <sup id="fnref:chen2023inference"><a href="#fn:chen2023inference" class="footnote" rel="footnote" role="doc-noteref">1</a></sup>.</p><p>Briefly, this paper is about using instrumental variables (SNPs) in causal inference with applications to genome-wide association studies (GWAS).</p><ul><li><strong>Exposure:</strong><ul><li>Metabolites: They serve as intermediate phenotypes connecting genetic variants to clinical outcomes and play a crucial role in biological processes.</ul><li><strong>Instrumental Variable (IV):</strong><ul><li>SNPs (Single Nucleotide Polymorphisms): These are used as instrumental variables to isolate the variability in metabolites that is independent of confounders.</ul><li><strong>Outcome:</strong><ul><li>Clinical outcomes: These are affected by metabolites, which in turn are influenced by SNPs. The causal effect of metabolites on clinical outcomes is what the study aims to determine.</ul></ul><p>For a more gradual background jump down to the “In context” section.</p><hr /><h2 id="abstract-summary"> <a href="#abstract-summary" class="anchor-heading" aria-labelledby="abstract-summary"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Abstract summary</h2><p>Structural equation models (SEMs) for inferring causal networks in metabolites and other complex traits. The method:</p><ol><li>Performs causal analysis to discover relationships among multiple traits.<li>Accounts for potential invalid IVs.<li>Allows for data analysis using only GWAS summary statistics.<li>Considers bidirectional relationships between traits.</ol><p>The approach uses a stepwise selection to identify invalid IVs, and demonstrates its superior performance using both real and simulated GWAS data.</p><hr /><h2 id="key-methods-summary"> <a href="#key-methods-summary" class="anchor-heading" aria-labelledby="key-methods-summary"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Key methods summary</h2><h3 id="21-one-sample-data"> <a href="#21-one-sample-data" class="anchor-heading" aria-labelledby="21-one-sample-data"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.1 One-sample data</h3><p>For one-sample GWAS individual-level data:</p><ul><li>For each of \(n\) individuals:<ul><li>We have \(p\) SNPs as IVs and \(M\) traits.<li>Traits are denoted by an \(n x M\) matrix \(Y = [y_1, y_2, ... , y_M]\).<ul><li>Where \(y_i = (y_1i, y_2i, ..., y_ni)^T\) is the vector of \(n\) observations for trait \(i = 1, 2, ..., M\).</ul><li>IVs are denoted as \(Z = [z_1, ... , z_p]\).<ul><li>Where \(z_j = (z_1j, ..., z_nj)^T\) is the vector of the \(n\) observations for IV \(j = 1, 2, ..., p\).</ul></ul></ul><h3 id="22-sem-with-individual-level-data"> <a href="#22-sem-with-individual-level-data" class="anchor-heading" aria-labelledby="22-sem-with-individual-level-data"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.2 SEM with individual-level data</h3><ul><li>SEM is used as a tool for multivariate causal inference in this framework.<li>The analysis is based on one-sample GWAS individual-level data.<li>The system considers linear equations for the ‘n’ individuals in the sample.<li>Notations follow Wang et al. 2016.<li>The model has random errors denoted by a ‘n x M’ matrix.<li>The matrix ‘E’ consists of vectors for the ‘n’ random errors for each trait from 1 to M.<li>The assumption is made that the expected value of error ‘e_i’ is 0.<li>For each trait, the error follows a normal distribution with mean 0 and variance ‘σ_i^2’.<li>The matrix ‘M x M’ denotes the coefficients for the traits, represented by ‘Γ’.<li>The coefficients for each trait ‘i’ range from 1 to M.<li>The linear model for the ‘i-th’ trait is represented by an equation where traits are connected linearly and the sum is 0.</ul><h3 id="23-sem-with-some-invalid-ivs"> <a href="#23-sem-with-some-invalid-ivs" class="anchor-heading" aria-labelledby="23-sem-with-some-invalid-ivs"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 2.3 SEM with some invalid IVs</h3><ul><li>Discusses the scenario when some IVs may be invalid.<li>Violation of one or more of the three valid IV Assumptions (A1)–(A3).<li>Proposes a method to account for invalid IVs in SEM.<li>For trait i, the first \({p}_{0i}\) IVs are considered invalid IVs.<li>Represents the matrix of invalid IVs with \({Z}_{ {\mathscr{A}}_{i}}=[{z}_{1},{\rm{\ldots }},{z}_{p_{0i}}]\).<li>\({z}_{j}={({z}_{1j},{\rm{\ldots }},{z}_{nj})}^{T}\) is the vector for the n observations of IV \(j=1,2,{\rm{\ldots }},{p}_{0i}\).<li>\(M\) vectors are present for the coefficients of invalid IVs: \({B}_{1},{\rm{\ldots }},{B}_{M}\).<li>\({B}_{i}={({\beta }_{1i},{\beta }_{2i},{\rm{\ldots }},{\beta }_{p_{0i}i})}^{T}\) represents the direct or (horizontal) pleiotropic effects of the invalid IVs on trait \(i=1,2,{\rm{\ldots }},M\).<li>The linear SEM for trait i is represented as: \({y}_{1}{\gamma }_{1i}+\cdots +{y}_{i}{\gamma }_{ii}+\cdots +{y}_{M}{\gamma }_{Mi}+{z}_{1}{\beta }_{1i}+\cdots +{z}_{p_{0}i}{\beta }_{p_{0i}i}+{e}_{i}=0,\).<li>\(\gamma\)’s and \(\beta\)’s are unknown parameters in the model.</ul><hr /><h2 id="code"> <a href="#code" class="anchor-heading" aria-labelledby="code"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Code</h2><h3 id="onesample-mvstepiv"> <a href="#onesample-mvstepiv" class="anchor-heading" aria-labelledby="onesample-mvstepiv"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> onesample mvstepIV</h3><p>The original source code is here: <a href="https://github.com/chen-siyi7/one-sample-stepwise-IV-selection/blob/main/one-sample%20stepwise%20IV%20code.R">https://github.com/chen-siyi7/one-sample-stepwise-IV-selection/blob/main/one-sample%20stepwise%20IV%20code.R</a></p><p>The function <code class="language-plaintext highlighter-rouge">onesample_mvstepIV</code> conducts one-sample stepwise IV.</p><p><strong>Input Parameters:</strong></p><ul><li><code class="language-plaintext highlighter-rouge">p</code>: Total number of predictors.<li><code class="language-plaintext highlighter-rouge">R</code>: Correlation matrix.<li><code class="language-plaintext highlighter-rouge">betaZX</code>: Regression coefficients for predictors.<li><code class="language-plaintext highlighter-rouge">betaZY</code>: Regression coefficients for outcomes.<li><code class="language-plaintext highlighter-rouge">se_betaZY</code>: Standard error of <code class="language-plaintext highlighter-rouge">betaZY</code>.<li><code class="language-plaintext highlighter-rouge">n</code>: Sample size.<li><code class="language-plaintext highlighter-rouge">gamma_hat</code>: Gamma hat values (prior information).</ul><p><strong>Main Computations:</strong></p><ol><li>Initialize <code class="language-plaintext highlighter-rouge">ZTZ</code> using <code class="language-plaintext highlighter-rouge">R</code> as: \(ZTZ = R\)<li>Compute <code class="language-plaintext highlighter-rouge">ZTY</code> as the element-wise product of <code class="language-plaintext highlighter-rouge">ZTZ</code> diagonal and <code class="language-plaintext highlighter-rouge">betaZY</code>: \(ZTY = \text{diag}(ZTZ) \times \beta{ZY}\)<li>Calculate the median of <code class="language-plaintext highlighter-rouge">YTY</code> for each predictor <code class="language-plaintext highlighter-rouge">SNP</code> as: \(YTY[SNP] = (n-1) \times ZTZ[SNP,SNP] \times (se_\beta{ZY}^2)[SNP] + ZTY[SNP] \times \beta{ZY}[SNP]\), excluding NA values.<li>Compute Bayesian Information Criterion (<code class="language-plaintext highlighter-rouge">BIC</code>) for each predictor. For each predictor <code class="language-plaintext highlighter-rouge">i</code>:<ul><li>a. Create a matrix <code class="language-plaintext highlighter-rouge">test11</code> with diagonal element <code class="language-plaintext highlighter-rouge">i</code> set to 1.<li>b. Create matrix <code class="language-plaintext highlighter-rouge">W1</code> by combining <code class="language-plaintext highlighter-rouge">test11</code> and <code class="language-plaintext highlighter-rouge">gamma_hat</code>.<li>c. Solve for <code class="language-plaintext highlighter-rouge">W1</code> using: \(\text{solve.W1} = W1^T \times ZTZ \times W1\)<li>d. Compute beta estimates <code class="language-plaintext highlighter-rouge">beta1</code> as: \(\beta1 = (solve.W1^{-1} \times W1^T \times ZTY)\)<li>e. Calculate <code class="language-plaintext highlighter-rouge">BIC</code> as: \(testbic[i] = n \times \log(YTY - \beta1^T \times W1^T \times ZTY) + \log(n) \times \sum_{i} \text{diag}(test11)\)</ul><li>Determine the optimal instrument variables (<code class="language-plaintext highlighter-rouge">IVs</code>) based on <code class="language-plaintext highlighter-rouge">BIC</code>:<ul><li>a. For each iteration <code class="language-plaintext highlighter-rouge">j</code>, select the predictor <code class="language-plaintext highlighter-rouge">i</code> with the smallest <code class="language-plaintext highlighter-rouge">BIC</code>.<li>b. Repeat the process by adding one predictor at a time.<li>c. Stop if the current and previous <code class="language-plaintext highlighter-rouge">IV</code> are the same.</ul><li>Compute the final beta estimates using the invalid IVs:<ul><li>a. Extract the invalid IVs from <code class="language-plaintext highlighter-rouge">whichIV</code> and set their diagonal elements in <code class="language-plaintext highlighter-rouge">test11</code> to 1.<li>b. Compute beta estimates <code class="language-plaintext highlighter-rouge">beta1</code> as: \(\beta1 = (solve.W1^{-1} \times W1^T \times ZTY)\)<li>c. Calculate variance of beta <code class="language-plaintext highlighter-rouge">Varbeta</code> as: \(\text{Var\beta} = \text{diag}(solve.W1 \times n) \times \sigma_u2\) where \(\sigma_u2 = YTY - \beta1^T \times W1^T \times ZTY\).</ul></ol><p><strong>Output:</strong></p><ul><li><code class="language-plaintext highlighter-rouge">invalidIV</code>: Indices of invalid IVs.<li><code class="language-plaintext highlighter-rouge">beta_est</code>: Estimated beta values.<li><code class="language-plaintext highlighter-rouge">beta_se</code>: Standard error of beta estimates.<li><code class="language-plaintext highlighter-rouge">K</code>: Number of invalid IVs.</ul><h3 id="onesample-mvstepiv-ind"> <a href="#onesample-mvstepiv-ind" class="anchor-heading" aria-labelledby="onesample-mvstepiv-ind"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> onesample mvstepIV ind</h3><p>The original source code is here: <a href="https://github.com/chen-siyi7/one-sample-stepwise-IV-selection/blob/main/onesample_mvstepIV_ind%20code.R">https://github.com/chen-siyi7/one-sample-stepwise-IV-selection/blob/main/onesample_mvstepIV_ind%20code.R</a></p><p>The function <code class="language-plaintext highlighter-rouge">onesample_mvstepIV_ind</code> performs one-sample stepwise IV for independent SNPs.</p><p><strong>Input Parameters:</strong></p><ul><li><code class="language-plaintext highlighter-rouge">Y</code>: Response variable.<li><code class="language-plaintext highlighter-rouge">Z</code>: Predictor matrix.<li><code class="language-plaintext highlighter-rouge">n</code>: Sample size.<li><code class="language-plaintext highlighter-rouge">gamma_hat</code>: Gamma hat values (prior information).</ul><p><strong>Main Computations:</strong></p><ol><li>Initialize <code class="language-plaintext highlighter-rouge">testbic</code> for Bayesian Information Criterion.<li>For each predictor <code class="language-plaintext highlighter-rouge">i</code>:<ul><li>a. Initialize a zero vector <code class="language-plaintext highlighter-rouge">l</code> with length <code class="language-plaintext highlighter-rouge">dim(Z)[2]</code> and set the <code class="language-plaintext highlighter-rouge">i</code>th element to 1.<li>b. Modify matrix <code class="language-plaintext highlighter-rouge">Z22</code> such that for each row <code class="language-plaintext highlighter-rouge">j</code>, <code class="language-plaintext highlighter-rouge">Z22[j,]</code> is <code class="language-plaintext highlighter-rouge">Z[j,]*l</code>.<li>c. Perform a linear regression (<code class="language-plaintext highlighter-rouge">lm_stage2</code>) of <code class="language-plaintext highlighter-rouge">Y</code> on <code class="language-plaintext highlighter-rouge">Z22</code> and <code class="language-plaintext highlighter-rouge">Z*gamma_hat</code>.<li>d. Calculate <code class="language-plaintext highlighter-rouge">BIC</code> for this predictor using: \(testbic[i] = n \times \log\left(\frac{\sum(lm\_stage2\text{residuals}^2)}{n}\right) + \log(n) \times \sum(l)\)</ul><li>Determine the optimal instrument variables (<code class="language-plaintext highlighter-rouge">IVs</code>) based on <code class="language-plaintext highlighter-rouge">BIC</code>:<ul><li>a. For each iteration <code class="language-plaintext highlighter-rouge">j</code>, select the predictor <code class="language-plaintext highlighter-rouge">i</code> with the smallest <code class="language-plaintext highlighter-rouge">BIC</code>.<li>b. Modify matrix <code class="language-plaintext highlighter-rouge">Z22</code> for the selected predictors and add one predictor at a time.<li>c. Repeat the linear regression and calculate <code class="language-plaintext highlighter-rouge">BIC</code> as in step 2.<li>d. Stop if the current and previous <code class="language-plaintext highlighter-rouge">IV</code> are the same.</ul><li>Extract the invalid IVs, <code class="language-plaintext highlighter-rouge">which.invalid</code>, from <code class="language-plaintext highlighter-rouge">whichIV</code> and sort them to obtain <code class="language-plaintext highlighter-rouge">K</code>.<li>Compute the final beta estimates using the invalid IVs:<ul><li>a. Extract columns <code class="language-plaintext highlighter-rouge">K</code> from <code class="language-plaintext highlighter-rouge">Z</code> to form <code class="language-plaintext highlighter-rouge">Z22</code>.<li>b. Perform linear regression (<code class="language-plaintext highlighter-rouge">lm_stage2</code>) of <code class="language-plaintext highlighter-rouge">Y</code> on <code class="language-plaintext highlighter-rouge">Z22</code> and <code class="language-plaintext highlighter-rouge">Z*gamma_hat</code>.<li>c. Calculate beta estimates <code class="language-plaintext highlighter-rouge">betaest</code> as: \(\beta{est} = \text{summary}(lm\_stage2)\text{coef[,1]}\)<li>d. Calculate variance of residuals <code class="language-plaintext highlighter-rouge">sigma_u2</code> as: \(\sigma_u2 = \frac{\sum(lm\_stage2\text{residuals}^2)}{n}\)<li>e. Calculate variance of beta <code class="language-plaintext highlighter-rouge">Varbeta</code> using: \(\text{Varbeta} = \text{diag}(ginv(X^TX)) \times \sigma_u2\) where \(X = \text{cbind}(Z22, Dhat)\) and \(Dhat = Z*gamma\_hat\).<li>f. Compute the standard error <code class="language-plaintext highlighter-rouge">betase</code> as: \(\beta{se} = \sqrt{\text{Varbeta}}\)</ul></ol><p><strong>Output:</strong></p><ul><li><code class="language-plaintext highlighter-rouge">beta_est</code>: Estimated beta values.<li><code class="language-plaintext highlighter-rouge">beta_se</code>: Standard error of beta estimates.<li><code class="language-plaintext highlighter-rouge">invalid IVs</code>: Indices of invalid IVs.<li><code class="language-plaintext highlighter-rouge">no. of invalid IV</code>: Number of invalid IVs.</ul><hr /><h2 id="in-context"> <a href="#in-context" class="anchor-heading" aria-labelledby="in-context"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> In context</h2><p>Recap:</p><ul><li><strong>Exposure:</strong><ul><li>Metabolites: They serve as intermediate phenotypes connecting genetic variants to clinical outcomes and play a crucial role in biological processes.</ul><li><strong>Instrumental Variable (IV):</strong><ul><li>SNPs (Single Nucleotide Polymorphisms): These are used as instrumental variables to isolate the variability in metabolites that is independent of confounders.</ul><li><strong>Outcome:</strong><ul><li>Clinical outcomes: These are affected by metabolites, which in turn are influenced by SNPs. The causal effect of metabolites on clinical outcomes is what the study aims to determine.</ul></ul><h3 id="background"> <a href="#background" class="anchor-heading" aria-labelledby="background"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Background</h3><p>In GWAS, associations are generally sought between single nucleotide polymorphisms (SNPs) and a single trait. But GWAS data can also be used to analyze multiple related traits, leading to improved power and new biological insights. Specifically, network analysis of multiple traits is gaining interest, especially when it comes to causal network analysis. This is pivotal for elucidating relationships among multiple traits, such as in gene network and protein network analyses. Metabolite network analysis, the focal point of this research, posits that metabolites are integral parts of many biological processes, often interacting with each other in regulatory networks. By inferring these networks, we can gain insight into relationships among metabolites in biological processes.</p><p>In causal networks, traits, including metabolites, proteins, and genes, serve as the nodes. Their causal relationships are represented by directed edges connecting them. SNPs are utilized as instrumental variables (IVs). To model these intricate biological networks, structural equation models (SEMs) have been adopted.</p><h3 id="what-is-an-instrumental-variable-iv"> <a href="#what-is-an-instrumental-variable-iv" class="anchor-heading" aria-labelledby="what-is-an-instrumental-variable-iv"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> What is an Instrumental Variable (IV)?</h3><p>An instrumental variable is associated with the exposure but does not have a direct association with the outcome, except through its relationship with the exposure. Its role is to isolate the variability in the exposure that is independent of the confounders.</p><h3 id="key-assumptions"> <a href="#key-assumptions" class="anchor-heading" aria-labelledby="key-assumptions"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Key Assumptions</h3><ul><li><strong>Relevance:</strong> The IV is correlated with the exposure.<li><strong>Exclusion:</strong> The IV only affects the outcome through its effect on the exposure.<li><strong>Exchangeability:</strong> The IV is not associated with unobserved confounders.</ul><h3 id="how-does-it-work"> <a href="#how-does-it-work" class="anchor-heading" aria-labelledby="how-does-it-work"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> How Does It Work?</h3><p>IV analysis uses the variation in the exposure explained by the instrument to estimate the causal effect of the exposure on the outcome.</p><h3 id="why-is-it-necessary-for-causal-inference"> <a href="#why-is-it-necessary-for-causal-inference" class="anchor-heading" aria-labelledby="why-is-it-necessary-for-causal-inference"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Why is it Necessary for Causal Inference?</h3><ul><li><strong>Control for Unmeasured Confounding:</strong> IVs can provide unbiased estimates of causal effects when unmeasured or unobserved confounding is present.<li><strong>Endogeneity:</strong> IVs can solve the problem of endogeneity.<li><strong>Natural Experiments:</strong> IVs can be employed in “natural experiments” where random assignment of treatments is not feasible.</ul><h3 id="usage"> <a href="#usage" class="anchor-heading" aria-labelledby="usage"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Usage</h3><p>In the context of GWAS and metabolite network analysis, IV methods are crucial. They help determine causal relationships in complex biological processes, especially when metabolites, which do not function in isolation, interact within metabolite regulatory networks.</p><h3 id="limitations"> <a href="#limitations" class="anchor-heading" aria-labelledby="limitations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Limitations</h3><ul><li><strong>Weak Instruments:</strong> Weak correlation between the IV and exposure can lead to biased IV estimates.<li><strong>Violations of Assumptions:</strong> IV estimates can be biased if any core assumptions are violated.<li><strong>Interpretability:</strong> The causal effect estimated through IV is often specific to a particular population, reducing generalizability.</ul><h3 id="summary"> <a href="#summary" class="anchor-heading" aria-labelledby="summary"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Summary</h3><p>Instrumental Variables are a pivotal tool in causal inference, especially in genome-wide association studies (GWAS). When utilized properly, they can provide valuable insights into causal relationships in settings laden with confounding and endogeneity. However, they come with their own assumptions and potential limitations.</p><hr /><h2 id="references"> <a href="#references" class="anchor-heading" aria-labelledby="references"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> References</h2><div class="footnotes" role="doc-endnotes"><ol><li id="fn:chen2023inference"><p>Chen, S., Lin, Z., Shen, X., Li, L., &amp; Pan, W. (2023). Inference of causal metabolite networks in the presence of invalid instrumental variables with GWAS summary data. Genetic Epidemiology, 1–15, <a href="https://doi.org/10.1002/gepi.22535">https://doi.org/10.1002/gepi.22535</a>. <a href="#fnref:chen2023inference" class="reversefootnote" role="doc-backlink">&#8617;</a></p></ol></div><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
