<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>PCA biplot 1000genomes | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="PCA biplot 1000genomes" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/pca_biplot_1kg.html" /><meta property="og:url" content="http://localhost:4000/pages/pca_biplot_1kg.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="PCA biplot 1000genomes" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"PCA biplot 1000genomes","url":"http://localhost:4000/pages/pca_biplot_1kg.html"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item active"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link active">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><h1 id="preparing-a-public-data-reference-panel-from-1000-genomes-project-for-pca"> <a href="#preparing-a-public-data-reference-panel-from-1000-genomes-project-for-pca" class="anchor-heading" aria-labelledby="preparing-a-public-data-reference-panel-from-1000-genomes-project-for-pca"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Preparing a public data reference panel from 1000 genomes project for PCA</h1><p>Last update: 20240807</p><details open="" class="no_toc"> <summary class="text-delta"> Table of contents </summary><ul id="markdown-toc"><li><a href="#preparing-a-public-data-reference-panel-from-1000-genomes-project-for-pca" id="markdown-toc-preparing-a-public-data-reference-panel-from-1000-genomes-project-for-pca">Preparing a public data reference panel from 1000 genomes project for PCA</a><ul><li><a href="#overview" id="markdown-toc-overview">Overview</a><li><a href="#data-source" id="markdown-toc-data-source">Data Source</a><li><a href="#tools-and-tutorials-referenced" id="markdown-toc-tools-and-tutorials-referenced">Tools and tutorials referenced</a><li><a href="#1000-genomes-project" id="markdown-toc-1000-genomes-project">1000 genomes project</a><ul><li><a href="#pedigree-data" id="markdown-toc-pedigree-data">Pedigree data</a><li><a href="#population-codes" id="markdown-toc-population-codes">Population codes</a></ul><li><a href="#pca-eigenvectors" id="markdown-toc-pca-eigenvectors">PCA eigenvectors</a><li><a href="#scripts-overview" id="markdown-toc-scripts-overview">Scripts Overview</a><li><a href="#step-by-step-summary" id="markdown-toc-step-by-step-summary">Step-by-Step Summary</a><li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></ul></ul></details><p><img src="/assets/images/PCA_biplot_1kg.png" width="100%" /> <strong>Figure 1</strong>: Principal component analysis (PCA) of 1000 genomes project, reference genome GRCh38, showing population structure.</p><h2 id="overview"> <a href="#overview" class="anchor-heading" aria-labelledby="overview"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Overview</h2><p>This documentation outlines the process of preparing a reference panel using the 1000 Genomes Project data, focusing on converting data to PLINK format and performing Principal Component Analysis (PCA). This enables mapping of cohort data to determine population labels and provides a reference for genetic diversity analysis.</p><h2 id="data-source"> <a href="#data-source" class="anchor-heading" aria-labelledby="data-source"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data Source</h2><p>The data utilized is derived from the GRCh38 release of the 1000 Genomes Project, specifically from:</p><ul><li>1000 Genomes Project - GRCh38:<li><a href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/release/20190312_biallelic_SNV_and_INDEL/">http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/release/20190312_biallelic_SNV_and_INDEL/</a></ul><h2 id="tools-and-tutorials-referenced"> <a href="#tools-and-tutorials-referenced" class="anchor-heading" aria-labelledby="tools-and-tutorials-referenced"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Tools and tutorials referenced</h2><ul><li><strong>Kevin Blighe’s Biostar Tutorial</strong>: “Produce PCA bi-plot for 1000 Genomes Phase III - Version 2” available at Biostars <a href="https://www.biostars.org/p/335605/">https://www.biostars.org/p/335605/</a>.<li><strong>1000 genomes project</strong>: Homepage <a href="https://www.internationalgenome.org">https://www.internationalgenome.org</a></ul><h2 id="1000-genomes-project"> <a href="#1000-genomes-project" class="anchor-heading" aria-labelledby="1000-genomes-project"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> 1000 genomes project</h2><p>This is a project which provides public WGS data in VCF format and related metadata. The pedigree informaiton comes from <code class="language-plaintext highlighter-rouge">20130606_g1k.ped</code>. We used the GRCh38 liftover data set <a href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/phase3_liftover_nygc_dir/">http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/phase3_liftover_nygc_dir/</a>.</p><blockquote><p>The phase 3 variant calls released by the 1000 Genomes project was on GRCh37 reference. To be able to compare them with the variant calls on the high coverage data they had to be lifted over to GRCh38. The liftover was performed at New York Genome Center (NYGC) using CrossMap version 0.5.4. The GRCh37 phase 3 calls used in the liftover are available here, ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/. The chain file used in liftover is available at UCSC and can be downloaded from https://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/ .</p></blockquote><blockquote><p>We did not attempt to liftover SVs that were in phase 3. CrossMap does not liftover any records that either had multiple hits on GRCh38 or after liftover the REF allele matches ALT allele. Additionally, we failed any record that was lifted over to a different chromosome or if the REF allele contained symbols (Y, W, Z etc.).</p></blockquote><p>[1] Download the files as VCF.gz (and tab-indices).</p><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#!/bin/bash
prefix="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/phase3_liftover_nygc_dir/phase3.chr" ;

suffix=".GRCh38.GT.crossmap.vcf.gz" ;

for chr in {1..22} X Y; do
    wget "${prefix}""${chr}""${suffix}" \
               "${prefix}""${chr}""${suffix}".tbi ;
done

wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/phase3_liftover_nygc_dir/phase3.crossmap.GRCh38.07302021.README.html
wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/phase3_liftover_nygc_dir/phase3.crossmap.GRCh38.07302021.manifest.tsv

mkdir 1000genomes
mv phase3* 1000genomes/
</code></pre></div></div><p>[2] Download 1000 Genomes PED file. <code class="language-plaintext highlighter-rouge">wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_g1k.ped </code></p><p>[3] Download (link to the existing copy) of the reference genome.</p><p>Read here: <a href="ref.html">Reference genome</a>.</p><h3 id="pedigree-data"> <a href="#pedigree-data" class="anchor-heading" aria-labelledby="pedigree-data"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Pedigree data</h3><div class="table-wrapper"><table><thead><tr><th>Family.ID<th>Individual.ID<th>Paternal.ID<th>Maternal.ID<th>Gender<th>Phenotype<th>Population<th>Relationship<tbody><tr><td>HG00096<td>HG00096<td>0<td>0<td>1<td>0<td>GBR<td>unrel<tr><td>HG00097<td>HG00097<td>0<td>0<td>2<td>0<td>GBR<td>unrel<tr><td>HG00099<td>HG00099<td>0<td>0<td>2<td>0<td>GBR<td>unrel<tr><td>HG00100<td>HG00100<td>0<td>0<td>2<td>0<td>GBR<td>unrel<tr><td>HG00101<td>HG00101<td>0<td>0<td>1<td>0<td>GBR<td>unrel<tr><td>HG00102<td>HG00102<td>0<td>0<td>2<td>0<td>GBR<td>unrel</table></div><h3 id="population-codes"> <a href="#population-codes" class="anchor-heading" aria-labelledby="population-codes"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Population codes</h3><div class="table-wrapper"><table><thead><tr><th>Code<th>Population Description<th>Geographic/Ethnic Details<tbody><tr><td>CHB<td>Han Chinese<td>Han Chinese in Beijing, China<tr><td>JPT<td>Japanese<td>Japanese in Tokyo, Japan<tr><td>CHS<td>Southern Han Chinese<td>Han Chinese South<tr><td>CDX<td>Dai Chinese<td>Chinese Dai in Xishuangbanna, China<tr><td>KHV<td>Kinh Vietnamese<td>Kinh in Ho Chi Minh City, Vietnam<tr><td>CHD<td>Denver Chinese<td>Chinese in Denver, Colorado (pilot 3 only)<tr><td>CEU<td>CEPH<td>Utah residents (CEPH) with Northern and Western European ancestry<tr><td>TSI<td>Tuscan<td>Toscani in Italia<tr><td>GBR<td>British<td>British in England and Scotland<tr><td>FIN<td>Finnish<td>Finnish in Finland<tr><td>IBS<td>Spanish<td>Iberian populations in Spain<tr><td>YRI<td>Yoruba<td>Yoruba in Ibadan, Nigeria<tr><td>LWK<td>Luhya<td>Luhya in Webuye, Kenya<tr><td>GWD<td>Gambian<td>Gambian in Western Division, The Gambia<tr><td>MSL<td>Mende<td>Mende in Sierra Leone<tr><td>ESN<td>Esan<td>Esan in Nigeria<tr><td>ASW<td>African-American SW<td>African Ancestry in Southwest US<tr><td>ACB<td>African-Caribbean<td>African Caribbean in Barbados<tr><td>MXL<td>Mexican-American<td>Mexican Ancestry in Los Angeles, California<tr><td>PUR<td>Puerto Rican<td>Puerto Rican in Puerto Rico<tr><td>CLM<td>Colombian<td>Colombian in Medellin, Colombia<tr><td>PEL<td>Peruvian<td>Peruvian in Lima, Peru<tr><td>GIH<td>Gujarati<td>Gujarati Indian in Houston, TX<tr><td>PJL<td>Punjabi<td>Punjabi in Lahore, Pakistan<tr><td>BEB<td>Bengali<td>Bengali in Bangladesh<tr><td>STU<td>Sri Lankan<td>Sri Lankan Tamil in the UK<tr><td>ITU<td>Indian<td>Indian Telugu in the UK</table></div><h2 id="pca-eigenvectors"> <a href="#pca-eigenvectors" class="anchor-heading" aria-labelledby="pca-eigenvectors"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> PCA eigenvectors</h2><div class="table-wrapper"><table><thead><tr><th>Individual<th>PC1<th>PC2<th>PC3<th>PC4<th>PC5<tbody><tr><td>HG00096<td>-0.01032<td>0.0270<td>0.0117<td>0.0192<td>0.002517<tr><td>HG00097<td>-0.01054<td>0.0275<td>0.0104<td>0.0180<td>0.003890<tr><td>HG00099<td>-0.01067<td>0.0275<td>0.0104<td>0.0168<td>0.001831<tr><td>HG00100<td>-0.00968<td>0.0275<td>0.0109<td>0.0191<td>-0.000839<tr><td>HG00101<td>-0.01038<td>0.0270<td>0.0116<td>0.0184<td>0.000796<tr><td>HG00102<td>-0.01063<td>0.0272<td>0.0110<td>0.0178<td>0.003824</table></div><h2 id="scripts-overview"> <a href="#scripts-overview" class="anchor-heading" aria-labelledby="scripts-overview"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Scripts Overview</h2><p>The process is currently set up in three scripts:</p><ul><li>pca_biplot_1kg.sh, pca_biplot_1kg_part2.sh, pca_biplot_1kg_part3_ggplot.R<li><strong>1</strong>: Parallel processing (per chromosome) for the conversion of 1000 Genomes VCF files to BCF and subsequently to PLINK format. The script handles data normalization, ID reformatting, and variant pruning to reduce linkage disequilibrium, enhancing the quality of genetic association analyses.<li><strong>2</strong>: Continues from Script 1, merging PLINK files across chromosomes into a single dataset, performing PCA to explore population structure.<li><strong>3</strong>: Uses <code class="language-plaintext highlighter-rouge">ggplot2</code> in R to visualize the PCA results, highlighting the population stratification among global populations. This script also integrates demographic data to color-code populations in the bi-plot, providing clear visual insights into genetic diversity.</ul><h2 id="step-by-step-summary"> <a href="#step-by-step-summary" class="anchor-heading" aria-labelledby="step-by-step-summary"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step Summary</h2><ol><li><strong>Data conversion and normalisation</strong><ul><li>Convert original VCF files to BCF using <code class="language-plaintext highlighter-rouge">bcftools</code>, ensuring all variants have unique IDs based on chromosome positions.<li>Annotate and reformat using <code class="language-plaintext highlighter-rouge">bcftools annotate</code> to adjust variant IDs.<li>Normalise and remove duplicate variants to clean the dataset for further analysis.</ul><li><strong>PLINK file preparation</strong><ul><li>Convert BCF files to PLINK format, ensuring allele orders are maintained.<li>Prune variants using PLINK to reduce the dataset based on minor allele frequency and linkage disequilibrium.</ul><li><strong>Data merging and PCA analysis</strong><ul><li>Merge all chromosome-specific PLINK files into a single dataset.<li>Perform PCA to identify principal components that explain the maximum variance, indicative of population stratification.</ul><li><strong>Visualisation and interpretation</strong><ul><li>Use <code class="language-plaintext highlighter-rouge">ggplot2</code> and <code class="language-plaintext highlighter-rouge">patchwork</code> in R to create bi-plots of the first few principal components.<li>Overlay population data to visually interpret population structure and genetic diversity.</ul></ol><h2 id="conclusion"> <a href="#conclusion" class="anchor-heading" aria-labelledby="conclusion"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Conclusion</h2><p>This reference panel and PCA analysis provide a framework for understanding genetic diversity and population structure within global populations, using the latest 1000 Genomes Project data. This methodological approach is essential for genomic studies requiring a comprehensive understanding of genetic backgrounds.</p><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
