<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Storage architecture plan | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Storage architecture plan" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/storage_architecture_plan.html" /><meta property="og:url" content="http://localhost:4000/pages/storage_architecture_plan.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Storage architecture plan" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Storage architecture plan","url":"http://localhost:4000/pages/storage_architecture_plan.html"}</script> <script type="text/javascript" defer src="/assets/js/mathjax-script-type.js"> </script> <script type="text/javascript" id="MathJax-script" defer src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"> </script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metadata_users.html" class="nav-list-link">WGS metadata users</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/my_voice.html" class="nav-list-link">QV - My voice</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item active"><a href="/pages/storage_architecture_plan.html" class="nav-list-link active">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><p>Last update: 20250727</p><h1 id="storage-architecture-plan"> <a href="#storage-architecture-plan" class="anchor-heading" aria-labelledby="storage-architecture-plan"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Storage architecture plan</h1><ul id="markdown-toc"><li><a href="#storage-architecture-plan" id="markdown-toc-storage-architecture-plan">Storage architecture plan</a><ul><li><a href="#storage" id="markdown-toc-storage">Storage</a><li><a href="#system-overview" id="markdown-toc-system-overview">System overview</a><ul><li><a href="#code-and-projects-src" id="markdown-toc-code-and-projects-src">Code and projects (<code class="language-plaintext highlighter-rouge">/src/</code>)</a><li><a href="#data-data" id="markdown-toc-data-data">Data (<code class="language-plaintext highlighter-rouge">/data/</code>)</a><li><a href="#media-media" id="markdown-toc-media-media">Media (<code class="language-plaintext highlighter-rouge">/media/</code>)</a><li><a href="#documentation-docs" id="markdown-toc-documentation-docs">Documentation (<code class="language-plaintext highlighter-rouge">/docs/</code>)</a><li><a href="#finance-finance" id="markdown-toc-finance-finance">Finance (<code class="language-plaintext highlighter-rouge">/finance/</code>)</a><li><a href="#offsite-backup-with-backblaze-b2" id="markdown-toc-offsite-backup-with-backblaze-b2">Offsite backup with Backblaze B2</a><li><a href="#summary" id="markdown-toc-summary">Summary</a><li><a href="#nas-and-raid-config" id="markdown-toc-nas-and-raid-config">NAS and RAID config</a><ul><li><a href="#ugreen-nasync-dxp4800-plus" id="markdown-toc-ugreen-nasync-dxp4800-plus">Ugreen NASync DXP4800 Plus</a><li><a href="#synology-models-for-reference" id="markdown-toc-synology-models-for-reference">Synology models for reference</a></ul><li><a href="#decision" id="markdown-toc-decision">Decision</a></ul></ul></ul><hr /><p>This page documents the planning for physical data storage and backup architecture.</p><p>➡️ See <a href="storage_use_sync_and_versioning">Storage, usage, and Git practices</a> for how we work with project folders, sync, and versioning day to day.</p><p>Here we provide the details of one pice of hardware in use; a Ugreen NASync DXP4800 Plus with RAID 5 for reliable onsite storage, combined with selective offsite backups (e.g. Backblaze B2) for critical data. This setup provides fast local access, resilience against hardware failure, and secure long-term protection without vendor lock-in. The document outlines the directory structure, NAS configuration, and storage planning rationale as a reference for maintaining a scalable and auditable data environment.</p><h2 id="storage"> <a href="#storage" class="anchor-heading" aria-labelledby="storage"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Storage</h2><div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>├── src/                     # local git projects (synced to NAS, pushed to GitHub)
│   ├── project1/
│   │   ├── analysis/
│   │   ├── scripts/
│   │   ├── latex/
│   │   ├── README.md
│   │   └── .git/
│   │
│   ├── project2/
│   │
│   ├── systematic_review/   # separate repo just for a stand alone paper
│   │   ├──  latex/
│   │   ├──  figures/
│   │   ├──  README.md
│   │   └── .git/
│   └── ...
│
├── data/                    # large datasets (mostly stored on NAS, copy locally if needed)
│   ├── project1/
│   │   ├── raw/             # not in git, synced to NAS
│   │   ├── processed/       # not in git, synced to NAS
│   │   ├── release/         # cleaned data for public release (upload to Zenodo, Figshare, etc.)
│   │   └── working/         # local scratch (NOT synced) - saved elsewhere when finished (raw, processed) for NAS sync
│   │
│   ├── project2/
│   └── ...
│
├── media/                   # media files (mostly stored on NAS, copy locally if needed)
│   ├── documents/
│   ├── graphics/
│   ├── videos/
│   │   └── working/         # local scratch (NOT synced) - saved elsewhere when finished (raw, processed) for NAS sync
│
└── docs/                    # company documentation (synced to NAS)
│   ├──  business_plan.md
│   ├──  marketing_notes.md
│   └──  ...
│
└── finance/
    └── ...
</code></pre></div></div><h2 id="system-overview"> <a href="#system-overview" class="anchor-heading" aria-labelledby="system-overview"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> System overview</h2><p>All work is organised under <code class="language-plaintext highlighter-rouge">~/so/</code> on the local hardware, structured into subfolders for code, data, media, and documentation. A local NAS, connected via ethernet to the office mesh Wi-Fi network, provides central storage and backup. The NAS syncs automatically with selected local folders, ensuring data redundancy and protection against hardware failures.</p><h3 id="code-and-projects-src"> <a href="#code-and-projects-src" class="anchor-heading" aria-labelledby="code-and-projects-src"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Code and projects (<code class="language-plaintext highlighter-rouge">/src/</code>)</h3><p>The <code class="language-plaintext highlighter-rouge">/src/</code> directory contains all software development projects, including analysis scripts and LaTeX or Markdown for writing academic papers. These projects are managed with local git commits and pushed to GitHub for version control and collaboration, while also being synchronised to the NAS for backup.</p><h3 id="data-data"> <a href="#data-data" class="anchor-heading" aria-labelledby="data-data"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data (<code class="language-plaintext highlighter-rouge">/data/</code>)</h3><p>Large datasets reside under <code class="language-plaintext highlighter-rouge">/data/</code>. Most data remain stored on the NAS to save local disk space, but can be copied locally when running compute-intensive tools. Keeping a clear auditable structure. Within each project’s data folder:</p><ul><li><code class="language-plaintext highlighter-rouge">raw/</code> and <code class="language-plaintext highlighter-rouge">processed/</code> (or equivalent) directories hold the main files.<li><code class="language-plaintext highlighter-rouge">release/</code> is prepared for public sharing through repositories like Zenodo, Figshare, EGA, customer products.<li><code class="language-plaintext highlighter-rouge">working/</code> serves as local scratch space for temporary files during analysis, providing faster performance for intermediate steps before results are saved back to the NAS.</ul><h3 id="media-media"> <a href="#media-media" class="anchor-heading" aria-labelledby="media-media"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Media (<code class="language-plaintext highlighter-rouge">/media/</code>)</h3><p>The <code class="language-plaintext highlighter-rouge">/media/</code> directory houses content that we generate such as documents, books, graphics, and video projects. Smaller files sync continuously with the NAS. As with analysis <code class="language-plaintext highlighter-rouge">data/</code>, for large media work such as videos, raw footage is copied locally into the <code class="language-plaintext highlighter-rouge">working/</code> subfolder to ensure fast read and write speeds during editing, and completed projects are then archived back to the NAS for long-term storage. No sensitive data or critical IP.</p><h3 id="documentation-docs"> <a href="#documentation-docs" class="anchor-heading" aria-labelledby="documentation-docs"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Documentation (<code class="language-plaintext highlighter-rouge">/docs/</code>)</h3><p>The <code class="language-plaintext highlighter-rouge">/docs/</code> folder stores company-level documentation and internal materials, all synced to the NAS to keep critical records safe and accessible. Keeping a clear auditable structure.</p><h3 id="finance-finance"> <a href="#finance-finance" class="anchor-heading" aria-labelledby="finance-finance"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Finance (<code class="language-plaintext highlighter-rouge">/finance/</code>)</h3><p>Equivalent to <code class="language-plaintext highlighter-rouge">/docs/</code> with a dedicated space. Keeping a clear auditable structure.</p><h3 id="offsite-backup-with-backblaze-b2"> <a href="#offsite-backup-with-backblaze-b2" class="anchor-heading" aria-labelledby="offsite-backup-with-backblaze-b2"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Offsite backup with Backblaze B2</h3><p>In addition to local and NAS storage, the system uses Backblaze B2 for offsite backup. The NAS connects to Backblaze B2 via Hyper Backup tool, which automatically schedules encrypted backups of selected folders (such as <code class="language-plaintext highlighter-rouge">/src/</code>, <code class="language-plaintext highlighter-rouge">/data/</code>, <code class="language-plaintext highlighter-rouge">/media/</code>, and <code class="language-plaintext highlighter-rouge">/docs/</code>). This ensures that all critical files are protected against physical disasters like fire, theft, or catastrophic hardware failure.</p><p>Hyper Backup allows versioning, so older copies of files can be restored if needed. The cost is low, typically around $5 per terabyte per month. Scratch folders like <code class="language-plaintext highlighter-rouge">working/</code> in both <code class="language-plaintext highlighter-rouge">/data/</code> and <code class="language-plaintext highlighter-rouge">/media/</code> are not included in these backups, as they are meant for temporary local use and cleared when projects are complete.</p><h3 id="summary"> <a href="#summary" class="anchor-heading" aria-labelledby="summary"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Summary</h3><p>This setup ensures that software is always run locally from desktops (e.g. Mac) for speed and control, while data and media can reside on the NAS unless local speed is required. Scratch folders in both <code class="language-plaintext highlighter-rouge">/data/</code> and <code class="language-plaintext highlighter-rouge">/media/</code> provide local-only working space for tasks needing high performance, keeping the main storage system organised and efficient. Meanwhile, Backblaze B2 provides a secure, offsite layer of protection for long-term data safety.</p><h3 id="nas-and-raid-config"> <a href="#nas-and-raid-config" class="anchor-heading" aria-labelledby="nas-and-raid-config"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> NAS and RAID config</h3><p>We use RAID 5 to balance redundancy and capacity for a compact research team. This provides protection against a single disk failure while maximising usable storage capacity. In case of a disk failure, the plan is to replace the failed drive and allow RAID to rebuild from parity. Additional safety comes from offsite backups to Backblaze B2, and from code and documents stored in remote repositories like GitHub, ensuring important media and data can be restored if needed. For now, RAID 5 is sufficient, with the option to switch to RAID 6 in the future if the project expands or data becomes more critical. Currently, our datasets are calculated from public resources which can be reproduced if necessary.</p><p>For a project of &lt; 10 TB:</p><div class="table-wrapper"><table><thead><tr><th>Model<th>CHF<th>Drives<th>RAID Config<th>Usable Space<th>CHF per usable TB<tbody><tr><td>DS224+ (2×8 TB)<td>753<td>2 × 8 TB<td>RAID 1<td>8 TB<td>~CHF 94 / TB<tr><td>DS423+ (4×4 TB)<td>919<td>4 × 4 TB<td>RAID 5<td>~12 TB<td>~CHF 76 / TB<tr><td>DS423+ (4×4 TB)<td>919<td>4 × 4 TB<td>RAID 6<td>~8 TB<td>~CHF 115 / TB<tr><td>DS923+ (4×4 TB)<td>1029<td>4 × 4 TB<td>RAID 5<td>~12 TB<td>~CHF 86 / TB<tr><td>Ugreen DXP4800 Plus (4×4 TB)<td>1029<td>4 × 4 TB<td>RAID 5<td>~12 TB<td>~CHF 86 / TB<tr><td>Ugreen DXP4800 Plus (3×8 TB)<td>1119<td>3 × 8 TB<td>RAID 5<td>~16 TB<td>~CHF 70 / TB</table></div><p>Spending:</p><ul><li>Ugreen NASync DXP4800 Plus chassis: CHF 588.–<li>WD Red Plus 8 TB drives × 3: CHF 177.– each → CHF 531.–<li><strong>NAS Total</strong>: CHF 1119.–<li>Apple Thunderbolt to Gigabit Ethernet Adapter (MD463ZM/A): CHF 14.89 + CHF 4.20 shipping → <strong>Total</strong> CHF 19.09</ul><hr /><h4 id="ugreen-nasync-dxp4800-plus"> <a href="#ugreen-nasync-dxp4800-plus" class="anchor-heading" aria-labelledby="ugreen-nasync-dxp4800-plus"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Ugreen NASync DXP4800 Plus</h4><p>The <strong>Ugreen NASync DXP4800 Plus</strong> is a modern NAS offering high hardware performance and flexibility:</p><ul><li><strong>CPU:</strong> Intel Pentium Gold 8505, 5 cores / 6 threads, up to 4.40 GHz<li><strong>RAM:</strong> 8 GB DDR5 (expandable up to 64 GB)<li><strong>Drive bays:</strong> 4 × SATA (2.5″ / 3.5″) + 2 × M.2 NVMe<li><strong>Maximum storage capacity:</strong> up to 104 TB (4 × 26 TB drives)<li><strong>Network ports:</strong> 1 × 10GbE + 1 × 2.5GbE<li><strong>RAID support:</strong> 0, 1, 5, 6, 10, JBOD<li><strong>Connectivity:</strong> HDMI 4K, SD card reader, multiple USB ports (USB-A, USB-C)<li><strong>Weight:</strong> 6.02 kg<li><strong>Dimensions:</strong> 25.75 × 17.80 × 17.80 cm<li><strong>Country of origin:</strong> China<li><strong>Release date:</strong> 7 October 2024</ul><p>Unlike Synology, Ugreen allows the use of <strong>third-party drives</strong> without vendor lock-in, which reduces future costs and provides flexibility in hardware upgrades. It also features built-in 10GbE networking for high-speed transfers, and offers significant headroom for RAM and CPU-intensive workloads, making it well-suited for future data growth and heavier operations.</p><hr /><h4 id="synology-models-for-reference"> <a href="#synology-models-for-reference" class="anchor-heading" aria-labelledby="synology-models-for-reference"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Synology models for reference</h4><p>While Synology remains a leader in NAS software (DSM) and integrated cloud services, their hardware offers comparatively modest specifications:</p><ul><li>Lower base RAM (2–4 GB), expandable only on some models<li>Often requires purchase of additional cards for 10GbE networking<li>Potential future restrictions on third-party drives, increasing long-term costs<li>Excellent software ecosystem, with mature support for cloud backups like Backblaze B2</ul><hr /><h3 id="decision"> <a href="#decision" class="anchor-heading" aria-labelledby="decision"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Decision</h3><p>Given the combination of hardware power, freedom from vendor lock-in, built-in 10GbE networking, and expandability, we have chosen the <strong>Ugreen NASync DXP4800 Plus</strong> as the NAS platform. While Synology remains an excellent solution for those prioritising a mature software environment, Ugreen offers superior hardware value and flexibility, which aligns better with the performance and cost-efficiency goals of the project.</p><hr /><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
