<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Structural variation detection | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Structural variation detection" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/sv.html" /><meta property="og:url" content="http://localhost:4000/pages/sv.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Structural variation detection" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Structural variation detection","url":"http://localhost:4000/pages/sv.html"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/storage.html" class="nav-list-link">storage</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style guide</a><li class="nav-list-item active"><a href="/pages/sv.html" class="nav-list-link active">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><h1 id="structural-variation-detection"> <a href="#structural-variation-detection" class="anchor-heading" aria-labelledby="structural-variation-detection"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Structural variation detection</h1><p>Last update: 20240820</p><h2 id="understanding-structural-variation-detection-with-smoove-lumpy-and-duphold"> <a href="#understanding-structural-variation-detection-with-smoove-lumpy-and-duphold" class="anchor-heading" aria-labelledby="understanding-structural-variation-detection-with-smoove-lumpy-and-duphold"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Understanding Structural Variation Detection with smoove, lumpy, and duphold</h2><p>Structural variations (SVs) such as deletions, duplications, inversions, and translocations play a critical role in genomic diversity and disease. Detecting these variations accurately in whole-genome sequencing data requires sophisticated bioinformatics tools. Here, we discuss how Smoove, Lumpy, and Duphold work together to detect and interpret SVs, focusing on their methodologies rather than specific commands or installation procedures.</p><ul><li>smoove - <a href="https://github.com/brentp/smoove">https://github.com/brentp/smoove</a><li>lumpy-sv - <a href="https://github.com/arq5x/lumpy-sv">https://github.com/arq5x/lumpy-sv</a><li>duphold - <a href="https://github.com/brentp/duphold">https://github.com/brentp/duphold</a></ul><p>Example results quoted from “https://github.com/brentp/duphold”:</p><blockquote><p>A clear deletion will have rapid drop in depth at the left and increase in depth at the right and a lower mean coverage. Duphold annotated this with: DHBFC: 0.6. These indicate that both break-points are consistent with a deletion and that the coverage is ~60% of expected. So this is a clear deletion.</p></blockquote><p><img src="https://user-images.githubusercontent.com/1739/45895721-2dbaf400-bd8f-11e8-88b3-9fd5a90ef39e.png" width="100%" /></p><blockquote><p>BND - when lumpy decides that a cluster of evidence does not match a DUP or DEL or INV, it creates a BND with 2 lines in the VCF. Sometimes these are actual deletions. For example this shows where a deletion is bounded by 2 BND calls. duphold annotates this with: DHBFC: 0.01 indicating a homozygous deletion with clear break-points.</p></blockquote><p><img src="https://user-images.githubusercontent.com/1739/45906495-987d2700-bdb1-11e8-8ba5-eacdf8221f68.png" width="100%" /></p><h3 id="smoove-streamlining-lumpys-process"> <a href="#smoove-streamlining-lumpys-process" class="anchor-heading" aria-labelledby="smoove-streamlining-lumpys-process"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Smoove: Streamlining Lumpy’s Process</h3><p>Smoove is a software that enhances Lumpy’s SV calling capabilities. It wraps around Lumpy and other tools to simplify and speed up the process, especially for large-scale genomic studies. Here’s how Smoove enhances the detection of structural variations:</p><ol><li><strong>Read Filtering:</strong><ul><li>Smoove employs <code class="language-plaintext highlighter-rouge">lumpy_filter</code> to extract split reads and discordant reads, which are critical for identifying potential SVs.<li>It filters out low-quality signals such as reads with excessive soft-clipping or multiple mismatches, and those aligned to multiple locations.<li>Reads in specified exclude regions or those contributing to regions of abnormally high coverage (suggestive of repetitive sequences) are also discarded.</ul><li><strong>Parallel Processing:</strong><ul><li>Smoove streams Lumpy’s output directly into <code class="language-plaintext highlighter-rouge">svtyper</code>, facilitating simultaneous genotyping across different genomic regions. This significantly reduces the computational time and memory requirements.</ul><li><strong>Population-Level Calling:</strong><ul><li>For larger datasets, Smoove enables joint calling by first calling variants in smaller groups (e.g., by family) and then merging these calls. This is followed by re-genotyping at these combined sites, further refined by integrating all findings into a comprehensive VCF using tools like <code class="language-plaintext highlighter-rouge">svtools</code> and <code class="language-plaintext highlighter-rouge">bcftools</code>.</ul></ol><h3 id="lumpy-a-probabilistic-framework-for-sv-discovery"> <a href="#lumpy-a-probabilistic-framework-for-sv-discovery" class="anchor-heading" aria-labelledby="lumpy-a-probabilistic-framework-for-sv-discovery"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Lumpy: A Probabilistic Framework for SV Discovery</h3><p>Lumpy serves as the foundational tool for SV detection used by Smoove. It offers a flexible and highly sensitive approach to detect SVs across different types and sizes by analyzing the signatures of split reads and discordant read pairs:</p><ul><li><strong>Probabilistic Modeling:</strong><ul><li>Lumpy uses a probabilistic model to integrate evidence from various read types, including paired-end and split reads, to predict SVs.<li>This method allows for the detection of complex SVs that might not be identified through traditional read-mapping approaches.</ul></ul><h3 id="duphold-enhancing-sv-calls-with-depth-information"> <a href="#duphold-enhancing-sv-calls-with-depth-information" class="anchor-heading" aria-labelledby="duphold-enhancing-sv-calls-with-depth-information"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Duphold: Enhancing SV Calls with Depth Information</h3><p>Duphold complements Smoove and Lumpy by adding depth-based annotations to SV calls, which helps in assessing the confidence of these variants:</p><ul><li><strong>Depth Annotations:</strong><ul><li><strong>DHFC:</strong> Compares the depth of coverage within the SV to the chromosome average.<li><strong>DHBFC:</strong> Compares the depth within the SV to genomic regions with similar GC content, which helps control for sequencing biases.<li><strong>DHFFC:</strong> Compares the depth within the SV to its immediate flanking regions, providing a localized assessment of depth changes.</ul><li><strong>Annotation Against SNP/Indel VCFs:</strong><ul><li>Duphold can use existing SNP/Indel data to annotate SVs, helping to identify unlikely deletions that contain multiple heterozygous SNP calls, for example.</ul></ul><h3 id="practical-application-in-cohort-studies"> <a href="#practical-application-in-cohort-studies" class="anchor-heading" aria-labelledby="practical-application-in-cohort-studies"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Practical Application in Cohort Studies</h3><p>In practice, these tools enable researchers to effectively identify and validate structural variations across large cohorts. By integrating read-based and depth-based evidence, researchers can filter out spurious calls and focus on variants that are most likely to be true positives. Visualisations such as histograms of depth changes and variant allele frequencies, along with comparative plots across samples or groups, provide intuitive insights into the data, aiding further in the interpretation of results.</p><h3 id="conclusion"> <a href="#conclusion" class="anchor-heading" aria-labelledby="conclusion"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Conclusion</h3><p>The combination of Smoove, Lumpy, and Duphold offers a robust framework for the detection and interpretation of structural variations in genomic datasets. By leveraging their individual strengths in read processing, probabilistic modeling, and depth annotation, researchers can achieve a high level of accuracy and efficiency in SV detection, essential for advancing our understanding of genomic structure and its impact on health and disease.</p><h2 id="running-smoove-using-singularity"> <a href="#running-smoove-using-singularity" class="anchor-heading" aria-labelledby="running-smoove-using-singularity"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Running Smoove using Singularity</h2><p>Singularity provides an effective way to utilize Docker containers in environments where Docker itself may not be suitable, such as shared HPC systems. Here’s a generalized example of how you can deploy and run <code class="language-plaintext highlighter-rouge">smoove</code> using a Singularity container, which is useful for calling and genotyping structural variants in genomic data.</p><h3 id="step-by-step-guide"> <a href="#step-by-step-guide" class="anchor-heading" aria-labelledby="step-by-step-guide"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Step-by-Step Guide:</h3><ol><li><strong>Obtaining the Smoove Singularity Image:</strong><ul><li>First, pull the <code class="language-plaintext highlighter-rouge">smoove</code> Docker image from Docker Hub and convert it into a Singularity image file (.sif). This step typically requires admin privileges or may be handled by your system administrator:<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>singularity pull docker://brentp/smoove
</code></pre></div></div><li>This command creates a <code class="language-plaintext highlighter-rouge">.sif</code> file that can be used directly on any system where Singularity is installed.</ul><li><strong>Preparing the Environment:</strong><ul><li>Load necessary modules, if applicable, such as <code class="language-plaintext highlighter-rouge">samtools</code>. Ensure all dependencies like reference genome files and exclude region BED files are accessible.</ul><li><strong>Running Smoove:</strong><ul><li>Use the Singularity <code class="language-plaintext highlighter-rouge">exec</code> command to run <code class="language-plaintext highlighter-rouge">smoove</code> within the Singularity container. It’s analogous to using <code class="language-plaintext highlighter-rouge">docker run</code> but tailored for Singularity environments. Here’s an example command structure based on a SLURM script for batch processing:<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>singularity <span class="nb">exec </span>smoove_latest.sif smoove call <span class="nt">--outdir</span> results-smoove/ <span class="se">\</span>
  <span class="nt">--exclude</span> exclude_regions.bed <span class="se">\</span>
  <span class="nt">--name</span> sample_name <span class="se">\</span>
  <span class="nt">--fasta</span> reference.fasta <span class="se">\</span>
  <span class="nt">-p</span> 1 <span class="se">\</span>
  <span class="nt">--genotype</span> sample.bam
</code></pre></div></div><li>This command runs <code class="language-plaintext highlighter-rouge">smoove</code> to call structural variants for a single sample. It specifies an output directory, an exclusion BED file for problematic regions, the sample name, the reference genome, and the input BAM file.</ul><li><strong>Handling Multiple Samples:</strong><ul><li>If processing multiple samples, consider using SLURM’s array job feature to parallelize the process. Each job can process a different sample, effectively distributing the workload across multiple compute nodes or cores.</ul><li><strong>Post-Processing:</strong><ul><li>After generating VCF files for individual samples, you may need to merge these using <code class="language-plaintext highlighter-rouge">smoove merge</code> to create a combined VCF file that includes variants from all samples:<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>singularity <span class="nb">exec </span>smoove_latest.sif smoove merge <span class="nt">--name</span> merged <span class="nt">-f</span> reference.fasta <span class="nt">--outdir</span> ./ results-smoove/<span class="k">*</span>.genotyped.vcf.gz
</code></pre></div></div><li>Subsequently, genotype each sample at the merged sites and optionally run <code class="language-plaintext highlighter-rouge">duphold</code> to add depth annotations, enhancing the interpretability of the SV calls.</ul><li><strong>Output and Validation:</strong><ul><li>The final output will typically be a set of VCF files containing the called and genotyped structural variants for your cohort. These can be further analyzed or visualized using additional tools to assess the quality and implications of the detected variants.</ul></ol><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
