<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>Variant to RDF concept | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="Variant to RDF concept" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/variant_concept.html" /><meta property="og:url" content="http://localhost:4000/pages/variant_concept.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="Variant to RDF concept" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"Variant to RDF concept","url":"http://localhost:4000/pages/variant_concept.html"}</script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_evidence_flag.html" class="nav-list-link">Design QV evidence framework</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metadata_users.html" class="nav-list-link">WGS metadata users</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/my_voice.html" class="nav-list-link">QV - My voice</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/qv_design.html" class="nav-list-link">QV design principles</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item active"><a href="/pages/variant_concept.html" class="nav-list-link active">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item"><a href="/pages/vsat.html" class="nav-list-link">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><p>Last update: 20240920</p><details open="" class="no_toc"> <summary class="text-delta">Table of contents</summary><ul id="markdown-toc"><li><a href="#variant-features-to-rdf-concept-metadata" id="markdown-toc-variant-features-to-rdf-concept-metadata">Variant features to RDF concept metadata</a><ul><li><a href="#overview" id="markdown-toc-overview">Overview</a><li><a href="#aims" id="markdown-toc-aims">Aims</a><li><a href="#current-version" id="markdown-toc-current-version">Current version</a><li><a href="#semantic-evidence-network" id="markdown-toc-semantic-evidence-network">Semantic evidence network</a><li><a href="#downloads" id="markdown-toc-downloads">Downloads</a><li><a href="#process-steps-for-variant-features-to-rdf-concept-mapping" id="markdown-toc-process-steps-for-variant-features-to-rdf-concept-mapping">Process Steps for Variant Features to RDF Concept Mapping</a><li><a href="#terms-used-in-wgs-logging" id="markdown-toc-terms-used-in-wgs-logging">Terms used in WGS logging</a><ul><li><a href="#descriptions-for-sequencing-assay-wgs-terms" id="markdown-toc-descriptions-for-sequencing-assay-wgs-terms">Descriptions for sequencing assay (WGS) terms</a><li><a href="#descriptions-for-fastqbam-readgroup-terms" id="markdown-toc-descriptions-for-fastqbam-readgroup-terms">Descriptions for FASTQ/BAM readgroup terms</a><li><a href="#descriptions-for-genetic-variants-terms" id="markdown-toc-descriptions-for-genetic-variants-terms">Descriptions for genetic variants terms</a></ul></ul></ul></details><h1 id="variant-features-to-rdf-concept-metadata"> <a href="#variant-features-to-rdf-concept-metadata" class="anchor-heading" aria-labelledby="variant-features-to-rdf-concept-metadata"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Variant features to RDF concept metadata</h1><p class="note">This process is starting with <code class="language-plaintext highlighter-rouge">sequencing_assay</code>, which includes <code class="language-plaintext highlighter-rouge">library_preparation</code>, <code class="language-plaintext highlighter-rouge">sequencing_run</code>, etc. We will continue through the pipeline until we reach the final end-point required to report a pathogenic variant.</p><p>This documentation outlines the transformation of variant information from whole genome sequence (WGS) data to a format adhering to RDF structure data concepts. The aim is to ensure that the omic output from genomic analyses can be seamlessly integrated into clinical data warehouses with high fidelity and clarity.</p><p><strong>Number of variables</strong>:</p><ul><li>All SPHN RDF concept info (see <code class="language-plaintext highlighter-rouge">SPHN_dataset_release_2024_2_20240502.xlsx</code>) = 1503<li>Subset of relevant concepts = 76 (see <code class="language-plaintext highlighter-rouge">example_subset_concepts.tsv</code>)<li>Relevant WGS pipeline logs = 62 (see <code class="language-plaintext highlighter-rouge">example_report.tsv</code>)<li><p>Currently automated match = 13 (see <code class="language-plaintext highlighter-rouge">example_report_concepts.Rds</code>)</p><p class="note">This repository uses a public dataset of example genetic variants and sequencing/analysis log data.</p></ul><h2 id="overview"> <a href="#overview" class="anchor-heading" aria-labelledby="overview"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Overview</h2><p>The process begins with the extraction of variant data from a genomic study, (no sensitive data is included in the public example set). The key variant features such as Chromosome (CHROM), Position (POS), Reference Allele (REF), and Alternate Allele (ALT) are formatted alongside metadata that describes their relationship to RDF concepts. This ensures downstream users can map these data accurately within clinical and research frameworks.</p><p class="note">This document is to be updated as we improve the linking of result terms to <code class="language-plaintext highlighter-rouge">SPHN_dataset_release_2024_2_20240502.xlsx</code> which is critical so that downstream users can correctly map data.</p><h2 id="aims"> <a href="#aims" class="anchor-heading" aria-labelledby="aims"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Aims</h2><ol><li><strong>Data preparation</strong>: Start with the extracted variant information from the genomic pipeline.<li><strong>Key term identification</strong>: Focus on essential genomic terms like CHROM, POS, REF, and ALT, Sequencing run, Sequencing instrument.<li><strong>Metadata addition</strong>: Attach metadata columns that specify RDF concept requirements such as type and cardinality.<li><strong>Validation checklist</strong>:<ul><li>Do we have all necessary variant descriptors present?<li>Is there inclusion and accuracy of all metadata explanations?<li>Is there alignment of metadata with SPHN omic concepts?<li>Downstream users (mapping) can choose from TSV, HTML, JSON, and Rds. Any others needed?</ul></ol><h2 id="current-version"> <a href="#current-version" class="anchor-heading" aria-labelledby="current-version"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Current version</h2><p>The observation column is highlighted in GREEN. It contains the data which we report as output from the pipeline for use in our database Here is the completed concept observations (this is file <code class="language-plaintext highlighter-rouge">example_report_concepts.html</code>):</p><div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; overflow-x: scroll; width:100%; "><div class="table-wrapper"><table class="table table-striped table-hover table-condensed" style="font-size: 14px; color: black; width: auto !important; "><thead><tr><th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> cardinalityViolated<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> concept_reference_general_concept_name<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> concept_reference<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> general_concept_name<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> observation<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> release<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> unique_ID<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> IRI<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> active_status_(yes/no)<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> deprecated_in<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> replaced_by<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> concept_or_concept_compositions_or_inherited<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> general_description<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> contextualized_concept_name<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> contextualized_description<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> parent<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> type<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> excluded_type_descendants<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> standard<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> value_set_or_subset<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> meaning_binding<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> additional_information<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> cardinality_for_composedOf<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> cardinality_for_concept_to_Administrative_Case<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> cardinality_for_concept_to_Data_Provider<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> cardinality_for_concept_to_Subject_Pseudo_Identifier<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> cardinality_for_concept_to_Source_System<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> sensitive_(yes/no)<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> color_inherited<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> color_reference<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> color_observation<th style="text-align:left;font-weight: bold;color: black !important;background-color: rgba(211, 211, 211, 255) !important;position: sticky; top:0; background-color: #FFFFFF;"> color_cardinality<tbody><tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_sequencing_assay<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> sequencing_assay<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#sequencingassay<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(124, 202, 255, 255) !important;"> concept<td style="text-align:left;"> an_assay_that_exploits_a_sequencer_as_the_instrument_to_generate_results<td style="text-align:left;"> sequencing_assay<td style="text-align:left;"> an_assay_that_exploits_a_sequencer_as_the_instrument_to_generate_results<td style="text-align:left;"> assay<td style="text-align:left;"> assay<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> efo:0003740_|assay_by_sequencer|<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:n<td style="text-align:left;"> 1:1<td style="text-align:left;"> 0:n<td style="text-align:left;"> 1:n<td style="text-align:left;"> NA<td style="text-align:left;"> #7CCAFF<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_standard_operating_procedure<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> standard_operating_procedure<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> wgs_with_illumina_novaseq_6000<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasstandardoperatingprocedure<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> standard_operating_procedure_associated_to_the_concept<td style="text-align:left;"> standard_operating_procedure<td style="text-align:left;"> standard_operating_procedure_that_was_followed_for_this_sequencing_assay<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> standard_operating_procedure<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_predecessor<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> predecessor<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> kispi_custom_sample_prep_v1<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#haspredecessor<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> process_preceding_this_concept<td style="text-align:left;"> predecessor<td style="text-align:left;"> sample_processing_preceding_the_sequencing_assay<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> sample_processing<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:n<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_code<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> code<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> efo_0022396<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hascode<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> coded_information_specifying_the_concept<td style="text-align:left;"> code<td style="text-align:left;"> code_specifying_the_type_of_sequencing_assay<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> code<td style="text-align:left;"> NA<td style="text-align:left;"> efo;_obi_or_other<td style="text-align:left;"> for_efo:_descendant_of:_efo:0001455_|assay|;_for_obi:_descendant_of:_obi:0000070_|assay|<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 1:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_identifier<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> identifier<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> obo:obi_002117_(wgs)<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasidentifier<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> unique_identifier_identifying_the_concept<td style="text-align:left;"> identifier<td style="text-align:left;"> unique_identifier_identifying_the_sequencing_assay<td style="text-align:left;"> sphnattributedatatype<td style="text-align:left;"> string<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_start_datetime<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> start_datetime<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> jul 01 2023 01:01:01 gmt / v0.9.0<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasstartdatetime<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> datetime_at_which_the_concept_started<td style="text-align:left;"> start_datetime<td style="text-align:left;"> datetime_at_which_the_sequencing_assay_was_first_executed<td style="text-align:left;"> hasdatetime<td style="text-align:left;"> temporal<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> yes<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_data_file<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> data_file<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> out.fastq<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasdatafile<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> data_file_associated_to_the_concept<td style="text-align:left;"> data_file<td style="text-align:left;"> data_file_associated_to_the_sequencing_assay<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> data_file<td style="text-align:left;"> time_series_data_file<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:n<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_sample<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> sample<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> blood_sample_1<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hassample<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> sample_associated_to_the_concept<td style="text-align:left;"> sample<td style="text-align:left;"> material_that_is_being_sequenced_by_this_sequencing_assay<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> sample<td style="text-align:left;"> tumor_specimen;_isolate<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:n<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_library_preparation<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> library_preparation<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> illumina_truseq_dna_pcr-free<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#haslibrarypreparation<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> library_preparation_associated_to_the_concept<td style="text-align:left;"> library_preparation<td style="text-align:left;"> the_library_preparation_that_is_part_of_the_sequencing_assay<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> library_preparation<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_sequencing_instrument<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> sequencing_instrument<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> a00485<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hassequencinginstrument<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> device_associated_to_the_concept<td style="text-align:left;"> sequencing_instrument<td style="text-align:left;"> the_device_which_is_used_to_perform_the_sequencing_assay<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> sequencing_instrument<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_sequencing_run<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> sequencing_run<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> 334<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hassequencingrun<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> sequencing_run_associated_to_the_concept<td style="text-align:left;"> sequencing_run<td style="text-align:left;"> sequencing_run_performed_as_part_of_the_sequencing_assay<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> sequencing_run<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:n<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_intended_read_length<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> intended_read_length<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> 150<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasintendedreadlength<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> intended_read_length_associated_to_the_concept<td style="text-align:left;"> intended_read_length<td style="text-align:left;"> the_number_of_nucleotides_intended_to_be_ordered_from_each_side_of_a_nucleic_acid_fragment_obtained_after_the_completion_of_a_sequencing_assay<td style="text-align:left;"> hasquantity<td style="text-align:left;"> quantity<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_assay_intended_read_depth<td style="text-align:left;background-color: rgba(247, 202, 201, 255) !important;"> sequencing_assay<td style="text-align:left;"> intended_read_depth<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> 30x<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasintendedreaddepth<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> intended_read_depth_associated_to_the_concept<td style="text-align:left;"> intended_read_depth<td style="text-align:left;"> the_number_of_times_a_particular_locus_(site,_nucleotide,_amplicon,_region)_was_intended_to_be_sequenced_as_part_of_the_sequencing_assay<td style="text-align:left;"> hasquantity<td style="text-align:left;"> quantity<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7cac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_library_preparation<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> library_preparation<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#librarypreparation<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(124, 202, 255, 255) !important;"> concept<td style="text-align:left;"> process_which_results_in_the_creation_of_a_library_from_fragments_of_dna<td style="text-align:left;"> library_preparation<td style="text-align:left;"> process_which_results_in_the_creation_of_a_library_from_fragments_of_dna<td style="text-align:left;"> sampleprocessing<td style="text-align:left;"> sample_processing<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> obi:0000711_|library_preparation|<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:n<td style="text-align:left;"> 1:1<td style="text-align:left;"> 0:n<td style="text-align:left;"> 1:n<td style="text-align:left;"> NA<td style="text-align:left;"> #7CCAFF<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_code<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> code<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hascode<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> coded_information_specifying_the_concept<td style="text-align:left;"> code<td style="text-align:left;"> code_specifying_the_type_of_library_preparation<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> code<td style="text-align:left;"> NA<td style="text-align:left;"> obi;_efo_or_other<td style="text-align:left;"> for_obi:_descendant_of:_obi:0000711_|library_preparation|<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_input<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> input<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasinput<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> input_associated_to_the_concept<td style="text-align:left;"> input<td style="text-align:left;"> the_sample_for_which_a_library_is_created<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> sample<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:n<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_output<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> output<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasoutput<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> output_associated_to_the_concept<td style="text-align:left;"> output<td style="text-align:left;"> the_ngs_library_that_is_produced<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> sample<td style="text-align:left;"> tumor_specimen<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_start_datetime<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> start_datetime<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasstartdatetime<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> datetime_at_which_the_concept_started<td style="text-align:left;"> start_datetime<td style="text-align:left;"> start_of_library_preparation<td style="text-align:left;"> hasdatetime<td style="text-align:left;"> temporal<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> yes<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_quality_control_metric<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> quality_control_metric<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasqualitycontrolmetric<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> quality_control_metric_associated_to_the_concept<td style="text-align:left;"> quality_control_metric<td style="text-align:left;"> quality_control_metric_related_to_the_output_of_the_library_preparation<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> quality_control_metric<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:n<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_predecessor<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> predecessor<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#haspredecessor<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> process_preceding_this_concept<td style="text-align:left;"> predecessor<td style="text-align:left;"> process_preceding_this_library_preparation<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> sample_processing<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:n<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_standard_operating_procedure<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> standard_operating_procedure<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasstandardoperatingprocedure<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(171, 177, 207, 255) !important;"> inherited<td style="text-align:left;"> standard_operating_procedure_associated_to_the_concept<td style="text-align:left;"> standard_operating_procedure<td style="text-align:left;"> standard_operating_procedure_that_was_followed_for_this_library_preparation<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> standard_operating_procedure<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #abb1cf<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_kit_code<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> kit_code<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#haskitcode<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> coded_information_specifying_the_kit_associated_to_the_concept<td style="text-align:left;"> library_preparation_kit_code<td style="text-align:left;"> pre-filled,_ready-to-use_reagent_cartridges_intended_to_improve_chemistry,_cluster_density_and_read_length_as_well_as_improve_quality_(q)_scores_for_this_sample._reagent_components_are_encoded_to_interact_with_the_sequencing_system_to_validate_compatibility_with_user-defined_applications.<td style="text-align:left;"> hascode<td style="text-align:left;"> code<td style="text-align:left;"> NA<td style="text-align:left;"> efo,_genepio,_fairgenomes_or_other<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_target_enrichment_kit_code<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> target_enrichment_kit_code<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hastargetenrichmentkitcode<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> coded_information_specifying_the_target_enrichment_kit_associated_to_the_concept<td style="text-align:left;"> target_enrichment_kit_code<td style="text-align:left;"> indicates_which_target_enrichment_kit_was_used_to_prepare_this_sample._target_enrichment_is_a_pre-sequencing_dna_preparation_step_where_dna_sequences_are_either_directly_amplified_(amplicon_or_multiplex_pcr-based)_or_captured_(hybrid_capture-based)_in_order_to_only_focus_on_specific_regions_of_a_genome_or_dna_sample.<td style="text-align:left;"> hascode<td style="text-align:left;"> code<td style="text-align:left;"> NA<td style="text-align:left;"> efo,_genepio,_fairgenomes_or_other<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_intended_insert_size<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> intended_insert_size<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasintendedinsertsize<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> intended_insert_size_associated_to_the_concept<td style="text-align:left;"> intended_insert_size<td style="text-align:left;"> in_paired-end_sequencing,_the_dna_between_the_adapter_sequences_is_the_insert._the_length_of_this_sequence_is_known_as_the_insert_size,_not_to_be_confused_with_the_inner_distance_between_reads._so,_fragment_length_equals_read_adapter_length_(2x)_plus_insert_size,_and_insert_size_equals_read_length_(2x)_plus_inner_distance.<td style="text-align:left;"> hasquantity<td style="text-align:left;"> quantity<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> library_preparation_gene_panel<td style="text-align:left;background-color: rgba(247, 246, 201, 255) !important;"> library_preparation<td style="text-align:left;"> gene_panel<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasgenepanel<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> gene_panel_associated_to_the_concept<td style="text-align:left;"> gene_panel<td style="text-align:left;"> collection_of_genes_that_are_the_focus_of_sequencing<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> gene_panel<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7f6c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_instrument_sequencing_instrument<td style="text-align:left;background-color: rgba(247, 231, 201, 255) !important;"> sequencing_instrument<td style="text-align:left;"> sequencing_instrument<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> a00485<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#sequencinginstrument<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(124, 202, 255, 255) !important;"> concept<td style="text-align:left;"> a_sequencing_instrument_that_is_used_in_a_sequencing_assay<td style="text-align:left;"> sequencing_instrument<td style="text-align:left;"> a_sequencing_instrument_that_is_used_in_a_sequencing_assay<td style="text-align:left;"> sphnconcept<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> efo:0000548_|instrument|<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 1:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #7CCAFF<td style="text-align:left;"> #f7e7c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_instrument_code<td style="text-align:left;background-color: rgba(247, 231, 201, 255) !important;"> sequencing_instrument<td style="text-align:left;"> code<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> a00485<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hascode<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> coded_information_specifying_the_concept<td style="text-align:left;"> code<td style="text-align:left;"> code_specifying_the_type_of_sequencing_instrument<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> code<td style="text-align:left;"> NA<td style="text-align:left;"> obi;_efo_or_other<td style="text-align:left;"> for_obi:_descendant_of:_obi:0400103_|dna_sequencer|;_for_efo:_descendant_of:_efo:0003739_|sequencer|<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 1:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7e7c9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_run_sequencing_run<td style="text-align:left;background-color: rgba(247, 218, 201, 255) !important;"> sequencing_run<td style="text-align:left;"> sequencing_run<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#sequencingrun<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(124, 202, 255, 255) !important;"> concept<td style="text-align:left;"> the_valid_and_completed_operation_of_a_high-throughput_sequencing_instrument_associated_with_a_sequencing_assay<td style="text-align:left;"> sequencing_run<td style="text-align:left;"> the_valid_and_completed_operation_of_a_high-throughput_sequencing_instrument_associated_with_a_sequencing_assay<td style="text-align:left;"> sphnconcept<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> ncit:c148088_|sequencing_run|<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 1:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #7CCAFF<td style="text-align:left;"> #f7dac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_run_identifier<td style="text-align:left;background-color: rgba(247, 218, 201, 255) !important;"> sequencing_run<td style="text-align:left;"> identifier<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasidentifier<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> unique_identifier_identifying_the_concept<td style="text-align:left;"> identifier<td style="text-align:left;"> unique_identifier_identifying_the_sequencing_run<td style="text-align:left;"> sphnattributedatatype<td style="text-align:left;"> string<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7dac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_run_datetime<td style="text-align:left;background-color: rgba(247, 218, 201, 255) !important;"> sequencing_run<td style="text-align:left;"> datetime<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasdatetime<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> datetime_of_the_concept<td style="text-align:left;"> datetime<td style="text-align:left;"> datetime_the_sequencing_run_was_performed<td style="text-align:left;"> sphnattributedatatype<td style="text-align:left;"> temporal<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> yes<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7dac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_run_read_count<td style="text-align:left;background-color: rgba(247, 218, 201, 255) !important;"> sequencing_run<td style="text-align:left;"> read_count<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasreadcount<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> ready_count_associated_with_to_concept<td style="text-align:left;"> read_count<td style="text-align:left;"> the_number_of_sequencing_reaction_results_that_were_pooled_to_assemble_a_sequence_for_a_genomic_region_of_interest_in_a_sequencing_run<td style="text-align:left;"> hasquantity<td style="text-align:left;"> quantity<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7dac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_run_average_insert_size<td style="text-align:left;background-color: rgba(247, 218, 201, 255) !important;"> sequencing_run<td style="text-align:left;"> average_insert_size<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasaverageinsertsize<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> average_insert_size_associated_to_the_concept<td style="text-align:left;"> average_insert_size<td style="text-align:left;"> the_average_insert_size_found_during_the_nucleic_acid_sequencing_run<td style="text-align:left;"> hasquantity<td style="text-align:left;"> quantity<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7dac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_run_average_read_length<td style="text-align:left;background-color: rgba(247, 218, 201, 255) !important;"> sequencing_run<td style="text-align:left;"> average_read_length<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasaveragereadlength<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> average_read_length_associated_to_the_concept<td style="text-align:left;"> average_read_length<td style="text-align:left;"> the_average_length_for_nucleic_acid_sequencing_reads_generated_in_a_sequencing_run<td style="text-align:left;"> hasquantity<td style="text-align:left;"> quantity<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7dac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_run_mean_read_depth<td style="text-align:left;background-color: rgba(247, 218, 201, 255) !important;"> sequencing_run<td style="text-align:left;"> mean_read_depth<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> NA<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasmeanreaddepth<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> mean_read_depth_associated_to_the_concept<td style="text-align:left;"> mean_read_depth<td style="text-align:left;"> the_number_of_times_a_particular_locus_(site,_nucleotide,_amplicon,_region)_was_sequenced_in_a_sequencing_run<td style="text-align:left;"> hasquantity<td style="text-align:left;"> quantity<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 0:1<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7dac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_run_data_file<td style="text-align:left;background-color: rgba(247, 218, 201, 255) !important;"> sequencing_run<td style="text-align:left;"> data_file<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> ../out/example_wgs_sequencing_report_deliverable_summary.tsv<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasdatafile<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> data_file_associated_to_the_concept<td style="text-align:left;"> data_file<td style="text-align:left;"> data_file_associated_to_the_sequencing_run<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> data_file<td style="text-align:left;"> time_series_data_file<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 1:n<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7dac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0<tr><td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> FALSE<td style="text-align:left;"> sequencing_run_quality_control_metric<td style="text-align:left;background-color: rgba(247, 218, 201, 255) !important;"> sequencing_run<td style="text-align:left;"> quality_control_metric<td style="text-align:left;background-color: rgba(142, 211, 160, 255) !important;"> 5f4dcc3b5aa765d61d8327deb882cf99<td style="text-align:left;"> 2024.1<td style="text-align:left;"> NA<td style="text-align:left;"> https://www.biomedit.ch/rdf/sphn-schema/sphn/2024/1#hasqualitycontrolmetric<td style="text-align:left;"> yes<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;background-color: rgba(146, 168, 209, 255) !important;"> composedof<td style="text-align:left;"> quality_control_metric_associated_to_the_concept<td style="text-align:left;"> quality_control_metric<td style="text-align:left;"> quality_control_metric_associated_with_the_sequencing_run<td style="text-align:left;"> sphnattributeobject<td style="text-align:left;"> quality_control_metric<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> 1:n<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> NA<td style="text-align:left;"> #92a8d1<td style="text-align:left;"> #f7dac9<td style="text-align:left;"> #8ed3a0<td style="text-align:left;"> #8ed3a0</table></div></div><h2 id="semantic-evidence-network"> <a href="#semantic-evidence-network" class="anchor-heading" aria-labelledby="semantic-evidence-network"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Semantic evidence network</h2><p>I have added the following method to automatically plot semantic evidence networks which show how evidence provenance has been generated. The dataset is organised into three hierarchical grouping levels based on the column <code class="language-plaintext highlighter-rouge">concept_or_concept_compositions_or_inherited</code>. The top level, Level 1, includes entries where this column equals concept. The subsequent levels, Level 2 and Level 3, contain entries where this column does not equal concept. The distinction between Levels 2 and 3 lies in the presence of distinct observations; Level 3 specifically represents the final observation associated with the general concept names from Level 2, differentiated further by non-empty values in the observation column, making Level 3 essentially a detailed continuation of Level 2.</p><p>Nodes within the network are structured with the following attributes: <code class="language-plaintext highlighter-rouge">general_concept_name</code>, <code class="language-plaintext highlighter-rouge">id</code>, <code class="language-plaintext highlighter-rouge">group</code>, <code class="language-plaintext highlighter-rouge">name</code>, and <code class="language-plaintext highlighter-rouge">observation</code>, where <code class="language-plaintext highlighter-rouge">general_concept_name</code> recurs in both Level 2 and Level 3 but differs based on the associated observation. Edges within this hierarchical setup link nodes from Level 1 to Level 2 and from Level 2 to Level 3 using <code class="language-plaintext highlighter-rouge">general_concept_name</code> as a consistent link identifier, facilitating a connection between the initial abstract concept level and its more detailed observational breakdowns.</p><p><img src="/assets/images/plot_semantic_evidence_plot_network.png" width="100%" /> <img src="/assets/images/plot_semantic_evidence_plot_sankey.png" width="100%" /></p><h2 id="downloads"> <a href="#downloads" class="anchor-heading" aria-labelledby="downloads"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Downloads</h2><p>Example output (in mutiple filetypes) can be downloaded from the public set:</p><div class="table-wrapper"><table><thead><tr><th>File Name<th>Download Link<tbody><tr><td><code class="language-plaintext highlighter-rouge">example_report_concepts.tsv</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/concept/out/example_report_concepts.tsv">Download</a><tr><td><code class="language-plaintext highlighter-rouge">example_report_concepts.html</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/concept/out/example_report_concepts.html">Download</a><tr><td><code class="language-plaintext highlighter-rouge">example_report_concepts.Rds</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/concept/out/example_report_concepts.Rds">Download</a><tr><td><code class="language-plaintext highlighter-rouge">example_report_concepts.Rds</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/concept/out/example_report_concepts.Rds">Download</a><tr><td><code class="language-plaintext highlighter-rouge">example_report_concepts.Rds</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/concept/out/example_report_concepts.Rds">Download</a><tr><td><code class="language-plaintext highlighter-rouge">pdf plot_semantic_evidence_plot_network.pdf</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/assets/images/plot_semantic_evidence_plot_network.pdf">Download</a><tr><td><code class="language-plaintext highlighter-rouge">pdf plot_semantic_evidence_plot_sankey.pdf</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/assets/images/plot_semantic_evidence_plot_sankey.pdf">Download</a><tr><td><code class="language-plaintext highlighter-rouge">html plot_semantic_evidence_plot_network.html</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/assets/images/plot_semantic_evidence_plot_network.html">Download</a><tr><td><code class="language-plaintext highlighter-rouge">html plot_semantic_evidence_plot_sankey.html</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/assets/images/plot_semantic_evidence_plot_sankey.html">Download</a></table></div><p>Example inputs can be downloaded from the public set:</p><div class="table-wrapper"><table><thead><tr><th>File Name<th>Download Link<tbody><tr><td><code class="language-plaintext highlighter-rouge">Canton_001_NGS000012345_NA_S46_L001_R1_001.fastq_head.text</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/Canton_001_NGS000012345_NA_S46_L001_R1_001.fastq_head.text">Download</a><tr><td><code class="language-plaintext highlighter-rouge">Canton_001_NGS000012345_NA_S46_L001_R1_001_sample_seq_assay_log.text</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/Canton_001_NGS000012345_NA_S46_L001_R1_001_sample_seq_assay_log.text">Download</a><tr><td><code class="language-plaintext highlighter-rouge">SPHN_dataset_release_2024_2_20240502.xlsx</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/SPHN_dataset_release_2024_2_20240502.xlsx">Download</a><tr><td><code class="language-plaintext highlighter-rouge">sequencing assay_van_der_Horst2023.txt</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/sequencing_assay_van_der_Horst2023.txt">Download</a><tr><td><code class="language-plaintext highlighter-rouge">bwa_10351_101_10453.out.text</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/bwa_10351_101_10453.out.text">Download</a><tr><td><code class="language-plaintext highlighter-rouge">example_variant.Rds</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/example_variant.Rds">Download</a><tr><td><code class="language-plaintext highlighter-rouge">example_variant.tsv</code><td><a href="https://github.com/SwissPedHealth-PipelineDev/docs/tree/main/_includes/example_variant.tsv">Download</a></table></div><h2 id="process-steps-for-variant-features-to-rdf-concept-mapping"> <a href="#process-steps-for-variant-features-to-rdf-concept-mapping" class="anchor-heading" aria-labelledby="process-steps-for-variant-features-to-rdf-concept-mapping"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Process Steps for Variant Features to RDF Concept Mapping</h2><p>This section outlines the sequential processing steps from data extraction through to the final merged dataset, prepared for RDF concept mapping. Each step corresponds to a specific script and handles distinct data types or stages in data preparation and merging.</p><ol><li><strong>Export Variant Data from Study</strong><ul><li>Extracts variant data from genomic projects focusing on specific genes and filtering for high-impact variants, saving them in formats like RDS and TSV for further processing.</ul><li><strong>Read Variant Report Data</strong><ul><li>Loads and transforms variant data into a long format to facilitate metadata annotation, preparing the data by adding a column for metadata requirements.</ul><li><strong>Read Sequencing Assay Data</strong><ul><li>Extracts key sequencing assay data such as identifiers, read depth, and file formats from logs or metadata files, providing crucial context for sequencing parameters.</ul><li><strong>Read BWA Read Group Data</strong><ul><li>Parses BWA and samtools log files to extract detailed read group information, including metadata about the sequencing run such as machine, file paths, and read group specifications.</ul><li><strong>Read Fastq Header Data</strong> - Analyzes headers from FASTQ files to extract sequencing instrument details and run metrics, offering a granular look at the sequencing runs which is instrumental in validating sequencing quality and parameters.<li><strong>Merge Datasets</strong><ul><li>Combines all processed data from the previous steps into a single dataset, aligning them by common identifiers and ensuring consistency across data types.</ul><li><strong>Map Pipeline Output to SPHN Concepts</strong><ul><li>Maps the merged dataset to standard SPHN RDF concepts, ensuring each data point is correctly classified according to standardized ontology, thus aligning detailed genomic data with broader healthcare data standards.</ul></ol><h2 id="terms-used-in-wgs-logging"> <a href="#terms-used-in-wgs-logging" class="anchor-heading" aria-labelledby="terms-used-in-wgs-logging"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Terms used in WGS logging</h2><h3 id="descriptions-for-sequencing-assay-wgs-terms"> <a href="#descriptions-for-sequencing-assay-wgs-terms" class="anchor-heading" aria-labelledby="descriptions-for-sequencing-assay-wgs-terms"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Descriptions for sequencing assay (WGS) terms</h3><div class="table-wrapper"><table><thead><tr><th>Column Name<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">seq_assay_identifier</code><td>The unique identifier for the sequencing assay, typically a standard ontology term such as obo:OBI_002117 for Whole Genome Sequencing.<tr><td><code class="language-plaintext highlighter-rouge">seq_assay_intended_read_depth</code><td>The targeted depth of coverage for the sequencing assay, indicating how many times each base is expected to be sequenced; in this case, 150x.<tr><td><code class="language-plaintext highlighter-rouge">seq_assay_intended_read_length</code><td>The expected length of each read in the sequencing process, measured in base pairs; here, 20 bp.<tr><td><code class="language-plaintext highlighter-rouge">data_file_identifier</code><td>Identifier for the data file output from the sequencing, used to trace and access the file within data systems.<tr><td><code class="language-plaintext highlighter-rouge">data_file_format</code><td>The format of the sequencing data files, specifying the standard used; here, EDAM format 1931, which is typical for FASTQ files from Illumina platforms.<tr><td><code class="language-plaintext highlighter-rouge">quality_control_name</code><td>The name of the metric used to assess the quality of the sequencing data; in this case, the Phred quality score.<tr><td><code class="language-plaintext highlighter-rouge">quality_control_value</code><td>The actual quality score achieved, indicating the reliability of the sequencing reads; 78.33% in this context.<tr><td><code class="language-plaintext highlighter-rouge">library_prep_kit</code><td>Specifies the kit used for preparing DNA libraries for sequencing, critical for understanding the sample preparation methodology; Illumina TruSeq DNA PCR-Free is noted for high fidelity.<tr><td><code class="language-plaintext highlighter-rouge">sample_identifier</code><td>The unique identifier for the sample being sequenced, used for tracking and reference throughout the sequencing process.<tr><td><code class="language-plaintext highlighter-rouge">sample_material_type</code><td>The type of biological material from which the sample was derived, with its specific ontology code; snomed:119297000 denotes a blood sample.<tr><td><code class="language-plaintext highlighter-rouge">seq_instrument_code</code><td>The identifier for the sequencing instrument used, linking to specific equipment details; obo: OBI_0002630 refers to the Illumina NovaSeq 6000.<tr><td><code class="language-plaintext highlighter-rouge">sop_name</code><td>The name of the Standard Operating Procedure followed during the sequencing, ensuring consistency and reproducibility; here, WGS with Illumina NovaSeq 6000.<tr><td><code class="language-plaintext highlighter-rouge">sop_description</code><td>A brief description of the SOP, providing context and specifics about the sequencing approach used.<tr><td><code class="language-plaintext highlighter-rouge">sop_version</code><td>The version number of the SOP, which helps in identifying any changes or updates that might affect the sequencing output or interpretation.</table></div><h3 id="descriptions-for-fastqbam-readgroup-terms"> <a href="#descriptions-for-fastqbam-readgroup-terms" class="anchor-heading" aria-labelledby="descriptions-for-fastqbam-readgroup-terms"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Descriptions for FASTQ/BAM readgroup terms</h3><div class="table-wrapper"><table><thead><tr><th>Column Name<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">START AT</code><td>The start timestamp of the sequencing or analysis process.<tr><td><code class="language-plaintext highlighter-rouge">END AT</code><td>The end timestamp of the sequencing or analysis process.<tr><td><code class="language-plaintext highlighter-rouge">sample_read_id</code><td>A unique identifier for each sample read in the process.<tr><td><code class="language-plaintext highlighter-rouge">rel_dir</code><td>Relative directory path where sequencing data is stored.<tr><td><code class="language-plaintext highlighter-rouge">dir_id</code><td>Directory identifier combining the project and sample ID.<tr><td><code class="language-plaintext highlighter-rouge">FILE1</code><td>Path to the first FASTQ file generated by sequencing.<tr><td><code class="language-plaintext highlighter-rouge">FILE2</code><td>Path to the second FASTQ file generated by sequencing.<tr><td><code class="language-plaintext highlighter-rouge">output_file</code><td>Path to the final BAM file generated after processing.<tr><td><code class="language-plaintext highlighter-rouge">ID</code><td>Internal identifier used to track the sample in analysis.<tr><td><code class="language-plaintext highlighter-rouge">SM</code><td>Sample name or identifier used within the BAM file.<tr><td><code class="language-plaintext highlighter-rouge">PL</code><td>Sequencing platform used, indicating technology type.<tr><td><code class="language-plaintext highlighter-rouge">PU</code><td>Platform unit (PU) tag, often a barcode identifier.<tr><td><code class="language-plaintext highlighter-rouge">LB</code><td>Library ID which is crucial for distinguishing between libraries prepared differently.<tr><td><code class="language-plaintext highlighter-rouge">RG</code><td>Read group identifier in a BAM file, encapsulating all other identifiers.</table></div><h3 id="descriptions-for-genetic-variants-terms"> <a href="#descriptions-for-genetic-variants-terms" class="anchor-heading" aria-labelledby="descriptions-for-genetic-variants-terms"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Descriptions for genetic variants terms</h3><div class="table-wrapper"><table><thead><tr><th>Column Name<th>Description<tbody><tr><td><code class="language-plaintext highlighter-rouge">sample.id</code><td>Unique identifier for each sample.<tr><td><code class="language-plaintext highlighter-rouge">rownames</code><td>Row names corresponding to data entries.<tr><td><code class="language-plaintext highlighter-rouge">CHROM</code><td>Chromosome number where the variant is located.<tr><td><code class="language-plaintext highlighter-rouge">REF</code><td>Reference allele at the variant locus.<tr><td><code class="language-plaintext highlighter-rouge">ALT</code><td>Alternate allele at the variant locus.<tr><td><code class="language-plaintext highlighter-rouge">POS</code><td>Position of the variant on the chromosome.<tr><td><code class="language-plaintext highlighter-rouge">start</code><td>Start position of the variant.<tr><td><code class="language-plaintext highlighter-rouge">end</code><td>End position of the variant.<tr><td><code class="language-plaintext highlighter-rouge">width</code><td>Width of the variant region.<tr><td><code class="language-plaintext highlighter-rouge">Gene</code><td>Gene name associated with the variant.<tr><td><code class="language-plaintext highlighter-rouge">SYMBOL</code><td>Gene symbol.<tr><td><code class="language-plaintext highlighter-rouge">HGNC_ID</code><td>HUGO Gene Nomenclature Committee ID.<tr><td><code class="language-plaintext highlighter-rouge">HGVSp</code><td>Human Genome Variation Society protein nomenclature.<tr><td><code class="language-plaintext highlighter-rouge">HGVSc</code><td>Human Genome Variation Society coding DNA sequence nomenclature.<tr><td><code class="language-plaintext highlighter-rouge">Consequence</code><td>Consequence of the variant.<tr><td><code class="language-plaintext highlighter-rouge">IMPACT</code><td>Impact of the variant on the gene or protein function.<tr><td><code class="language-plaintext highlighter-rouge">genotype</code><td>Genotype showing the variant alleles.<tr><td><code class="language-plaintext highlighter-rouge">Feature_type</code><td>Type of genomic feature (e.g., transcript, regulatory).<tr><td><code class="language-plaintext highlighter-rouge">Feature</code><td>Specific feature affected by the variant (e.g., exon, intron).<tr><td><code class="language-plaintext highlighter-rouge">BIOTYPE</code><td>Biological type of the feature affected (e.g., protein_coding, miRNA).<tr><td><code class="language-plaintext highlighter-rouge">VARIANT_CLASS</code><td>Classification of the variant based on its genomic context.<tr><td><code class="language-plaintext highlighter-rouge">CANONICAL</code><td>Indicates if the transcript is the canonical transcript.<tr><td><code class="language-plaintext highlighter-rouge">CHROM_Metadata</code><td>Type: Chromosome; Cardinality: 1:1; Value Set: SNOMED CT: 91272006, LOINC:48000-4<tr><td><code class="language-plaintext highlighter-rouge">POS_Metadata</code><td>Type: Genomic Position; Cardinality: 1:1; Value Set: GENO:0000902<tr><td><code class="language-plaintext highlighter-rouge">REF_Metadata</code><td>Type: Reference Allele; Cardinality: 1:1; Value Set: string<tr><td><code class="language-plaintext highlighter-rouge">ALT_Metadata</code><td>Type: Alternate Allele; Cardinality: 1:1; Value Set: string</table></div><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
