<!DOCTYPE html><html lang="en-US"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=Edge"><link rel="stylesheet" href="/assets/css/just-the-docs-default.css"> <script src="/assets/js/vendor/lunr.min.js"></script> <script src="/assets/js/just-the-docs.js"></script><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"><link rel="manifest" href="/site.webmanifest"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#edbe7a"><link rel="shortcut icon" href="/favicon.ico"><meta name="msapplication-TileColor" content="#da532c"><meta name="msapplication-config" content="/browserconfig.xml"><meta name="theme-color" content="#ffffff"><title>VSAT with SKAT | Dev docs</title><meta name="generator" content="Jekyll v4.4.1" /><meta property="og:title" content="VSAT with SKAT" /><meta property="og:locale" content="en_US" /><meta name="description" content="Dev docs made with Jeykll using the Just the Docs theme." /><meta property="og:description" content="Dev docs made with Jeykll using the Just the Docs theme." /><link rel="canonical" href="http://localhost:4000/pages/vsat.html" /><meta property="og:url" content="http://localhost:4000/pages/vsat.html" /><meta property="og:site_name" content="Dev docs" /><meta property="og:type" content="website" /><meta name="twitter:card" content="summary" /><meta property="twitter:title" content="VSAT with SKAT" /> <script type="application/ld+json"> {"@context":"https://schema.org","@type":"WebPage","description":"Dev docs made with Jeykll using the Just the Docs theme.","headline":"VSAT with SKAT","url":"http://localhost:4000/pages/vsat.html"}</script> <script type="text/javascript" defer src="/assets/js/mathjax-script-type.js"> </script> <script type="text/javascript" id="MathJax-script" defer src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"> </script><body> <a class="skip-to-main" href="#main-content">Skip to main content</a> <svg xmlns="http://www.w3.org/2000/svg" class="d-none"> <symbol id="svg-link" viewBox="0 0 24 24"><title>Link</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-link"><path d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"></path><path d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"></path> </svg> </symbol> <symbol id="svg-menu" viewBox="0 0 24 24"><title>Menu</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-menu"><line x1="3" y1="12" x2="21" y2="12"></line><line x1="3" y1="6" x2="21" y2="6"></line><line x1="3" y1="18" x2="21" y2="18"></line> </svg> </symbol> <symbol id="svg-arrow-right" viewBox="0 0 24 24"><title>Expand</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline> </svg> </symbol> <symbol id="svg-external-link" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-external-link"><title id="svg-external-link-title">(external link)</title><path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path><polyline points="15 3 21 3 21 9"></polyline><line x1="10" y1="14" x2="21" y2="3"></line> </symbol> <symbol id="svg-doc" viewBox="0 0 24 24"><title>Document</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-file"><path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline> </svg> </symbol> <symbol id="svg-search" viewBox="0 0 24 24"><title>Search</title><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"> <circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line> </svg> </symbol> <symbol id="svg-copy" viewBox="0 0 16 16"><title>Copy</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard" viewBox="0 0 16 16"><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1h1a1 1 0 0 1 1 1V14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V3.5a1 1 0 0 1 1-1h1v-1z"/><path d="M9.5 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3zm-3-1A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3z"/> </svg> </symbol> <symbol id="svg-copied" viewBox="0 0 16 16"><title>Copied</title><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-clipboard-check-fill" viewBox="0 0 16 16"><path d="M6.5 0A1.5 1.5 0 0 0 5 1.5v1A1.5 1.5 0 0 0 6.5 4h3A1.5 1.5 0 0 0 11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3Zm3 1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5h3Z"/><path d="M4 1.5H3a2 2 0 0 0-2 2V14a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V3.5a2 2 0 0 0-2-2h-1v1A2.5 2.5 0 0 1 9.5 5h-3A2.5 2.5 0 0 1 4 2.5v-1Zm6.854 7.354-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708.708Z"/> </svg> </symbol> </svg><div class="side-bar"><div class="site-header"> <a href="/" class="site-title lh-tight"> Dev docs </a> <a href="#" id="menu-button" class="site-button"> <svg viewBox="0 0 24 24" class="icon"><use xlink:href="#svg-menu"></use></svg> </a></div><nav aria-label="Main" id="site-nav" class="site-nav"><ul class="nav-list"><li class="nav-list-item"><a href="/" class="nav-list-link">Home</a><li class="nav-list-item"><a href="/pages/bam2fastq.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/bayesian_example1.html" class="nav-list-link">Bayes 1 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_example2.html" class="nav-list-link">Bayesian 2 probability in placenta previa</a><li class="nav-list-item"><a href="/pages/bayesian_genetic_carrier.html" class="nav-list-link">Bayesian discrete probability example in genetics</a><li class="nav-list-item"><a href="/pages/bayesian_mcmc_samplers.html" class="nav-list-link">Bayes MCMC samplers</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_bioassay.html" class="nav-list-link">Bayes multiparameter bioassay demo</a><li class="nav-list-item"><a href="/pages/bayesian_multiparameter_models.html" class="nav-list-link">Bayes multiparameter models</a><li class="nav-list-item"><a href="/pages/benchmark_pipelines.html" class="nav-list-link">Benchmarking pipeline output</a><li class="nav-list-item"><a href="/pages/bevimed.html" class="nav-list-link">BeviMed</a><li class="nav-list-item"><a href="/pages/biomedit.html" class="nav-list-link">BioMedIT</a><li class="nav-list-item"><a href="/pages/bookmarks.html" class="nav-list-link">Bookmarks</a><li class="nav-list-item"><a href="/pages/bwa.html" class="nav-list-link">BWA</a><li class="nav-list-item"><a href="/pages/causal_inference_stats.html" class="nav-list-link">Causal inference stats</a><li class="nav-list-item"><a href="/pages/causal_inference_whole_game.html" class="nav-list-link">Causal inference mosquito nets</a><li class="nav-list-item"><a href="/pages/data_concepts.html" class="nav-list-link">Data concepts</a><li class="nav-list-item"><a href="/pages/data_stream.html" class="nav-list-link">Data stream</a><li class="nav-list-item"><a href="#" class="nav-list-expander" aria-label="toggle links in Design documents category"> <svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg> </a><a href="/pages/design_doc.html" class="nav-list-link">Design documents</a><ul class="nav-list"><li class="nav-list-item "><a href="/pages/design_PCA_SNV_INDEL_V1.html" class="nav-list-link">Design PCA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1.html" class="nav-list-link">Design DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v1_release.html" class="nav-list-link">Design release DNA SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_dna_snvindel_v2_release.html" class="nav-list-link">Design release DNA SNV INDEL v2</a><li class="nav-list-item "><a href="/pages/design_qv_evidence_flag.html" class="nav-list-link">Project design QV-ES</a><li class="nav-list-item "><a href="/pages/design_qv_snvindel_v1.html" class="nav-list-link">Design QV SNV INDEL v1</a><li class="nav-list-item "><a href="/pages/design_statistical_genomics_v1.html" class="nav-list-link">Design statistical genomics v1</a><li class="nav-list-item "><a href="/pages/design_quant_es.html" class="nav-list-link">Project design Quant ES</a></ul><li class="nav-list-item"><a href="/pages/dna_annotation.html" class="nav-list-link">DNA annotation</a><li class="nav-list-item"><a href="/pages/dna_interpretation.html" class="nav-list-link">DNA interpretation</a><li class="nav-list-item"><a href="/pages/dna_qc.html" class="nav-list-link">DNA QC</a><li class="nav-list-item"><a href="/pages/docker_singularity.html" class="nav-list-link">Docker with singularity</a><li class="nav-list-item"><a href="/pages/exomiser.html" class="nav-list-link">Exomiser</a><li class="nav-list-item"><a href="/pages/exomiser_phenodigm.html" class="nav-list-link">Exomiser phenodigm</a><li class="nav-list-item"><a href="/pages/fastp.html" class="nav-list-link">FASTP</a><li class="nav-list-item"><a href="/pages/fastq.html" class="nav-list-link">FASTQ format data</a><li class="nav-list-item"><a href="/pages/filter_vcf_bcftools.html" class="nav-list-link">Filter VCF with bcftools</a><li class="nav-list-item"><a href="/pages/financial_management.html" class="nav-list-link">Financial management</a><li class="nav-list-item"><a href="/pages/guru.html" class="nav-list-link">Guru variant interpretation</a><li class="nav-list-item"><a href="/pages/gwas.html" class="nav-list-link">GWAS analysis</a><li class="nav-list-item"><a href="/pages/hpc.html" class="nav-list-link">HPC infrastructure</a><li class="nav-list-item"><a href="/pages/inf_causal_metab_sem.html" class="nav-list-link">Inference of causal metabolite networks</a><li class="nav-list-item"><a href="/pages/layout.html" class="nav-list-link">Layout</a><li class="nav-list-item"><a href="/pages/mathjax.html" class="nav-list-link">MathJax config</a><li class="nav-list-item"><a href="/pages/mbdf_models.html" class="nav-list-link">MBDF models</a><li class="nav-list-item"><a href="/pages/mbdf_supervised.html" class="nav-list-link">MBDF supervised</a><li class="nav-list-item"><a href="/pages/mbdf_unsupervised.html" class="nav-list-link">MBDF unsupervised</a><li class="nav-list-item"><a href="/pages/metadata.html" class="nav-list-link">WGS metadata</a><li class="nav-list-item"><a href="/pages/metadata_users.html" class="nav-list-link">WGS metadata users</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolscounts.html" class="nav-list-link">Metrics Bcftools counts</a><li class="nav-list-item"><a href="/pages/metrics_bcftoolsstats.html" class="nav-list-link">Metrics Bcftools stats</a><li class="nav-list-item"><a href="/pages/metrics_collectwgsmetrics.html" class="nav-list-link">Metrics CollectWgsMetrics</a><li class="nav-list-item"><a href="/pages/multiblock_data_fusion.html" class="nav-list-link">Multiblock data fusion</a><li class="nav-list-item"><a href="/pages/my_voice.html" class="nav-list-link">QV - My voice</a><li class="nav-list-item"><a href="/pages/panels_disease.html" class="nav-list-link">Panels disease gene</a><li class="nav-list-item"><a href="/pages/pca_biplot_1kg.html" class="nav-list-link">PCA biplot 1000genomes</a><li class="nav-list-item"><a href="/pages/pca_features.html" class="nav-list-link">PCA features</a><li class="nav-list-item"><a href="/pages/platform_updates.html" class="nav-list-link">Platform updates</a><li class="nav-list-item"><a href="/pages/precision_med.html" class="nav-list-link">Precision Medicine Unit (PMU)</a><li class="nav-list-item"><a href="/pages/present/presentations.html" class="nav-list-link">Presentations</a><li class="nav-list-item"><a href="/pages/qv_design.html" class="nav-list-link">QV design principles</a><li class="nav-list-item"><a href="/pages/read_group.html" class="nav-list-link">Read group</a><li class="nav-list-item"><a href="/pages/ref.html" class="nav-list-link">Reference genome</a><li class="nav-list-item"><a href="/pages/rl_finte_mdp.html" class="nav-list-link">RL finite MDP</a><li class="nav-list-item"><a href="/pages/slurm_manager.html" class="nav-list-link">SLURM monitoring</a><li class="nav-list-item"><a href="/pages/slurm_sbatch.html" class="nav-list-link">SLURM sbatch headers</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_analysis_of_methods.html" class="nav-list-link">Stats Analysis of methods</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_ci_from_p.html" class="nav-list-link">Stats CI from P</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_correlation.html" class="nav-list-link">Stats Correlation, regression and repeated data</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_odds_ratios.html" class="nav-list-link">Stats Odds ratios, SE & CI</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_roc_curve.html" class="nav-list-link">Stats Receiver operating characteristic plots</a><li class="nav-list-item"><a href="/pages/stats_altman_bland_sensitivity_specificity.html" class="nav-list-link">Stats Sensitivity and specificity</a><li class="nav-list-item"><a href="/pages/storage_architecture_plan.html" class="nav-list-link">Storage architecture plan</a><li class="nav-list-item"><a href="/pages/storage_estimates.html" class="nav-list-link">Storage estimates</a><li class="nav-list-item"><a href="/pages/storage_use_sync_and_versioning.html" class="nav-list-link">Storage, usage, and Git practices</a><li class="nav-list-item"><a href="/pages/style.html" class="nav-list-link">Style page guide</a><li class="nav-list-item"><a href="/pages/style_writing.html" class="nav-list-link">Style writing guide</a><li class="nav-list-item"><a href="/pages/sv.html" class="nav-list-link">Structural variation detection</a><li class="nav-list-item"><a href="/pages/synth_data.html" class="nav-list-link">Synthetic data</a><li class="nav-list-item"><a href="/pages/variables.html" class="nav-list-link">Variables</a><li class="nav-list-item"><a href="/pages/Phoenix_Sepsis_Score_logic.html" class="nav-list-link">Sepsis score: Phoenix</a><li class="nav-list-item"><a href="/pages/vcf.html" class="nav-list-link">VCF - Variant Call Format</a><li class="nav-list-item"><a href="/pages/vcf_gvcf.html" class="nav-list-link">VCF and gVCF</a><li class="nav-list-item"><a href="/pages/virtual_panels.html" class="nav-list-link">Virtual gene panels</a><li class="nav-list-item active"><a href="/pages/vsat.html" class="nav-list-link active">VSAT with SKAT</a><li class="nav-list-item"><a href="/pages/vsat_setID.html" class="nav-list-link">SetID for VSAT</a><li class="nav-list-item"><a href="/pages/variant_concept.html" class="nav-list-link">Variant to RDF concept</a><li class="nav-list-item"><a href="/pages/acat.html" class="nav-list-link">ACAT</a><li class="nav-list-item"><a href="/pages/acmg_criteria_table_main.html" class="nav-list-link">ACMG criteria</a><li class="nav-list-item"><a href="/pages/aggregate_multiplex.html" class="nav-list-link">Aggregate multiplexed data</a><li class="nav-list-item"><a href="/pages/annotation_table.html" class="nav-list-link">Annotation table</a><li class="nav-list-item"><a href="/pages/documentation_log.html" class="nav-list-link">Documentation log</a></ul></nav><footer class="site-footer"> Maintained by <a href="https://github.com/DylanLawless">Dylan Lawless</a> for <a href="https://switzerlandomics.ch">SwitzerlandOmics.ch</a>.</footer></div><div class="main" id="top"><div id="main-header" class="main-header"><div class="search"><div class="search-input-wrap"> <input type="text" id="search-input" class="search-input" tabindex="0" placeholder="Search Dev docs" aria-label="Search Dev docs" autocomplete="off"> <label for="search-input" class="search-label"><svg viewBox="0 0 24 24" class="search-icon"><use xlink:href="#svg-search"></use></svg></label></div><div id="search-results" class="search-results"></div></div><nav aria-label="Auxiliary" class="aux-nav"><ul class="aux-nav-list"><li class="aux-nav-list-item"> <a href="https://github.com/docs-switzerlandomics.github.io" class="site-button" > Github </a><li class="aux-nav-list-item"> <a href="https://switzerlandomics.ch" class="site-button" > SwitzerlandOmics.ch </a></ul></nav></div><div id="main-content-wrap" class="main-content-wrap"><div id="main-content" class="main-content" role="main"><p>Last update: 20230425</p><h1 id="variant-set-association-testing-vsat-with-burden-testing-skat-and-skat-o"> <a href="#variant-set-association-testing-vsat-with-burden-testing-skat-and-skat-o" class="anchor-heading" aria-labelledby="variant-set-association-testing-vsat-with-burden-testing-skat-and-skat-o"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Variant Set Association Testing (VSAT) with burden testing, SKAT, and SKAT-O</h1><h2 id="introduction"> <a href="#introduction" class="anchor-heading" aria-labelledby="introduction"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Introduction</h2><p>Variant Set Association Testing (VSAT) involves statistical methods like burden testing, SKAT (Sequence Kernel Association Test), and SKAT-O (Optimal Unified SKAT), which help identify associations between groups of genetic variants and phenotypic traits. These methods are crucial for understanding the genetic basis of diseases, particularly when analyzing rare variants.</p><p>Methods for variant collapse <sup id="fnref:povysil2019rare"><a href="#fn:povysil2019rare" class="footnote" rel="footnote" role="doc-noteref">1</a></sup>, <a href="https://www.nature.com/articles/s41576-019-0177-4/">https://www.nature.com/articles/s41576-019-0177-4/</a>. One of the most important starting positions in our analysis is determined on variant collapse. Each project and biological system has unique features that determine how variants should be grouped for joint analysis. Some more context is available in this blogpost <a href="https://lawlessgenomics.com/2021/05/28/pathway_analysis.html">https://lawlessgenomics.com/2021/05/28/pathway_analysis.html</a>.</p><h2 id="historical-context-and-evolution"> <a href="#historical-context-and-evolution" class="anchor-heading" aria-labelledby="historical-context-and-evolution"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Historical context and evolution</h2><p>VSAT has evolved from simpler burden tests to more sophisticated models that can handle complex variant interactions and incorporate various data types. Key publications have shaped the field:</p><h3 id="main-papers-in-order"> <a href="#main-papers-in-order" class="anchor-heading" aria-labelledby="main-papers-in-order"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Main papers in order</h3><ul><li>Methods for detecting associations with rare variants for common diseases: application to analysis of sequence data<sup id="fnref:li2008methods"><a href="#fn:li2008methods" class="footnote" rel="footnote" role="doc-noteref">2</a></sup>.<li>A groupwise association test for rare mutations using a weighted sum statistic<sup id="fnref:madsen2009groupwise"><a href="#fn:madsen2009groupwise" class="footnote" rel="footnote" role="doc-noteref">3</a></sup>.<ul><li>Introduced a weighted sum statistic to improve the power of groupwise association tests for rare mutations.</ul><li>An evaluation of statistical approaches to rare variant analysis in genetic association studies<sup id="fnref:morris2010evaluation"><a href="#fn:morris2010evaluation" class="footnote" rel="footnote" role="doc-noteref">4</a></sup>.<li>Pooled association tests for rare variants in exon-resequencing studies<sup id="fnref:price2010pooled"><a href="#fn:price2010pooled" class="footnote" rel="footnote" role="doc-noteref">5</a></sup>.<li>Testing for an unusual distribution of rare variants<sup id="fnref:neale2011testing"><a href="#fn:neale2011testing" class="footnote" rel="footnote" role="doc-noteref">6</a></sup>.<li>Rare-Variant Association Testing for Sequencing Data with the Sequence Kernel Association Test<sup id="fnref:Wu2011Rare"><a href="#fn:Wu2011Rare" class="footnote" rel="footnote" role="doc-noteref">7</a></sup>.<ul><li>Proposed SKAT, which uses a kernel-based approach to account for both common and rare variants in association tests.</ul><li>Optimal tests for rare variant effects in sequencing association studies<sup id="fnref:Lee2012Optimal"><a href="#fn:Lee2012Optimal" class="footnote" rel="footnote" role="doc-noteref">8</a></sup>.<ul><li>Developed SKAT-O, an optimal approach that blends SKAT and burden testing methods to maximize statistical power across different genetic architectures.</ul><li>Optimal Unified Approach for Rare-Variant Association Testing with Application to Small-Sample Case-Control Whole-Exome Sequencing Studies<sup id="fnref:Lee2012Optimalunified"><a href="#fn:Lee2012Optimalunified" class="footnote" rel="footnote" role="doc-noteref">9</a></sup>.<li>Sequence Kernel Association Tests for the Combined Effect of Rare and Common Variants<sup id="fnref:IonitaLaza2013Sequence"><a href="#fn:IonitaLaza2013Sequence" class="footnote" rel="footnote" role="doc-noteref">10</a></sup>.</ul><h3 id="major-classes-of-tests"> <a href="#major-classes-of-tests" class="anchor-heading" aria-labelledby="major-classes-of-tests"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Major classes of tests</h3><ul><li>Burden/Collapsing tests<li>Supervised/Adaptive Burden/Collapsing tests<li>Variance component (similarity) based tests<li>Omnibus tests: hedge against difference scenarios</ul><p>These methods address the challenge of detecting the collective effect of multiple genetic variants, acknowledging that these effects might vary in direction and magnitude.</p><h2 id="burden-testing"> <a href="#burden-testing" class="anchor-heading" aria-labelledby="burden-testing"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Burden testing</h2><p>Burden tests aggregate the effects of multiple variants within a genomic region into a single score, which is then tested for association with the phenotype. These tests are powerful when most variants in the region contribute to the phenotype in a similar manner.</p><h2 id="skat-and-skat-o"> <a href="#skat-and-skat-o" class="anchor-heading" aria-labelledby="skat-and-skat-o"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> SKAT and SKAT-O</h2><p>Sequence Kernel Association Test (SKAT) and SKAT-Optimal (SKAT-O) are advanced statistical methods used for identifying associations between sets of genetic variants and a phenotype. They are particularly useful for complex genetic traits influenced by multiple rare variants.</p><h3 id="data-notation-and-setup"> <a href="#data-notation-and-setup" class="anchor-heading" aria-labelledby="data-notation-and-setup"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Data Notation and Setup</h3><ul><li><strong>Subjects:</strong> Let \(n\) be the number of subjects.<li><strong>Variants:</strong> Assume \(p\) variant sites are observed within a region.<li><strong>Phenotype:</strong> Denote \(y_i\) as the phenotype for subject \(i\).<li><strong>Genotypes:</strong> \(G_i = (G_{i1}, G_{i2}, \ldots, G_{ip})\) represents the genotypes for the \(p\) variants, where \(G_{ij}\) can be 0, 1, or 2 (copies of the minor allele).<li><strong>Covariates:</strong> \(X_i\) includes covariates such as age, gender, and principal components to adjust for population stratification.</ul><h3 id="skat-model"> <a href="#skat-model" class="anchor-heading" aria-labelledby="skat-model"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> SKAT Model</h3><p>The SKAT model assesses the impact of multiple genetic variants simultaneously by considering the variants’ collective effect on the phenotype. It is based on a regression model:</p>\[y_i = \beta_0 + X_i'\beta_X + G_i'\beta_G + \epsilon_i,\]<p>where:</p><ul><li>\(\beta_0\) is the intercept.<li>\(\beta_X\) are the coefficients for the covariates.<li>\(\beta_G\) are the effects of the genetic variants.<li>\(\epsilon_i\) is the error term.</ul><h3 id="kernel-function"> <a href="#kernel-function" class="anchor-heading" aria-labelledby="kernel-function"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Kernel function</h3><p>SKAT uses a kernel function to measure the genetic similarity between subjects based on their genotypes. This function is crucial as it allows SKAT to model the correlation structure among genetic variants. The kernel \(K\) is defined as:</p>\[K = G W G^T,\]<p>where \(W\) is a diagonal matrix of weights for the genetic variants. These weights can be based on variant properties such as minor allele frequency or predicted functional impact.</p><h3 id="variance-component-test"> <a href="#variance-component-test" class="anchor-heading" aria-labelledby="variance-component-test"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Variance component test</h3><p>SKAT performs a variance component test using the kernel matrix to test the null hypothesis \(H_0: \beta_G = 0\) against the alternative \(H_a: \beta_G \neq 0\). Under the null hypothesis, there is no effect of the genetic variants on the phenotype.</p><h3 id="implementation-steps"> <a href="#implementation-steps" class="anchor-heading" aria-labelledby="implementation-steps"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Implementation steps</h3><ol><li><strong>Compute the Kernel Matrix:</strong> Calculate \(K\) using the genotypes and variant weights.<li><strong>Fit the Null Model:</strong> Estimate the coefficients for the covariates without including genetic effects.<li><strong>Compute the SKAT Statistic:</strong> Use the kernel matrix to calculate the SKAT statistic, which follows a chi-squared distribution under the null hypothesis.</ol><h3 id="skat-o"> <a href="#skat-o" class="anchor-heading" aria-labelledby="skat-o"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> SKAT-O</h3><p>SKAT-Optimal (SKAT-O) combines the burden test and SKAT to maximize power. It computes a weighted sum of the burden test statistic and the SKAT statistic:</p>\[Q_{SKAT-O} = \omega Q_{Burden} + (1 - \omega) Q_{SKAT},\]<p>where \(\omega\) is a weight factor that is optimally chosen based on the data to maximize the test’s power.</p><h2 id="practical-implementation"> <a href="#practical-implementation" class="anchor-heading" aria-labelledby="practical-implementation"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Practical implementation</h2><ul><li><strong>Data Preparation:</strong> Requires a genotype matrix, phenotypic data, and potentially other covariates. Typical plink format is best.<li><strong>Statistical Testing:</strong> Uses regression-based methods to associate genetic variability with phenotypes while adjusting for covariates.<li><strong>Result Interpretation:</strong> Involves understanding p-values in the context of multiple testing, requiring adjustments such as Bonferroni (usually this) or false discovery rate (FDR) corrections.</ul><h2 id="challenges-and-considerations"> <a href="#challenges-and-considerations" class="anchor-heading" aria-labelledby="challenges-and-considerations"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Challenges and considerations</h2><ul><li><strong>Rare Variants:</strong> Both SKAT and burden tests are particularly useful for analyzing rare variants, which may be overlooked by traditional GWAS.<li><strong>Direction of Effects:</strong> Unlike burden tests, SKAT and SKAT-O can handle variants that have both protective and deleterious effects.<li><strong>Computational Complexity:</strong> The choice of kernel functions in SKAT and the integration of multiple tests in SKAT-O can increase computational demands.</ul><h2 id="conclusion"> <a href="#conclusion" class="anchor-heading" aria-labelledby="conclusion"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> Conclusion</h2><p>The development of VSAT methods like SKAT and SKAT-O represents a significant advancement in the field of genetic epidemiology, offering tools that can detect subtle and complex genetic influences on phenotypes. These tools are crucial for uncovering the genetic architecture of complex traits, especially in the context of rare genetic variants.</p><h2 id="references"> <a href="#references" class="anchor-heading" aria-labelledby="references"><svg viewBox="0 0 16 16" aria-hidden="true"><use xlink:href="#svg-link"></use></svg></a> References</h2><div class="footnotes" role="doc-endnotes"><ol><li id="fn:povysil2019rare"><p>Povysil, G. et al., 2019. Rare-variant collapsing analyses for complex traits: guidelines and applications. <em>Nature Reviews Genetics</em>, 20(12), pp.747-759. DOI: <a href="https://doi.org/10.1038/s41576-019-0177-4">10.1038/s41576-019-0177-4</a>. <a href="#fnref:povysil2019rare" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:li2008methods"><p>Li, B. and Leal, S.M., 2008. Methods for detecting associations with rare variants for common diseases: application to analysis of sequence data. <em>The American Journal of Human Genetics</em>, 83(3), pp.311-321. <a href="#fnref:li2008methods" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:madsen2009groupwise"><p>Madsen, B.E. and Browning, S.R., 2009. A groupwise association test for rare mutations using a weighted sum statistic. <em>PLoS Genet</em>, 5(2), e1000384. <a href="#fnref:madsen2009groupwise" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:morris2010evaluation"><p>Morris, A.P. and Zeggini, E., 2010. An evaluation of statistical approaches to rare variant analysis in genetic association studies. <em>Genetic epidemiology</em>, 34(2), pp.188-193. <a href="#fnref:morris2010evaluation" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:price2010pooled"><p>Price, A.L. et al., 2010. Pooled association tests for rare variants in exon-resequencing studies. <em>The American Journal of Human Genetics</em>, 86(6), pp.832-838. <a href="#fnref:price2010pooled" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:neale2011testing"><p>Neale, B.M. et al., 2011. Testing for an unusual distribution of rare variants. <em>PLoS genetics</em>, 7(3), e1001322. <a href="#fnref:neale2011testing" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:Wu2011Rare"><p>Wu, M.C. et al., 2011. Rare-Variant Association Testing for Sequencing Data with the Sequence Kernel Association Test. <em>The American Journal of Human Genetics</em>, 89(1), pp.82-93. <a href="#fnref:Wu2011Rare" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:Lee2012Optimal"><p>Lee, S., Wu, M.C., and Lin, X., 2012. Optimal tests for rare variant effects in sequencing association studies. <em>Biostatistics</em>, 13(4), pp.762-775. <a href="#fnref:Lee2012Optimal" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:Lee2012Optimalunified"><p>Lee, S. et al., 2012. Optimal Unified Approach for Rare-Variant Association Testing with Application to Small-Sample Case-Control Whole-Exome Sequencing Studies. <em>The American Journal of Human Genetics</em>, 91(2), pp.224-237. <a href="#fnref:Lee2012Optimalunified" class="reversefootnote" role="doc-backlink">&#8617;</a></p><li id="fn:IonitaLaza2013Sequence"><p>Ionita-Laza, I. et al., 2013. Sequence Kernel Association Tests for the Combined Effect of Rare and Common Variants. <em>The American Journal of Human Genetics</em>, 92(6), pp.841-853. <a href="#fnref:IonitaLaza2013Sequence" class="reversefootnote" role="doc-backlink">&#8617;</a></p></ol></div><hr><footer><div class="d-flex mt-2"></div></footer></div></div><div class="search-overlay"></div></div>
